{
 "cells": [
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Analysis of deep mutational scanning of influenza A/Aichi/1968 (H3N2) nucleoprotein in perturbed host proteostasis environments"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Anna Ponomarenko, March 2018"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Load required modules"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Using matplotlib version 2.0.2\n",
      "Using dms_tools version 1.2.2\n"
     ]
    }
   ],
   "source": [
    "import os\n",
    "import glob\n",
    "import sys\n",
    "import string\n",
    "import time\n",
    "import multiprocessing\n",
    "\n",
    "import pandas as pd\n",
    "\n",
    "basedir = os.getcwd()\n",
    "if not os.path.isdir('notebook_figures'):\n",
    "    !mkdir notebook_figures\n",
    "if not os.path.isdir('FASTQ_files'):\n",
    "    !mkdir FASTQ_files\n",
    "if not os.path.isdir('preferences'):\n",
    "    !mkdir preferences\n",
    "    \n",
    "import matplotlib\n",
    "print \"Using matplotlib version %s\" % matplotlib.__version__\n",
    "matplotlib.use(\"Pdf\")\n",
    "orange = \"#E69F00\"\n",
    "lightblue = \"#56B4E9\"\n",
    "green = \"#009E73\"\n",
    "yellow = \"#F0E442\"\n",
    "darkblue = \"#0072B2\"\n",
    "rust =  \"#D55E00\"\n",
    "purple =  \"#CC79A7\"\n",
    "from matplotlib.ticker import MaxNLocator\n",
    "from matplotlib.patches import Polygon\n",
    "from matplotlib import gridspec\n",
    "import pylab as plt\n",
    "\n",
    "from IPython.display import Image, display\n",
    "\n",
    "import dms_tools\n",
    "print \"Using dms_tools version %s\" % dms_tools.__version__\n",
    "import dms_tools.utils\n",
    "import dms_tools.file_io\n",
    "import dms_tools.plot"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Define some functions and variables used below by multiple cells"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Shared functions and variables are now defined.\n"
     ]
    }
   ],
   "source": [
    "def RunScriptDms(rundir, run_name, script_name, commands, use_sbatch, sbatch_cpus, walltime=None):\n",
    "    \"\"\"Runs a ``dmstools`` script.\n",
    "\n",
    "    *rundir* is the directory in which we run the job. Created if it does\n",
    "    not exist.\n",
    "\n",
    "    *run_name* is the name of the run, which should be a string without\n",
    "    spaces. The input file has this prefix followed by ``_infile.txt``.\n",
    "\n",
    "    *script_name* is the name of the script that we run.\n",
    "\n",
    "    *commands* is a list of command-line arguments. These arguments should appear \n",
    "    in the order in which one would place them on the command line. The elements in this\n",
    "    list are joined together by spaces, and given to *script_name*. All elements\n",
    "    should be strings.\n",
    "\n",
    "    It is assumed that the script can be run at the command line using::\n",
    "\n",
    "        script_name infile\n",
    "\n",
    "    Returns *runfailed*: *True* if run failed, and *False* otherwise.\n",
    "    \"\"\"\n",
    "    print \"Running %s for %s in directory %s...\" % (script_name, run_name, rundir)\n",
    "    currdir = os.getcwd()\n",
    "    if not os.path.isdir(rundir):\n",
    "        os.mkdir(rundir)\n",
    "    os.chdir(rundir)\n",
    "    if (not run_name) or not all([x not in string.whitespace for x in run_name]):\n",
    "        raise ValueError(\"Invalid run_name of %s\" % run_name)\n",
    "    commandline = ' '.join(['%s' % command for command in commands])\n",
    "    if use_sbatch:\n",
    "        sbatchfile = '%s.sbatch' % run_name # sbatch command file\n",
    "        jobidfile = 'sbatch_%s_jobid' % run_name # holds sbatch job id\n",
    "        jobstatusfile = 'sbatch_%s_jobstatus' % run_name # holds sbatch job status\n",
    "        joberrorsfile = 'sbatch_%s_errors' % run_name # holds sbatch job errors\n",
    "        sbatch_f = open(sbatchfile, 'w')\n",
    "        sbatch_f.write('#!/bin/sh\\n#SBATCH\\n')\n",
    "        if walltime:\n",
    "            sbatch_f.write('#PBS -l walltime=%d:00:00\\n' % walltime)\n",
    "        sbatch_f.write('%s %s' % (script_name, commandline))\n",
    "        sbatch_f.close()\n",
    "        os.system('sbatch -c %d -e %s %s > %s' % (sbatch_cpus, joberrorsfile, sbatchfile, jobidfile))\n",
    "        time.sleep(60) # short 1 minute delay\n",
    "        jobid = int(open(jobidfile).read().split()[-1])\n",
    "        nslurmfails = 0\n",
    "        while True:\n",
    "            time.sleep(60) # delay 1 minute\n",
    "            returncode = os.system('squeue -j %d > %s' % (jobid, jobstatusfile))\n",
    "            if returncode != 0:\n",
    "                nslurmfails += 1\n",
    "                if nslurmfails > 180: # error if squeue fails at least 180 consecutive times\n",
    "                    raise ValueError(\"squeue is continually failing, which means that slurm is not working on your system. Note that although this script has crashed, many of the jobs submitted via slurm may still be running. You'll want to monitor (squeue) or kill them (scancel) -- unfortunately you can't do that until slurm starts working again.\")\n",
    "                continue # we got an error while trying to run squeue\n",
    "            nslurmfails = 0\n",
    "            lines = open(jobstatusfile).readlines()\n",
    "            if len(lines) < 2:\n",
    "                break # no longer in slurm queue\n",
    "        errors = open(joberrorsfile).read().strip()\n",
    "    else:\n",
    "        errors = os.system('%s %s' % (script_name, commandline))\n",
    "    os.chdir(currdir)\n",
    "    if errors:\n",
    "        print \"ERROR running %s for %s in directory %s.\" % (script_name, run_name, rundir)\n",
    "        return True\n",
    "    else:\n",
    "        print \"Successfully completed running %s for %s in directory %s.\" % (script_name, run_name, rundir)\n",
    "        return False\n",
    "\n",
    "    \n",
    "def RunProcesses(processes, nmultiruns):\n",
    "    \"\"\"Runs a list *multiprocessing.Process* processes.\n",
    "\n",
    "    *processes* is a list of *multiprocessing.Process* objects that\n",
    "    have not yet been started. If an empty list, then just returns \n",
    "    with no action taken.\n",
    "\n",
    "    *nmultiruns* is an integer >= 1 indicating the number of simultaneous\n",
    "    processes to run.\n",
    "\n",
    "    Runs the processes in *processes*, making sure to never have more than\n",
    "    *nmultiruns* running at a time. If any of the processes fail (return\n",
    "    an exitcode with a boolean value other than *False*), an exception\n",
    "    is raised immediately. Otherwise, this function finishes when all\n",
    "    processes have completed.\n",
    "    \"\"\"\n",
    "    if not processes:\n",
    "        return\n",
    "    if not (nmultiruns >= 1 and isinstance(nmultiruns, int)):\n",
    "        raise ValueError(\"nmultiruns must be an integer >= 1\")\n",
    "    processes_started = [False] * len(processes)\n",
    "    processes_running = [False] * len(processes)\n",
    "    processes_finished = [False] * len(processes)\n",
    "    while not all(processes_finished):\n",
    "        if (processes_running.count(True) < nmultiruns) and not all(processes_started):\n",
    "            i = processes_started.index(False)\n",
    "            processes[i].start()\n",
    "            processes_started[i] = True\n",
    "            processes_running[i] = True\n",
    "        for i in range(len(processes)):\n",
    "            if processes_running[i]:\n",
    "                if not processes[i].is_alive():\n",
    "                    processes_running[i] = False\n",
    "                    processes_finished[i] = True\n",
    "                    if processes[i].exitcode:\n",
    "                        raise IOError(\"One of the processes failed to complete.\")\n",
    "        time.sleep(60)\n",
    "\n",
    "max_cpus = 1\n",
    "## Do we use sbatch when appropriate? Some simple operations will still\n",
    "## be run without sbatch even when True\n",
    "#use_sbatch = False\n",
    "\n",
    "# Specify reference sequence to map reads to, numbering must follow what is specified by alignspecs\n",
    "refseq = '%s/Aichi_NP_reference.fasta' % basedir\n",
    "\n",
    "# Specify the biological replicates and sample names. Sample names are stored as list of 2-tuples or in a dictionary\n",
    "replicates = ['replicate-1', 'replicate-2', 'replicate-3']\n",
    "replicate_sample = [('replicate-1', 'wt_plasmid'),('replicate-1', 'mutant_plasmid'),('replicate-1', 'p0_wt'),('replicate-1', 'p0_mutant'),('replicate-1', 'wt_37C'),('replicate-1', 'basal_37C'),('replicate-1', 'HSF1i_37C'), ('replicate-1', 'Hsp90i_37C'),('replicate-1', 'wt_39C'),('replicate-1', 'basal_39C'),('replicate-1', 'HSF1i_39C'), ('replicate-1', 'Hsp90i_39C'),('replicate-2', 'wt_plasmid'),('replicate-2', 'mutant_plasmid'),('replicate-2', 'p0_wt'),('replicate-2', 'p0_mutant'),('replicate-2', 'wt_37C'),('replicate-2', 'basal_37C'),('replicate-2', 'HSF1i_37C'), ('replicate-2', 'Hsp90i_37C'),('replicate-2', 'wt_39C'),('replicate-2', 'basal_39C'),('replicate-2', 'HSF1i_39C'), ('replicate-2', 'Hsp90i_39C'),('replicate-3', 'wt_plasmid'),('replicate-3', 'mutant_plasmid'),('replicate-3', 'p0_wt'),('replicate-3', 'p0_mutant'),('replicate-3', 'wt_37C'),('replicate-3', 'basal_37C'),('replicate-3', 'HSF1i_37C'), ('replicate-3', 'Hsp90i_37C'),('replicate-3', 'wt_39C'),('replicate-3', 'basal_39C'),('replicate-3', 'HSF1i_39C'), ('replicate-3', 'Hsp90i_39C')]\n",
    "\n",
    "replicate_sample_dict = dict([ (replicate, []) for replicate in replicates])\n",
    "for replicate, sample in replicate_sample:\n",
    "    replicate_sample_dict[replicate].append(sample)\n",
    "        \n",
    "print(\"Shared functions and variables are now defined.\")"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Create reference file"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Writing /Users/annapon/Documents/NP_DMS_2018/2018_NP_DMS/Aichi_NP_reference.fasta\n"
     ]
    }
   ],
   "source": [
    "%%writefile $refseq\n",
    ">gi|392340231|gb|CY121120.1| Influenza A virus (A/Aichi/2/1968(H3N2)) nucleocapsid protein (NP) gene, complete cds\n",
    "ATGGCGTCCCAAGGCACCAAACGGTCTTATGAACAGATGGAAACTGATGGGGAACGCCAGAATGCAACTGAGATCAGAGCATCCGTCGGGAAGATGATTGATGGAAT\n",
    "TGGACGATTCTACATCCAAATGTGCACTGAACTTAAACTCAGTGATTATGAGGGGCGACTGATCCAGAAC\n",
    "AGCTTAACAATAGAGAGAATGGTGCTCTCTGCTTTTGACGAAAGAAGGAATAAATATCTGGAAGAACATC\n",
    "CCAGCGCGGGGAAGGATCCTAAGAAAACTGGAGGACCCATATACAAGAGAGTAGATAGAAAGTGGATGAG\n",
    "GGAACTCGTCCTTTATGACAAAGAAGAAATAAGGCGAATCTGGCGCCAAGCCAATAATGGTGATGATGCA\n",
    "ACAGCTGGTCTGACTCACATGATGATCTGGCATTCCAATTTGAATGATACAACATACCAGAGGACAAGAG\n",
    "CTCTTGTTCGCACCGGCATGGATCCCAGGATGTGCTCTCTGATGCAGGGTTCGACTCTCCCTAGGAGGTC\n",
    "TGGAGCTGCAGGCGCTGCAGTCAAAGGAGTTGGGACAATGGTGATGGAGTTGATAAGGATGATCAAACGT\n",
    "GGGATCAATGATCGGAACTTCTGGAGAGGTGAAAATGGACGAAAAACAAGGAGTGCTTACGAGAGAATGT\n",
    "GCAACATTCTCAAAGGAAAATTTCAAACAGCTGCACAAAGGGCAATGATGGATCAAGTGAGAGAAAGTCG\n",
    "GAACCCAGGAAATGCTGAGATCGAAGATCTCATCTTTCTGGCACGGTCTGCACTCATATTGAGAGGGTCA\n",
    "GTTGCTCACAAATCTTGTCTGCCCGCCTGTGTGTATGGACCTGCCGTAGCCAGTGGCTACGACTTCGAAA\n",
    "AAGAGGGATACTCTTTAGTGGGAATAGACCCTTTCAAACTGCTTCAAAACAGCCAAGTATACAGCCTAAT\n",
    "CAGACCGAACGAGAATCCAGCACACAAGAGTCAGCTGGTGTGGATGGCATGCAATTCTGCTGCATTTGAA\n",
    "GATCTAAGAGTATTAAGCTTCATCAGAGGGACCAAAGTATCCCCAAGGGGGAAACTTTCCACTAGAGGAG\n",
    "TACAAATTGCTTCAAATGAAAACATGGATGCTATGGAATCAAGTACTCTTGAACTGAGAAGCAGGTACTG\n",
    "GGCCATAAGAACCAGAAGTGGAGGAAACACTAATCAACAGAGGGCCTCTGCAGGTCAAATCAGTGTGCAA\n",
    "CCTGCATTTTCTGTGCAAAGAAACCTCCCATTTGACAAACCAACCATCATGGCAGCATTCACTGGGAATA\n",
    "CAGAGGGAAGAACATCAGACATGAGGGCAGAAATTATAAGGATGATGGAAGGTGCAAAACCAGAAGAAAT\n",
    "GTCCTTCCAGGGGCGGGGAGTCTTCGAGCTCTCGGACGAAAGGGCAGCGAACCCGATCGTGCCCTCTTTT\n",
    "GACATGAGTAATGAAGGATCTTATTTCTTCGGAGACAATGCAGAGGAGTACGACAATTAA\n"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Process and analyze deep sequencing data"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Align paired-end reads"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Found FASTQ_path\n",
      "Skipping barcodedsubamplicons section of masterscript\n"
     ]
    }
   ],
   "source": [
    "def main():\n",
    "    \"\"\"Main body of script.\"\"\"\n",
    "\n",
    "    # Specify R1 and R2 read lengths and gene range of each subamplicon for dms_barcodedsubamplicons\n",
    "    alignspecs = '1,252,34,32 253,501,38,30 502,750,30,36 751,1002,32,41 1003,1251,29,34 1252,1497,34,41'\n",
    "\n",
    "    # Specify range of residues to consider\n",
    "    sitesrange = ' '.join(['%s' %i for i in range(1,499)]) # range of residues to consider\n",
    "\n",
    "    # Set R1 and R2 trim lengths for dms_barcodedsubamplicons\n",
    "    R1trimlength = 200\n",
    "    R2trimlength = 160\n",
    "\n",
    "    # Group all barcodes with this many reads or more for dms_summarizealignments\n",
    "    maxperbarcode = 4\n",
    " \n",
    "    # Directory path for FASTQ files for each replicate and amplicon\n",
    "    FASTQ_path = '%s/FASTQ_files/' % basedir\n",
    "    \n",
    "    print \"Found FASTQ_path\"\n",
    "\n",
    "    # Toggle which scripts to run or skip:\n",
    "    # however, this script is not smart enough to know if the existing output \n",
    "    # exists necessary to run the next strip exist - so use with caution!\n",
    "    run_barcodedsubamplicons = False\n",
    "    run_summarizealignments = True\n",
    "    run_inferdiffprefs = False\n",
    "    run_diffprefs_editsites = False\n",
    "    run_diffprefs_merge = False\n",
    "    run_diffprefs_correlate = False\n",
    "    run_diffprefs_logoplot = False\n",
    "    #########################################\n",
    "    #########################################\n",
    "    # Now run dms_barcodedsubamplicons\n",
    "    # Each sample is run its own subdirectory\n",
    "    #########################################\n",
    "    #########################################\n",
    "    if run_barcodedsubamplicons:\n",
    "        print \"entering barcodedsubamplicons section of masterscript\"\n",
    "\n",
    "        processes = []\n",
    "\n",
    "        # Store (r1files, r2files) in dictionary keyed by (replicate, sample)\n",
    "        fastqfiles = {} \n",
    "        for replicate, sample in replicate_sample:\n",
    "            r1files = sorted(glob.glob('%s/%s/%s/%s_1.fastq' % (FASTQ_path, replicate, sample, sample)))\n",
    "            r2files = sorted(glob.glob('%s/%s/%s/%s_2.fastq' % (FASTQ_path, replicate, sample, sample)))\n",
    "            assert r1files and r2files\n",
    "            fastqfiles[(replicate, sample)] = (r1files, r2files)\n",
    "            #print replicate, sample, r1files, r2files\n",
    "\n",
    "            if not os.path.isdir(replicate):\n",
    "                os.mkdir(replicate)\n",
    "            if not os.path.isdir(\"%s/%s\" % (replicate, sample)):\n",
    "                os.mkdir(\"%s/%s\" % (replicate, sample))\n",
    "\n",
    "        # Make the barcoded alignments for each sample\n",
    "        for replicate, sample in replicate_sample:\n",
    "            subdir = '%s/%s' % (replicate, sample)\n",
    "            alignprefix = '%s_' % sample\n",
    "            (r1files, r2files) = fastqfiles[(replicate, sample)]\n",
    "            commands = ['--R1trimlength %s' % R1trimlength, '--R2trimlength %s' % R2trimlength, alignprefix, refseq, ','.join(r1files), ','.join(r2files), alignspecs]\n",
    "\n",
    "            # Add this makealignments.py call to the list of processes.\n",
    "            # Use *list* to make a copy of *commands* since lists are mutable on subsequent runs\n",
    "            # of making replicate-amplicon combinations.\n",
    "            processes.append(multiprocessing.Process(target=RunScriptDms, args=(subdir, 'barcodedsubamplicons', 'dms_barcodedsubamplicons', list(commands), use_sbatch, 1)))\n",
    "    \n",
    "\n",
    "        # run ALL the barcodedsubamplicons processes\n",
    "        RunProcesses(processes, nmultiruns=max_cpus)\n",
    "\n",
    "        # end barcodedsubamplicons block of masterscript\n",
    "    else:\n",
    "        print \"Skipping barcodedsubamplicons section of masterscript\"\n",
    "        \n",
    "if __name__ == '__main__':\n",
    "    main() # run the script"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Summarize alignments"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 10,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "['--maxperbarcode 4', 'alignmentsummary_', 'barcodedsubamplicons', 'wt_plasmid/wt_plasmid_,wt_plasmid mutant_plasmid/mutant_plasmid_,mutant_plasmid p0_wt/p0_wt_,p0_wt p0_mutant/p0_mutant_,p0_mutant wt_37C/wt_37C_,wt_37C basal_37C/basal_37C_,basal_37C HSF1i_37C/HSF1i_37C_,HSF1i_37C Hsp90i_37C/Hsp90i_37C_,Hsp90i_37C wt_39C/wt_39C_,wt_39C basal_39C/basal_39C_,basal_39C HSF1i_39C/HSF1i_39C_,HSF1i_39C Hsp90i_39C/Hsp90i_39C_,Hsp90i_39C']\n",
      "Running dms_summarizealignments for summarizealignments in directory replicate-1...\n",
      "Successfully completed running dms_summarizealignments for summarizealignments in directory replicate-1.\n",
      "['--maxperbarcode 4', 'alignmentsummary_', 'barcodedsubamplicons', 'wt_plasmid/wt_plasmid_,wt_plasmid mutant_plasmid/mutant_plasmid_,mutant_plasmid p0_wt/p0_wt_,p0_wt p0_mutant/p0_mutant_,p0_mutant wt_37C/wt_37C_,wt_37C basal_37C/basal_37C_,basal_37C HSF1i_37C/HSF1i_37C_,HSF1i_37C Hsp90i_37C/Hsp90i_37C_,Hsp90i_37C wt_39C/wt_39C_,wt_39C basal_39C/basal_39C_,basal_39C HSF1i_39C/HSF1i_39C_,HSF1i_39C Hsp90i_39C/Hsp90i_39C_,Hsp90i_39C']\n",
      "Running dms_summarizealignments for summarizealignments in directory replicate-2...\n",
      "Successfully completed running dms_summarizealignments for summarizealignments in directory replicate-2.\n",
      "['--maxperbarcode 4', 'alignmentsummary_', 'barcodedsubamplicons', 'wt_plasmid/wt_plasmid_,wt_plasmid mutant_plasmid/mutant_plasmid_,mutant_plasmid p0_wt/p0_wt_,p0_wt p0_mutant/p0_mutant_,p0_mutant wt_37C/wt_37C_,wt_37C basal_37C/basal_37C_,basal_37C HSF1i_37C/HSF1i_37C_,HSF1i_37C Hsp90i_37C/Hsp90i_37C_,Hsp90i_37C wt_39C/wt_39C_,wt_39C basal_39C/basal_39C_,basal_39C HSF1i_39C/HSF1i_39C_,HSF1i_39C Hsp90i_39C/Hsp90i_39C_,Hsp90i_39C']\n",
      "Running dms_summarizealignments for summarizealignments in directory replicate-3...\n",
      "Successfully completed running dms_summarizealignments for summarizealignments in directory replicate-3.\n"
     ]
    }
   ],
   "source": [
    "def main():\n",
    "    \"\"\"Main body of script.\"\"\"\n",
    "\n",
    "    # Group all barcodes with this many reads or more for dms_summarizealignments\n",
    "    maxperbarcode = 4\n",
    "\n",
    "    # Directory path for FASTQ files for each replicate and amplicon\n",
    "    FASTQ_path = '%s/FASTQ_files/' % basedir\n",
    "\n",
    "\n",
    "    # Run summarizealignments on individual samples\n",
    "    for replicate, sample in replicate_sample:\n",
    "        subdir = '%s/%s' % (replicate, sample)\n",
    "        alignprefix_name = '%s_,%s' % (sample, sample)\n",
    "        outprefix = 'alignmentsummary_'\n",
    "        commands = ['--maxperbarcode %s' % maxperbarcode, outprefix, 'barcodedsubamplicons', alignprefix_name]            #print commands\n",
    "#        RunScriptDms(subdir, 'summarizealignments', 'dms_summarizealignments', commands, False, 1)\n",
    "\n",
    "    # Run summarizealignments on all samples from a replicate     \n",
    "    for replicate in replicates:\n",
    "        alignprefixes_names = []\n",
    "        subdir = replicate\n",
    "        outprefix = 'alignmentsummary_'\n",
    "        for sample in replicate_sample_dict[replicate]:\n",
    "            if replicate == 'replicate-1' and sample != 'DNA':   # only 7 alignments can be summarized currently\n",
    "                alignprefix = '%s/%s_' % (sample, sample)\n",
    "                alignprefixes_names.append(\"%s,%s\" % (alignprefix, sample))\n",
    "            if replicate != 'replicate-1':\n",
    "                alignprefix = '%s/%s_' % (sample, sample)\n",
    "                alignprefixes_names.append(\"%s,%s\" % (alignprefix, sample))\n",
    "        commands = ['--maxperbarcode %s' % maxperbarcode, outprefix, 'barcodedsubamplicons', ' '.join(alignprefixes_names)]\n",
    "        print commands\n",
    "        RunScriptDms(subdir, 'summarizealignments', 'dms_summarizealignments', commands, False, 1)\n",
    "        \n",
    "if __name__ == '__main__':\n",
    "    main() # run the script"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Plot mutational and error rates, depth of coverage"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "/bin/sh: convert: command not found\r\n"
     ]
    },
    {
     "data": {
      "image/png": "cmVwbGljYXRlLTEvYWxpZ25tZW50c3VtbWFyeV9iYXJjb2Rlcy5wbmc=\n",
      "text/plain": [
       "<IPython.core.display.Image object>"
      ]
     },
     "metadata": {
      "image/png": {
       "width": 500
      }
     },
     "output_type": "display_data"
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "/bin/sh: convert: command not found\r\n"
     ]
    },
    {
     "data": {
      "image/png": "cmVwbGljYXRlLTEvYWxpZ25tZW50c3VtbWFyeV9kZXB0aC5wbmc=\n",
      "text/plain": [
       "<IPython.core.display.Image object>"
      ]
     },
     "metadata": {
      "image/png": {
       "width": 500
      }
     },
     "output_type": "display_data"
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "/bin/sh: convert: command not found\r\n"
     ]
    },
    {
     "data": {
      "image/png": "cmVwbGljYXRlLTEvYWxpZ25tZW50c3VtbWFyeV9tdXRjb3VudHNfYWxsLnBuZw==\n",
      "text/plain": [
       "<IPython.core.display.Image object>"
      ]
     },
     "metadata": {
      "image/png": {
       "width": 500
      }
     },
     "output_type": "display_data"
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "/bin/sh: convert: command not found\r\n"
     ]
    },
    {
     "data": {
      "image/png": "cmVwbGljYXRlLTEvYWxpZ25tZW50c3VtbWFyeV9tdXRjb3VudHNfbXVsdGlfbnQucG5n\n",
      "text/plain": [
       "<IPython.core.display.Image object>"
      ]
     },
     "metadata": {
      "image/png": {
       "width": 500
      }
     },
     "output_type": "display_data"
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "/bin/sh: convert: command not found\r\n"
     ]
    },
    {
     "data": {
      "image/png": "cmVwbGljYXRlLTEvYWxpZ25tZW50c3VtbWFyeV9tdXRkZXB0aC5wbmc=\n",
      "text/plain": [
       "<IPython.core.display.Image object>"
      ]
     },
     "metadata": {
      "image/png": {
       "width": 500
      }
     },
     "output_type": "display_data"
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "/bin/sh: convert: command not found\r\n"
     ]
    },
    {
     "data": {
      "image/png": "cmVwbGljYXRlLTEvYWxpZ25tZW50c3VtbWFyeV9tdXRmcmVxcy5wbmc=\n",
      "text/plain": [
       "<IPython.core.display.Image object>"
      ]
     },
     "metadata": {
      "image/png": {
       "width": 500
      }
     },
     "output_type": "display_data"
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "/bin/sh: convert: command not found\r\n"
     ]
    },
    {
     "data": {
      "image/png": "cmVwbGljYXRlLTEvYWxpZ25tZW50c3VtbWFyeV9yZWFkcy5wbmc=\n",
      "text/plain": [
       "<IPython.core.display.Image object>"
      ]
     },
     "metadata": {
      "image/png": {
       "width": 500
      }
     },
     "output_type": "display_data"
    }
   ],
   "source": [
    "from IPython.display import Image, display\n",
    "\n",
    "filepath = 'replicate-1/alignmentsummary_barcodes.pdf'\n",
    "\n",
    "png = filepath.rstrip('.pdf') + '.png'\n",
    "!convert -density 192 -trim $filepath $png\n",
    "display(Image(png, width=500))\n",
    "\n",
    "filepath = 'replicate-1/alignmentsummary_depth.pdf'\n",
    "\n",
    "png = filepath.rstrip('.pdf') + '.png'\n",
    "!convert -density 192 -trim $filepath $png\n",
    "display(Image(png, width=500))\n",
    "\n",
    "filepath = 'replicate-1/alignmentsummary_mutcounts_all.pdf'\n",
    "\n",
    "png = filepath.rstrip('.pdf') + '.png'\n",
    "!convert -density 192 -trim $filepath $png\n",
    "display(Image(png, width=500))\n",
    "\n",
    "filepath = 'replicate-1/alignmentsummary_mutcounts_multi_nt.pdf'\n",
    "\n",
    "png = filepath.rstrip('.pdf') + '.png'\n",
    "!convert -density 192 -trim $filepath $png\n",
    "display(Image(png, width=500))\n",
    "\n",
    "filepath = 'replicate-1/alignmentsummary_mutdepth.pdf'\n",
    "\n",
    "png = filepath.rstrip('.pdf') + '.png'\n",
    "!convert -density 192 -trim $filepath $png\n",
    "display(Image(png, width=500))\n",
    "\n",
    "filepath = 'replicate-1/alignmentsummary_mutfreqs.pdf'\n",
    "\n",
    "png = filepath.rstrip('.pdf') + '.png'\n",
    "!convert -density 192 -trim $filepath $png\n",
    "display(Image(png, width=500))\n",
    "\n",
    "filepath = 'replicate-1/alignmentsummary_reads.pdf'\n",
    "\n",
    "png = filepath.rstrip('.pdf') + '.png'\n",
    "!convert -density 192 -trim $filepath $png\n",
    "display(Image(png, width=500))"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Calculate differential selection for each mutation and each site in NP"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Run dms_diffselection on all perturbed proteostasis environments relative to basal conditions at 37 C for each of three replicates."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 25,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "/Users/annapon/Documents/NP_DMS_2018/2018_NP_DMS/preferences\n",
      "\n",
      "Beginning execution of dms_diffselection in directory /Users/annapon/Documents/NP_DMS_2018/2018_NP_DMS/preferences\n",
      "\n",
      "Version information for dms_tools and associated packages.\n",
      "\tTime and date: Sat Mar 24 16:16:27 2018\n",
      "\tPlatform: Darwin-15.6.0-x86_64-i386-64bit\n",
      "\tPython version: 2.7.13 |Anaconda 4.4.0 (x86_64)| (default, Dec 20 2016, 23:05:08)  [GCC 4.2.1 Compatible Apple LLVM 6.0 (clang-600.0.57)]\n",
      "\tdms_tools version: 1.2.2\n",
      "\tBio version: 1.70\n",
      "\tnumpy version: 1.12.1\n",
      "\tscipy version: 0.19.0\n",
      "\tmatplotlib version: 2.0.2\n",
      "\tcython version: 0.25.2\n",
      "\tpystan version: 2.16.0.0\n",
      "\tweblogolib version: 3.5.0\n",
      "\tPyPDF2 version: 1.26.0\n",
      "\tpandas version: 0.20.1\n",
      "Parsed the following arguments:\n",
      "\tchartype = codon_to_aa\n",
      "\tpseudocount = 10\n",
      "\tmincounts = 0\n",
      "\terrorcontrolcounts = None\n",
      "\tmockcounts = Rep1-DMSO37_counts.txt\n",
      "\tno_scale_pseudocounts = False\n",
      "\toutprefix = HSF1i37vsDMSO37-rep1_\n",
      "\tselectedcounts = Rep1-HSF1i37_counts.txt\n",
      "\tincludestop = False\n",
      "\n",
      "\n",
      "Read counts data for 499 sites.\n",
      "\n",
      "Mutations with the lowest differential selection:\n",
      "site wt mut  diffsel\n",
      " 15  T   P    -1.36\n",
      "497  D   I    -1.23\n",
      " 72  D   N    -1.20\n",
      "233  A   Q    -1.17\n",
      "128  D   T    -1.14\n",
      "404  G   R    -1.13\n",
      " 82  H   F    -1.13\n",
      " 37  G   E    -1.05\n",
      "200  G   E    -1.04\n",
      "140  H   P    -1.04\n",
      "\n",
      "Mutations with the highest differential selection:\n",
      "site wt mut  diffsel\n",
      "382  R   N     3.00\n",
      "395  N   G     2.26\n",
      " 63  I   D     1.53\n",
      "458  F   H     1.41\n",
      "446  R   P     1.40\n",
      "443  E   N     1.36\n",
      "492  N   A     1.22\n",
      " 77  K   V     1.09\n",
      "497  D   C     1.08\n",
      "238  M   I     1.08\n",
      "\n",
      "Sites with greatest total absolute differential selection:\n",
      "site  abs_diffsel  positive_diffsel  negative_diffsel\n",
      "497         9.25              2.95             -6.31\n",
      "431         8.15              2.90             -5.25\n",
      "450         7.62              4.13             -3.49\n",
      " 34         6.85              4.51             -2.34\n",
      "446         6.77              3.29             -3.47\n",
      "395         6.74              4.04             -2.70\n",
      " 81         6.73              1.65             -5.08\n",
      "409         6.69              1.20             -5.49\n",
      "421         6.61              3.03             -3.58\n",
      "294         6.53              1.43             -5.10\n",
      "\n",
      "Writing mutation differential selection to HSF1i37vsDMSO37-rep1_mutdiffsel.txt\n",
      "\n",
      "Writing site differential selection to HSF1i37vsDMSO37-rep1_sitediffsel.txt\n",
      "\n",
      "Program complete.\n"
     ]
    }
   ],
   "source": [
    "#REPLICATE 1: -/+ HSF1i 37C\n",
    "%cd preferences\n",
    "\n",
    "!dms_diffselection Rep1-DMSO37_counts.txt Rep1-HSF1i37_counts.txt HSF1i37vsDMSO37-rep1_"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 26,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[Errno 2] No such file or directory: 'preferences'\n",
      "/Users/annapon/Documents/NP_DMS_2018/2018_NP_DMS/preferences\n",
      "\n",
      "Beginning execution of dms_diffselection in directory /Users/annapon/Documents/NP_DMS_2018/2018_NP_DMS/preferences\n",
      "\n",
      "Version information for dms_tools and associated packages.\n",
      "\tTime and date: Sat Mar 24 16:17:23 2018\n",
      "\tPlatform: Darwin-15.6.0-x86_64-i386-64bit\n",
      "\tPython version: 2.7.13 |Anaconda 4.4.0 (x86_64)| (default, Dec 20 2016, 23:05:08)  [GCC 4.2.1 Compatible Apple LLVM 6.0 (clang-600.0.57)]\n",
      "\tdms_tools version: 1.2.2\n",
      "\tBio version: 1.70\n",
      "\tnumpy version: 1.12.1\n",
      "\tscipy version: 0.19.0\n",
      "\tmatplotlib version: 2.0.2\n",
      "\tcython version: 0.25.2\n",
      "\tpystan version: 2.16.0.0\n",
      "\tweblogolib version: 3.5.0\n",
      "\tPyPDF2 version: 1.26.0\n",
      "\tpandas version: 0.20.1\n",
      "Parsed the following arguments:\n",
      "\tchartype = codon_to_aa\n",
      "\tpseudocount = 10\n",
      "\tmincounts = 0\n",
      "\terrorcontrolcounts = None\n",
      "\tmockcounts = Rep1-DMSO37_counts.txt\n",
      "\tno_scale_pseudocounts = False\n",
      "\toutprefix = Hsp90i37vsDMSO37-rep1_\n",
      "\tselectedcounts = Rep1-Hsp90i37_counts.txt\n",
      "\tincludestop = False\n",
      "\n",
      "\n",
      "Read counts data for 499 sites.\n",
      "\n",
      "Mutations with the lowest differential selection:\n",
      "site wt mut  diffsel\n",
      "163  M   C    -1.81\n",
      "301  I   V    -1.33\n",
      "103  K   I    -1.32\n",
      "404  G   R    -1.32\n",
      "233  A   Q    -1.32\n",
      "292  E   Y    -1.30\n",
      " 82  H   F    -1.20\n",
      "171  T   A    -1.14\n",
      "126  G   A    -1.13\n",
      "257  I   S    -1.11\n",
      "\n",
      "Mutations with the highest differential selection:\n",
      "site wt mut  diffsel\n",
      "382  R   N     3.19\n",
      " 63  I   D     2.57\n",
      "446  R   P     1.85\n",
      "395  N   G     1.82\n",
      "492  N   A     1.77\n",
      "439  D   V     1.64\n",
      "457  S   H     1.57\n",
      "448  M   R     1.54\n",
      "450  G   D     1.43\n",
      " 77  K   V     1.39\n",
      "\n",
      "Sites with greatest total absolute differential selection:\n",
      "site  abs_diffsel  positive_diffsel  negative_diffsel\n",
      "451         8.09              4.85             -3.24\n",
      " 18         7.88              3.27             -4.61\n",
      "450         7.77              4.78             -2.98\n",
      "382         7.54              5.60             -1.94\n",
      "319         7.28              4.14             -3.13\n",
      "283         7.21              2.85             -4.35\n",
      "498         7.11              2.58             -4.53\n",
      " 63         7.11              5.89             -1.22\n",
      " 83         6.96              5.22             -1.75\n",
      "424         6.96              4.10             -2.86\n",
      "\n",
      "Writing mutation differential selection to Hsp90i37vsDMSO37-rep1_mutdiffsel.txt\n",
      "\n",
      "Writing site differential selection to Hsp90i37vsDMSO37-rep1_sitediffsel.txt\n",
      "\n",
      "Program complete.\n"
     ]
    }
   ],
   "source": [
    "#REPLICATE 1: -/+ Hsp90i 37C\n",
    "%cd preferences\n",
    "\n",
    "!dms_diffselection Rep1-DMSO37_counts.txt Rep1-Hsp90i37_counts.txt Hsp90i37vsDMSO37-rep1_"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 27,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[Errno 2] No such file or directory: 'preferences'\n",
      "/Users/annapon/Documents/NP_DMS_2018/2018_NP_DMS/preferences\n",
      "\n",
      "Beginning execution of dms_diffselection in directory /Users/annapon/Documents/NP_DMS_2018/2018_NP_DMS/preferences\n",
      "\n",
      "Version information for dms_tools and associated packages.\n",
      "\tTime and date: Sat Mar 24 16:17:52 2018\n",
      "\tPlatform: Darwin-15.6.0-x86_64-i386-64bit\n",
      "\tPython version: 2.7.13 |Anaconda 4.4.0 (x86_64)| (default, Dec 20 2016, 23:05:08)  [GCC 4.2.1 Compatible Apple LLVM 6.0 (clang-600.0.57)]\n",
      "\tdms_tools version: 1.2.2\n",
      "\tBio version: 1.70\n",
      "\tnumpy version: 1.12.1\n",
      "\tscipy version: 0.19.0\n",
      "\tmatplotlib version: 2.0.2\n",
      "\tcython version: 0.25.2\n",
      "\tpystan version: 2.16.0.0\n",
      "\tweblogolib version: 3.5.0\n",
      "\tPyPDF2 version: 1.26.0\n",
      "\tpandas version: 0.20.1\n",
      "Parsed the following arguments:\n",
      "\tchartype = codon_to_aa\n",
      "\tpseudocount = 10\n",
      "\tmincounts = 0\n",
      "\terrorcontrolcounts = None\n",
      "\tmockcounts = Rep1-DMSO37_counts.txt\n",
      "\tno_scale_pseudocounts = False\n",
      "\toutprefix = HSF1i39vsDMSO37-rep1_\n",
      "\tselectedcounts = Rep1-HSF1i39_counts.txt\n",
      "\tincludestop = False\n",
      "\n",
      "\n",
      "Read counts data for 499 sites.\n",
      "\n",
      "Mutations with the lowest differential selection:\n",
      "site wt mut  diffsel\n",
      " 82  H   F    -2.40\n",
      "446  R   A    -1.95\n",
      "309  N   L    -1.93\n",
      "497  D   I    -1.89\n",
      "165  S   A    -1.89\n",
      "467  S   R    -1.88\n",
      "321  N   K    -1.87\n",
      "131  A   S    -1.86\n",
      "163  M   C    -1.83\n",
      "193  L   C    -1.80\n",
      "\n",
      "Mutations with the highest differential selection:\n",
      "site wt mut  diffsel\n",
      "334  N   H     4.92\n",
      "283  P   A     3.72\n",
      "283  P   C     3.58\n",
      "456  M   H     2.98\n",
      "289  Y   R     2.98\n",
      "283  P   T     2.93\n",
      "268  G   P     2.81\n",
      "283  P   Q     2.79\n",
      "463  V   I     2.58\n",
      "427  A   S     2.53\n",
      "\n",
      "Sites with greatest total absolute differential selection:\n",
      "site  abs_diffsel  positive_diffsel  negative_diffsel\n",
      "283        30.65             30.41             -0.24\n",
      " 34        17.08             14.24             -2.84\n",
      "446        16.30              0.79            -15.51\n",
      " 89        14.52              1.18            -13.34\n",
      "334        13.93              9.44             -4.48\n",
      "466        12.75              3.78             -8.97\n",
      "294        12.71              5.89             -6.81\n",
      "497        12.70              1.41            -11.29\n",
      "455        12.66              2.94             -9.72\n",
      " 82        12.34              0.25            -12.09\n",
      "\n",
      "Writing mutation differential selection to HSF1i39vsDMSO37-rep1_mutdiffsel.txt\n",
      "\n",
      "Writing site differential selection to HSF1i39vsDMSO37-rep1_sitediffsel.txt\n",
      "\n",
      "Program complete.\n"
     ]
    }
   ],
   "source": [
    "#REPLICATE 1: -/+ HSF1i 39C\n",
    "%cd preferences\n",
    "\n",
    "!dms_diffselection Rep1-DMSO37_counts.txt Rep1-HSF1i39_counts.txt HSF1i39vsDMSO37-rep1_"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 28,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[Errno 2] No such file or directory: 'preferences'\n",
      "/Users/annapon/Documents/NP_DMS_2018/2018_NP_DMS/preferences\n",
      "\n",
      "Beginning execution of dms_diffselection in directory /Users/annapon/Documents/NP_DMS_2018/2018_NP_DMS/preferences\n",
      "\n",
      "Version information for dms_tools and associated packages.\n",
      "\tTime and date: Sat Mar 24 16:18:46 2018\n",
      "\tPlatform: Darwin-15.6.0-x86_64-i386-64bit\n",
      "\tPython version: 2.7.13 |Anaconda 4.4.0 (x86_64)| (default, Dec 20 2016, 23:05:08)  [GCC 4.2.1 Compatible Apple LLVM 6.0 (clang-600.0.57)]\n",
      "\tdms_tools version: 1.2.2\n",
      "\tBio version: 1.70\n",
      "\tnumpy version: 1.12.1\n",
      "\tscipy version: 0.19.0\n",
      "\tmatplotlib version: 2.0.2\n",
      "\tcython version: 0.25.2\n",
      "\tpystan version: 2.16.0.0\n",
      "\tweblogolib version: 3.5.0\n",
      "\tPyPDF2 version: 1.26.0\n",
      "\tpandas version: 0.20.1\n",
      "Parsed the following arguments:\n",
      "\tchartype = codon_to_aa\n",
      "\tpseudocount = 10\n",
      "\tmincounts = 0\n",
      "\terrorcontrolcounts = None\n",
      "\tmockcounts = Rep1-DMSO37_counts.txt\n",
      "\tno_scale_pseudocounts = False\n",
      "\toutprefix = Hsp90i39vsDMSO37-rep1_\n",
      "\tselectedcounts = Rep1-Hsp90i39_counts.txt\n",
      "\tincludestop = False\n",
      "\n",
      "\n",
      "Read counts data for 499 sites.\n",
      "\n",
      "Mutations with the lowest differential selection:\n",
      "site wt mut  diffsel\n",
      " 82  H   F    -2.41\n",
      "371  M   C    -1.96\n",
      "309  N   L    -1.72\n",
      " 83  P   I    -1.55\n",
      "193  L   C    -1.54\n",
      "431  G   F    -1.53\n",
      "131  A   S    -1.52\n",
      "469  E   V    -1.52\n",
      "296  Y   A    -1.51\n",
      " 89  P   A    -1.50\n",
      "\n",
      "Mutations with the highest differential selection:\n",
      "site wt mut  diffsel\n",
      "334  N   H     2.68\n",
      "283  P   C     2.36\n",
      "359  S   K     1.94\n",
      "289  Y   R     1.91\n",
      "353  S   F     1.82\n",
      "353  S   I     1.80\n",
      "353  S   L     1.68\n",
      " 39  F   Y     1.59\n",
      "268  G   S     1.58\n",
      "372  D   G     1.54\n",
      "\n",
      "Sites with greatest total absolute differential selection:\n",
      "site  abs_diffsel  positive_diffsel  negative_diffsel\n",
      "283        15.33             14.59             -0.74\n",
      " 89        11.75              0.00            -11.75\n",
      " 50        11.63              0.49            -11.14\n",
      "446        11.17              0.11            -11.06\n",
      " 83        10.75              0.88             -9.87\n",
      " 82        10.32              0.47             -9.85\n",
      "377        10.17              6.75             -3.42\n",
      "334        10.14              7.04             -3.09\n",
      "466        10.05              2.02             -8.04\n",
      " 51        10.03              0.00            -10.03\n",
      "\n",
      "Writing mutation differential selection to Hsp90i39vsDMSO37-rep1_mutdiffsel.txt\n",
      "\n",
      "Writing site differential selection to Hsp90i39vsDMSO37-rep1_sitediffsel.txt\n",
      "\n",
      "Program complete.\n"
     ]
    }
   ],
   "source": [
    "#REPLICATE 1: -/+ Hsp90i 39C\n",
    "%cd preferences\n",
    "\n",
    "!dms_diffselection Rep1-DMSO37_counts.txt Rep1-Hsp90i39_counts.txt Hsp90i39vsDMSO37-rep1_"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 29,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Beginning execution of dms_diffselection in directory /Users/annapon/Documents/NP_DMS_2018/2018_NP_DMS/preferences\n",
      "\n",
      "Version information for dms_tools and associated packages.\n",
      "\tTime and date: Sat Mar 24 16:20:09 2018\n",
      "\tPlatform: Darwin-15.6.0-x86_64-i386-64bit\n",
      "\tPython version: 2.7.13 |Anaconda 4.4.0 (x86_64)| (default, Dec 20 2016, 23:05:08)  [GCC 4.2.1 Compatible Apple LLVM 6.0 (clang-600.0.57)]\n",
      "\tdms_tools version: 1.2.2\n",
      "\tBio version: 1.70\n",
      "\tnumpy version: 1.12.1\n",
      "\tscipy version: 0.19.0\n",
      "\tmatplotlib version: 2.0.2\n",
      "\tcython version: 0.25.2\n",
      "\tpystan version: 2.16.0.0\n",
      "\tweblogolib version: 3.5.0\n",
      "\tPyPDF2 version: 1.26.0\n",
      "\tpandas version: 0.20.1\n",
      "Parsed the following arguments:\n",
      "\tchartype = codon_to_aa\n",
      "\tpseudocount = 10\n",
      "\tmincounts = 0\n",
      "\terrorcontrolcounts = None\n",
      "\tmockcounts = Rep1-DMSO37_counts.txt\n",
      "\tno_scale_pseudocounts = False\n",
      "\toutprefix = DMSO39vsDMSO37-rep1_\n",
      "\tselectedcounts = Rep1-DMSO39_counts.txt\n",
      "\tincludestop = False\n",
      "\n",
      "\n",
      "Read counts data for 499 sites.\n",
      "\n",
      "Mutations with the lowest differential selection:\n",
      "site wt mut  diffsel\n",
      " 82  H   F    -2.28\n",
      "371  M   C    -1.58\n",
      "460  G   R    -1.57\n",
      "100  V   E    -1.53\n",
      "404  G   R    -1.49\n",
      " 50  S   G    -1.46\n",
      " 57  I   L    -1.45\n",
      "309  N   L    -1.41\n",
      " 12  Q   D    -1.37\n",
      "321  N   K    -1.35\n",
      "\n",
      "Mutations with the highest differential selection:\n",
      "site wt mut  diffsel\n",
      "334  N   H     2.09\n",
      "353  S   L     1.96\n",
      "353  S   F     1.72\n",
      "283  P   C     1.68\n",
      "289  Y   R     1.56\n",
      "165  S   G     1.49\n",
      " 39  F   Y     1.47\n",
      "283  P   K     1.40\n",
      "359  S   K     1.39\n",
      "283  P   T     1.36\n",
      "\n",
      "Sites with greatest total absolute differential selection:\n",
      "site  abs_diffsel  positive_diffsel  negative_diffsel\n",
      "283        12.96             12.15             -0.81\n",
      " 89         9.93              0.13             -9.80\n",
      " 82         9.23              0.69             -8.53\n",
      " 34         8.90              6.22             -2.68\n",
      "497         8.87              2.43             -6.44\n",
      " 50         8.67              0.56             -8.11\n",
      "446         8.50              0.98             -7.52\n",
      "421         8.31              1.04             -7.27\n",
      " 51         8.12              0.39             -7.73\n",
      "100         7.99              2.05             -5.95\n",
      "\n",
      "Writing mutation differential selection to DMSO39vsDMSO37-rep1_mutdiffsel.txt\n",
      "\n",
      "Writing site differential selection to DMSO39vsDMSO37-rep1_sitediffsel.txt\n",
      "\n",
      "Program complete.\n"
     ]
    }
   ],
   "source": [
    "#REPLICATE 1: DMSO 39 vs DMSO 37\n",
    "\n",
    "!dms_diffselection Rep1-DMSO37_counts.txt Rep1-DMSO39_counts.txt DMSO39vsDMSO37-rep1_"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 30,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Beginning execution of dms_diffselection in directory /Users/annapon/Documents/NP_DMS_2018/2018_NP_DMS/preferences\n",
      "\n",
      "Version information for dms_tools and associated packages.\n",
      "\tTime and date: Sat Mar 24 16:20:54 2018\n",
      "\tPlatform: Darwin-15.6.0-x86_64-i386-64bit\n",
      "\tPython version: 2.7.13 |Anaconda 4.4.0 (x86_64)| (default, Dec 20 2016, 23:05:08)  [GCC 4.2.1 Compatible Apple LLVM 6.0 (clang-600.0.57)]\n",
      "\tdms_tools version: 1.2.2\n",
      "\tBio version: 1.70\n",
      "\tnumpy version: 1.12.1\n",
      "\tscipy version: 0.19.0\n",
      "\tmatplotlib version: 2.0.2\n",
      "\tcython version: 0.25.2\n",
      "\tpystan version: 2.16.0.0\n",
      "\tweblogolib version: 3.5.0\n",
      "\tPyPDF2 version: 1.26.0\n",
      "\tpandas version: 0.20.1\n",
      "Parsed the following arguments:\n",
      "\tchartype = codon_to_aa\n",
      "\tpseudocount = 10\n",
      "\tmincounts = 0\n",
      "\terrorcontrolcounts = None\n",
      "\tmockcounts = Rep1-HSF1i37_counts.txt\n",
      "\tno_scale_pseudocounts = False\n",
      "\toutprefix = HSF1i39vsHSF1i37-rep1_\n",
      "\tselectedcounts = Rep1-HSF1i39_counts.txt\n",
      "\tincludestop = False\n",
      "\n",
      "\n",
      "Read counts data for 499 sites.\n",
      "\n",
      "Mutations with the lowest differential selection:\n",
      "site wt mut  diffsel\n",
      "382  R   N    -3.53\n",
      "395  N   G    -2.36\n",
      "446  R   A    -2.13\n",
      " 63  I   D    -2.08\n",
      " 50  S   A    -2.01\n",
      "443  E   N    -1.99\n",
      "161  P   R    -1.99\n",
      "319  N   A    -1.92\n",
      " 73  E   L    -1.80\n",
      "193  L   C    -1.79\n",
      "\n",
      "Mutations with the highest differential selection:\n",
      "site wt mut  diffsel\n",
      "334  N   H     4.98\n",
      "283  P   A     4.16\n",
      "283  P   C     3.22\n",
      "283  P   Q     2.84\n",
      "427  A   S     2.63\n",
      "456  M   H     2.57\n",
      "268  G   P     2.53\n",
      "283  P   T     2.46\n",
      "146  T   A     2.45\n",
      "463  V   I     2.39\n",
      "\n",
      "Sites with greatest total absolute differential selection:\n",
      "site  abs_diffsel  positive_diffsel  negative_diffsel\n",
      "283        32.05             32.05              0.00\n",
      " 34        14.75             11.88             -2.86\n",
      "446        13.84              0.19            -13.64\n",
      "466        13.66              3.06            -10.60\n",
      "455        12.91              3.80             -9.11\n",
      "334        12.86             10.05             -2.81\n",
      " 89        12.85              0.00            -12.85\n",
      " 82        12.54              0.10            -12.44\n",
      "294        12.28              7.60             -4.68\n",
      " 51        12.23              0.00            -12.23\n",
      "\n",
      "Writing mutation differential selection to HSF1i39vsHSF1i37-rep1_mutdiffsel.txt\n",
      "\n",
      "Writing site differential selection to HSF1i39vsHSF1i37-rep1_sitediffsel.txt\n",
      "\n",
      "Program complete.\n"
     ]
    }
   ],
   "source": [
    "#REPLICATE 1: HSF1i 39C vs HSF1i 37C\n",
    "\n",
    "!dms_diffselection Rep1-HSF1i37_counts.txt Rep1-HSF1i39_counts.txt HSF1i39vsHSF1i37-rep1_"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 31,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Beginning execution of dms_diffselection in directory /Users/annapon/Documents/NP_DMS_2018/2018_NP_DMS/preferences\n",
      "\n",
      "Version information for dms_tools and associated packages.\n",
      "\tTime and date: Sat Mar 24 16:21:47 2018\n",
      "\tPlatform: Darwin-15.6.0-x86_64-i386-64bit\n",
      "\tPython version: 2.7.13 |Anaconda 4.4.0 (x86_64)| (default, Dec 20 2016, 23:05:08)  [GCC 4.2.1 Compatible Apple LLVM 6.0 (clang-600.0.57)]\n",
      "\tdms_tools version: 1.2.2\n",
      "\tBio version: 1.70\n",
      "\tnumpy version: 1.12.1\n",
      "\tscipy version: 0.19.0\n",
      "\tmatplotlib version: 2.0.2\n",
      "\tcython version: 0.25.2\n",
      "\tpystan version: 2.16.0.0\n",
      "\tweblogolib version: 3.5.0\n",
      "\tPyPDF2 version: 1.26.0\n",
      "\tpandas version: 0.20.1\n",
      "Parsed the following arguments:\n",
      "\tchartype = codon_to_aa\n",
      "\tpseudocount = 10\n",
      "\tmincounts = 0\n",
      "\terrorcontrolcounts = None\n",
      "\tmockcounts = Rep1-DMSO39_counts.txt\n",
      "\tno_scale_pseudocounts = False\n",
      "\toutprefix = HSF1i39vsDMSO39-rep1_\n",
      "\tselectedcounts = Rep1-HSF1i39_counts.txt\n",
      "\tincludestop = False\n",
      "\n",
      "\n",
      "Read counts data for 499 sites.\n",
      "\n",
      "Mutations with the lowest differential selection:\n",
      "site wt mut  diffsel\n",
      "497  D   L    -1.73\n",
      "165  S   A    -1.68\n",
      "424  T   N    -1.66\n",
      "422  K   S    -1.65\n",
      " 80  E   I    -1.49\n",
      "161  P   R    -1.34\n",
      "289  Y   I    -1.27\n",
      " 82  H   S    -1.26\n",
      "157  T   A    -1.22\n",
      "446  R   G    -1.19\n",
      "\n",
      "Mutations with the highest differential selection:\n",
      "site wt mut  diffsel\n",
      "283  P   A     2.80\n",
      "334  N   H     2.79\n",
      "222  M   T     2.26\n",
      "253  I   V     2.18\n",
      "453  P   D     2.07\n",
      "283  P   M     2.03\n",
      "283  P   Q     1.86\n",
      "283  P   H     1.82\n",
      "283  P   C     1.80\n",
      "463  V   I     1.70\n",
      "\n",
      "Sites with greatest total absolute differential selection:\n",
      "site  abs_diffsel  positive_diffsel  negative_diffsel\n",
      "283        19.44             18.71             -0.74\n",
      " 34         8.69              8.00             -0.70\n",
      "455         8.29              3.13             -5.15\n",
      " 80         8.12              2.05             -6.07\n",
      "424         7.85              2.49             -5.35\n",
      "450         7.82              4.61             -3.21\n",
      "294         7.81              5.22             -2.59\n",
      "421         7.68              4.32             -3.36\n",
      "466         7.44              3.16             -4.27\n",
      "334         7.40              5.30             -2.11\n",
      "\n",
      "Writing mutation differential selection to HSF1i39vsDMSO39-rep1_mutdiffsel.txt\n",
      "\n",
      "Writing site differential selection to HSF1i39vsDMSO39-rep1_sitediffsel.txt\n",
      "\n",
      "Program complete.\n"
     ]
    }
   ],
   "source": [
    "#REPLICATE 1: HSF1i 39C vs DMSO 39C\n",
    "\n",
    "!dms_diffselection Rep1-DMSO39_counts.txt Rep1-HSF1i39_counts.txt HSF1i39vsDMSO39-rep1_"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 32,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Beginning execution of dms_diffselection in directory /Users/annapon/Documents/NP_DMS_2018/2018_NP_DMS/preferences\n",
      "\n",
      "Version information for dms_tools and associated packages.\n",
      "\tTime and date: Sat Mar 24 16:27:25 2018\n",
      "\tPlatform: Darwin-15.6.0-x86_64-i386-64bit\n",
      "\tPython version: 2.7.13 |Anaconda 4.4.0 (x86_64)| (default, Dec 20 2016, 23:05:08)  [GCC 4.2.1 Compatible Apple LLVM 6.0 (clang-600.0.57)]\n",
      "\tdms_tools version: 1.2.2\n",
      "\tBio version: 1.70\n",
      "\tnumpy version: 1.12.1\n",
      "\tscipy version: 0.19.0\n",
      "\tmatplotlib version: 2.0.2\n",
      "\tcython version: 0.25.2\n",
      "\tpystan version: 2.16.0.0\n",
      "\tweblogolib version: 3.5.0\n",
      "\tPyPDF2 version: 1.26.0\n",
      "\tpandas version: 0.20.1\n",
      "Parsed the following arguments:\n",
      "\tchartype = codon_to_aa\n",
      "\tpseudocount = 10\n",
      "\tmincounts = 0\n",
      "\terrorcontrolcounts = None\n",
      "\tmockcounts = Rep2-DMSO37_counts.txt\n",
      "\tno_scale_pseudocounts = False\n",
      "\toutprefix = HSF1i37vsDMSO37-rep2_\n",
      "\tselectedcounts = Rep2-HSF1i37_counts.txt\n",
      "\tincludestop = False\n",
      "\n",
      "\n",
      "Read counts data for 499 sites.\n",
      "\n",
      "Mutations with the lowest differential selection:\n",
      "site wt mut  diffsel\n",
      " 43  M   N    -1.85\n",
      "173  P   N    -1.43\n",
      "460  G   K    -1.41\n",
      " 29  V   P    -1.41\n",
      "397  N   Y    -1.25\n",
      " 79  L   R    -1.21\n",
      "418  L   C    -1.20\n",
      "189  M   G    -1.19\n",
      " 51  D   V    -1.17\n",
      "129  A   I    -1.14\n",
      "\n",
      "Mutations with the highest differential selection:\n",
      "site wt mut  diffsel\n",
      "455  E   H     1.08\n",
      " 60  S   L     1.07\n",
      "459  Q   N     1.06\n",
      " 73  E   R     1.06\n",
      " 27  A   S     1.05\n",
      "409  Q   H     1.04\n",
      " 13  M   S     0.94\n",
      "424  T   Y     0.93\n",
      "392  S   G     0.92\n",
      " 59  N   F     0.91\n",
      "\n",
      "Sites with greatest total absolute differential selection:\n",
      "site  abs_diffsel  positive_diffsel  negative_diffsel\n",
      "397         7.72              2.32             -5.40\n",
      " 58         7.69              0.58             -7.11\n",
      " 59         7.62              2.35             -5.27\n",
      "429         7.61              3.37             -4.24\n",
      "438         7.18              1.21             -5.97\n",
      "129         7.09              0.68             -6.41\n",
      "401         6.94              2.22             -4.72\n",
      " 80         6.85              2.66             -4.19\n",
      " 82         6.82              0.43             -6.39\n",
      "101         6.75              2.41             -4.34\n",
      "\n",
      "Writing mutation differential selection to HSF1i37vsDMSO37-rep2_mutdiffsel.txt\n",
      "\n",
      "Writing site differential selection to HSF1i37vsDMSO37-rep2_sitediffsel.txt\n",
      "\n",
      "Program complete.\n"
     ]
    }
   ],
   "source": [
    "#REPLICATE 2: -/+ HSF1i 37C\n",
    "\n",
    "!dms_diffselection Rep2-DMSO37_counts.txt Rep2-HSF1i37_counts.txt HSF1i37vsDMSO37-rep2_"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 33,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Beginning execution of dms_diffselection in directory /Users/annapon/Documents/NP_DMS_2018/2018_NP_DMS/preferences\n",
      "\n",
      "Version information for dms_tools and associated packages.\n",
      "\tTime and date: Sat Mar 24 16:28:41 2018\n",
      "\tPlatform: Darwin-15.6.0-x86_64-i386-64bit\n",
      "\tPython version: 2.7.13 |Anaconda 4.4.0 (x86_64)| (default, Dec 20 2016, 23:05:08)  [GCC 4.2.1 Compatible Apple LLVM 6.0 (clang-600.0.57)]\n",
      "\tdms_tools version: 1.2.2\n",
      "\tBio version: 1.70\n",
      "\tnumpy version: 1.12.1\n",
      "\tscipy version: 0.19.0\n",
      "\tmatplotlib version: 2.0.2\n",
      "\tcython version: 0.25.2\n",
      "\tpystan version: 2.16.0.0\n",
      "\tweblogolib version: 3.5.0\n",
      "\tPyPDF2 version: 1.26.0\n",
      "\tpandas version: 0.20.1\n",
      "Parsed the following arguments:\n",
      "\tchartype = codon_to_aa\n",
      "\tpseudocount = 10\n",
      "\tmincounts = 0\n",
      "\terrorcontrolcounts = None\n",
      "\tmockcounts = Rep2-DMSO37_counts.txt\n",
      "\tno_scale_pseudocounts = False\n",
      "\toutprefix = Hsp90i37vsDMSO37-rep2_\n",
      "\tselectedcounts = Rep2-Hsp90i37_counts.txt\n",
      "\tincludestop = False\n",
      "\n",
      "\n",
      "Read counts data for 499 sites.\n",
      "\n",
      "Mutations with the lowest differential selection:\n",
      "site wt mut  diffsel\n",
      "403  A   S    -1.18\n",
      "403  A   L    -1.15\n",
      "446  R   W    -1.10\n",
      " 17  G   P    -1.09\n",
      "170  S   W    -1.04\n",
      "438  S   N    -1.04\n",
      "397  N   Y    -1.03\n",
      "345  S   V    -1.01\n",
      "241  Q   V    -0.96\n",
      "485  G   C    -0.96\n",
      "\n",
      "Mutations with the highest differential selection:\n",
      "site wt mut  diffsel\n",
      " 29  V   A     1.26\n",
      " 56  L   E     1.10\n",
      " 98  K   H     1.07\n",
      "447  M   S     1.04\n",
      "372  D   N     1.00\n",
      "363  V   P     1.00\n",
      "399  Q   A     1.00\n",
      " 18  E   H     0.91\n",
      "331  M   I     0.90\n",
      "469  E   A     0.90\n",
      "\n",
      "Sites with greatest total absolute differential selection:\n",
      "site  abs_diffsel  positive_diffsel  negative_diffsel\n",
      " 18         6.50              3.99             -2.51\n",
      "446         6.33              2.94             -3.40\n",
      " 56         6.20              4.51             -1.69\n",
      "398         6.17              2.67             -3.50\n",
      " 90         6.01              3.65             -2.36\n",
      " 58         5.98              1.43             -4.55\n",
      "403         5.97              1.04             -4.93\n",
      "432         5.85              4.45             -1.40\n",
      " 25         5.80              2.71             -3.09\n",
      "442         5.76              2.00             -3.76\n",
      "\n",
      "Writing mutation differential selection to Hsp90i37vsDMSO37-rep2_mutdiffsel.txt\n",
      "\n",
      "Writing site differential selection to Hsp90i37vsDMSO37-rep2_sitediffsel.txt\n",
      "\n",
      "Program complete.\n"
     ]
    }
   ],
   "source": [
    "#REPLICATE 1: -/+ Hsp90i 37C\n",
    "\n",
    "!dms_diffselection Rep2-DMSO37_counts.txt Rep2-Hsp90i37_counts.txt Hsp90i37vsDMSO37-rep2_"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 34,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Beginning execution of dms_diffselection in directory /Users/annapon/Documents/NP_DMS_2018/2018_NP_DMS/preferences\n",
      "\n",
      "Version information for dms_tools and associated packages.\n",
      "\tTime and date: Sat Mar 24 16:29:37 2018\n",
      "\tPlatform: Darwin-15.6.0-x86_64-i386-64bit\n",
      "\tPython version: 2.7.13 |Anaconda 4.4.0 (x86_64)| (default, Dec 20 2016, 23:05:08)  [GCC 4.2.1 Compatible Apple LLVM 6.0 (clang-600.0.57)]\n",
      "\tdms_tools version: 1.2.2\n",
      "\tBio version: 1.70\n",
      "\tnumpy version: 1.12.1\n",
      "\tscipy version: 0.19.0\n",
      "\tmatplotlib version: 2.0.2\n",
      "\tcython version: 0.25.2\n",
      "\tpystan version: 2.16.0.0\n",
      "\tweblogolib version: 3.5.0\n",
      "\tPyPDF2 version: 1.26.0\n",
      "\tpandas version: 0.20.1\n",
      "Parsed the following arguments:\n",
      "\tchartype = codon_to_aa\n",
      "\tpseudocount = 10\n",
      "\tmincounts = 0\n",
      "\terrorcontrolcounts = None\n",
      "\tmockcounts = Rep2-DMSO37_counts.txt\n",
      "\tno_scale_pseudocounts = False\n",
      "\toutprefix = HSF1i39vsDMSO37-rep2_\n",
      "\tselectedcounts = Rep2-HSF1i39_counts.txt\n",
      "\tincludestop = False\n",
      "\n",
      "\n",
      "Read counts data for 499 sites.\n",
      "\n",
      "Mutations with the lowest differential selection:\n",
      "site wt mut  diffsel\n",
      "367  S   G    -2.45\n",
      " 88  D   N    -2.39\n",
      "268  G   T    -2.11\n",
      " 43  M   N    -2.10\n",
      "453  P   R    -2.03\n",
      " 81  E   L    -2.03\n",
      " 82  H   F    -1.95\n",
      "363  V   C    -1.93\n",
      "427  A   R    -1.85\n",
      "239  M   A    -1.84\n",
      "\n",
      "Mutations with the highest differential selection:\n",
      "site wt mut  diffsel\n",
      "353  S   L     5.25\n",
      "283  P   M     5.20\n",
      "283  P   A     4.26\n",
      "377  S   M     3.89\n",
      "334  N   H     3.54\n",
      "283  P   F     3.28\n",
      "283  P   W     3.27\n",
      "283  P   G     3.23\n",
      "353  S   H     3.22\n",
      "136  M   I     3.04\n",
      "\n",
      "Sites with greatest total absolute differential selection:\n",
      "site  abs_diffsel  positive_diffsel  negative_diffsel\n",
      "283        39.60             38.46             -1.14\n",
      "353        17.04             15.86             -1.18\n",
      " 82        16.65              0.79            -15.86\n",
      " 34        15.57             10.82             -4.74\n",
      " 50        14.44              0.00            -14.44\n",
      "466        14.07              3.76            -10.30\n",
      " 81        13.72              3.82             -9.90\n",
      " 83        13.30              0.07            -13.23\n",
      " 86        13.21              1.12            -12.09\n",
      "334        12.92              9.94             -2.99\n",
      "\n",
      "Writing mutation differential selection to HSF1i39vsDMSO37-rep2_mutdiffsel.txt\n",
      "\n",
      "Writing site differential selection to HSF1i39vsDMSO37-rep2_sitediffsel.txt\n",
      "\n",
      "Program complete.\n"
     ]
    }
   ],
   "source": [
    "#REPLICATE 2: -/+ HSF1i 39C\n",
    "\n",
    "!dms_diffselection Rep2-DMSO37_counts.txt Rep2-HSF1i39_counts.txt HSF1i39vsDMSO37-rep2_"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 36,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Beginning execution of dms_diffselection in directory /Users/annapon/Documents/NP_DMS_2018/2018_NP_DMS/preferences\n",
      "\n",
      "Version information for dms_tools and associated packages.\n",
      "\tTime and date: Sat Mar 24 16:30:41 2018\n",
      "\tPlatform: Darwin-15.6.0-x86_64-i386-64bit\n",
      "\tPython version: 2.7.13 |Anaconda 4.4.0 (x86_64)| (default, Dec 20 2016, 23:05:08)  [GCC 4.2.1 Compatible Apple LLVM 6.0 (clang-600.0.57)]\n",
      "\tdms_tools version: 1.2.2\n",
      "\tBio version: 1.70\n",
      "\tnumpy version: 1.12.1\n",
      "\tscipy version: 0.19.0\n",
      "\tmatplotlib version: 2.0.2\n",
      "\tcython version: 0.25.2\n",
      "\tpystan version: 2.16.0.0\n",
      "\tweblogolib version: 3.5.0\n",
      "\tPyPDF2 version: 1.26.0\n",
      "\tpandas version: 0.20.1\n",
      "Parsed the following arguments:\n",
      "\tchartype = codon_to_aa\n",
      "\tpseudocount = 10\n",
      "\tmincounts = 0\n",
      "\terrorcontrolcounts = None\n",
      "\tmockcounts = Rep2-DMSO37_counts.txt\n",
      "\tno_scale_pseudocounts = False\n",
      "\toutprefix = Hsp90i39vsDMSO37-rep2_\n",
      "\tselectedcounts = Rep2-Hsp90i39_counts.txt\n",
      "\tincludestop = False\n",
      "\n",
      "\n",
      "Read counts data for 499 sites.\n",
      "\n",
      "Mutations with the lowest differential selection:\n",
      "site wt mut  diffsel\n",
      " 82  H   Q    -2.31\n",
      " 88  D   N    -2.19\n",
      "367  S   G    -2.18\n",
      "367  S   A    -2.12\n",
      " 82  H   W    -2.06\n",
      "243  R   F    -1.92\n",
      "424  T   V    -1.90\n",
      "421  D   L    -1.89\n",
      "463  V   L    -1.85\n",
      " 50  S   Q    -1.83\n",
      "\n",
      "Mutations with the highest differential selection:\n",
      "site wt mut  diffsel\n",
      "353  S   L     2.55\n",
      "334  N   M     2.26\n",
      "377  S   M     2.04\n",
      "353  S   H     1.96\n",
      "136  M   I     1.81\n",
      "283  P   W     1.79\n",
      "283  P   A     1.78\n",
      "311  Q   R     1.78\n",
      "283  P   S     1.71\n",
      "353  S   R     1.66\n",
      "\n",
      "Sites with greatest total absolute differential selection:\n",
      "site  abs_diffsel  positive_diffsel  negative_diffsel\n",
      "283        20.75             19.56             -1.19\n",
      " 82        18.27              0.09            -18.18\n",
      " 50        16.31              0.00            -16.31\n",
      " 77        13.93              0.27            -13.66\n",
      " 49        13.74              0.43            -13.31\n",
      " 56        13.67              0.14            -13.53\n",
      " 89        13.60              0.22            -13.39\n",
      " 86        12.80              0.22            -12.58\n",
      " 83        12.43              0.00            -12.43\n",
      "446        12.21              0.33            -11.88\n",
      "\n",
      "Writing mutation differential selection to Hsp90i39vsDMSO37-rep2_mutdiffsel.txt\n",
      "\n",
      "Writing site differential selection to Hsp90i39vsDMSO37-rep2_sitediffsel.txt\n",
      "\n",
      "Program complete.\n"
     ]
    }
   ],
   "source": [
    "#REPLICATE 2: -/+ Hsp90i 39C\n",
    "\n",
    "!dms_diffselection Rep2-DMSO37_counts.txt Rep2-Hsp90i39_counts.txt Hsp90i39vsDMSO37-rep2_"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 37,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Beginning execution of dms_diffselection in directory /Users/annapon/Documents/NP_DMS_2018/2018_NP_DMS/preferences\n",
      "\n",
      "Version information for dms_tools and associated packages.\n",
      "\tTime and date: Sat Mar 24 16:31:46 2018\n",
      "\tPlatform: Darwin-15.6.0-x86_64-i386-64bit\n",
      "\tPython version: 2.7.13 |Anaconda 4.4.0 (x86_64)| (default, Dec 20 2016, 23:05:08)  [GCC 4.2.1 Compatible Apple LLVM 6.0 (clang-600.0.57)]\n",
      "\tdms_tools version: 1.2.2\n",
      "\tBio version: 1.70\n",
      "\tnumpy version: 1.12.1\n",
      "\tscipy version: 0.19.0\n",
      "\tmatplotlib version: 2.0.2\n",
      "\tcython version: 0.25.2\n",
      "\tpystan version: 2.16.0.0\n",
      "\tweblogolib version: 3.5.0\n",
      "\tPyPDF2 version: 1.26.0\n",
      "\tpandas version: 0.20.1\n",
      "Parsed the following arguments:\n",
      "\tchartype = codon_to_aa\n",
      "\tpseudocount = 10\n",
      "\tmincounts = 0\n",
      "\terrorcontrolcounts = None\n",
      "\tmockcounts = Rep2-DMSO37_counts.txt\n",
      "\tno_scale_pseudocounts = False\n",
      "\toutprefix = DMSO39vsDMSO37-rep2_\n",
      "\tselectedcounts = Rep2-DMSO39_counts.txt\n",
      "\tincludestop = False\n",
      "\n",
      "\n",
      "Read counts data for 499 sites.\n",
      "\n",
      "Mutations with the lowest differential selection:\n",
      "site wt mut  diffsel\n",
      "367  S   G    -2.09\n",
      " 77  K   V    -1.60\n",
      " 51  D   P    -1.59\n",
      "318  P   E    -1.59\n",
      "321  N   R    -1.56\n",
      "183  V   T    -1.56\n",
      " 77  K   S    -1.54\n",
      "440  M   G    -1.49\n",
      " 70  A   P    -1.48\n",
      " 88  D   N    -1.47\n",
      "\n",
      "Mutations with the highest differential selection:\n",
      "site wt mut  diffsel\n",
      "334  N   M     2.20\n",
      "334  N   H     2.09\n",
      "353  S   L     1.75\n",
      "353  S   F     1.43\n",
      "283  P   M     1.31\n",
      "268  G   P     1.31\n",
      "136  M   I     1.29\n",
      "130  T   V     1.26\n",
      "283  P   A     1.19\n",
      "377  S   M     1.16\n",
      "\n",
      "Sites with greatest total absolute differential selection:\n",
      "site  abs_diffsel  positive_diffsel  negative_diffsel\n",
      " 82        14.35              0.00            -14.35\n",
      " 83        12.88              0.29            -12.59\n",
      " 50        12.88              0.04            -12.83\n",
      "283        12.13             11.34             -0.80\n",
      " 56        11.66              0.24            -11.42\n",
      "446        11.48              0.49            -10.99\n",
      " 51        11.46              0.69            -10.77\n",
      " 49        11.00              0.39            -10.61\n",
      " 86        10.85              1.18             -9.67\n",
      " 59        10.29              0.51             -9.78\n",
      "\n",
      "Writing mutation differential selection to DMSO39vsDMSO37-rep2_mutdiffsel.txt\n",
      "\n",
      "Writing site differential selection to DMSO39vsDMSO37-rep2_sitediffsel.txt\n",
      "\n",
      "Program complete.\n"
     ]
    }
   ],
   "source": [
    "#REPLICATE 2: DMSO 39C vs DMSO 37C\n",
    "\n",
    "!dms_diffselection Rep2-DMSO37_counts.txt Rep2-DMSO39_counts.txt DMSO39vsDMSO37-rep2_"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 38,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Beginning execution of dms_diffselection in directory /Users/annapon/Documents/NP_DMS_2018/2018_NP_DMS/preferences\n",
      "\n",
      "Version information for dms_tools and associated packages.\n",
      "\tTime and date: Sat Mar 24 16:32:43 2018\n",
      "\tPlatform: Darwin-15.6.0-x86_64-i386-64bit\n",
      "\tPython version: 2.7.13 |Anaconda 4.4.0 (x86_64)| (default, Dec 20 2016, 23:05:08)  [GCC 4.2.1 Compatible Apple LLVM 6.0 (clang-600.0.57)]\n",
      "\tdms_tools version: 1.2.2\n",
      "\tBio version: 1.70\n",
      "\tnumpy version: 1.12.1\n",
      "\tscipy version: 0.19.0\n",
      "\tmatplotlib version: 2.0.2\n",
      "\tcython version: 0.25.2\n",
      "\tpystan version: 2.16.0.0\n",
      "\tweblogolib version: 3.5.0\n",
      "\tPyPDF2 version: 1.26.0\n",
      "\tpandas version: 0.20.1\n",
      "Parsed the following arguments:\n",
      "\tchartype = codon_to_aa\n",
      "\tpseudocount = 10\n",
      "\tmincounts = 0\n",
      "\terrorcontrolcounts = None\n",
      "\tmockcounts = Rep2-HSF1i37_counts.txt\n",
      "\tno_scale_pseudocounts = False\n",
      "\toutprefix = HSF1i39vsHSF1i37-rep2_\n",
      "\tselectedcounts = Rep2-HSF1i39_counts.txt\n",
      "\tincludestop = False\n",
      "\n",
      "\n",
      "Read counts data for 499 sites.\n",
      "\n",
      "Mutations with the lowest differential selection:\n",
      "site wt mut  diffsel\n",
      "453  P   R    -2.16\n",
      "466  L   I    -2.04\n",
      " 81  E   L    -2.01\n",
      "367  S   G    -1.98\n",
      " 51  D   A    -1.96\n",
      " 27  A   S    -1.91\n",
      "240  D   Q    -1.83\n",
      " 73  E   R    -1.83\n",
      "268  G   T    -1.81\n",
      " 88  D   N    -1.79\n",
      "\n",
      "Mutations with the highest differential selection:\n",
      "site wt mut  diffsel\n",
      "283  P   M     5.96\n",
      "353  S   L     5.01\n",
      "283  P   A     4.14\n",
      "283  P   F     3.98\n",
      "377  S   M     3.63\n",
      "353  S   H     3.36\n",
      "450  G   K     3.17\n",
      "334  N   H     3.13\n",
      "283  P   G     3.11\n",
      "283  P   W     3.09\n",
      "\n",
      "Sites with greatest total absolute differential selection:\n",
      "site  abs_diffsel  positive_diffsel  negative_diffsel\n",
      "283        41.41             39.53             -1.89\n",
      "353        16.60             15.90             -0.70\n",
      " 34        14.71             10.92             -3.79\n",
      "466        12.58              4.53             -8.05\n",
      " 81        12.25              3.73             -8.52\n",
      " 82        12.13              1.42            -10.71\n",
      " 83        11.59              0.36            -11.23\n",
      " 50        11.41              0.51            -10.91\n",
      "497        11.40              1.39            -10.01\n",
      " 89        10.97              0.00            -10.97\n",
      "\n",
      "Writing mutation differential selection to HSF1i39vsHSF1i37-rep2_mutdiffsel.txt\n",
      "\n",
      "Writing site differential selection to HSF1i39vsHSF1i37-rep2_sitediffsel.txt\n",
      "\n",
      "Program complete.\n"
     ]
    }
   ],
   "source": [
    "#REPLICATE 2: HSF1i 39C vs HSF1i 37C\n",
    "\n",
    "!dms_diffselection Rep2-HSF1i37_counts.txt Rep2-HSF1i39_counts.txt HSF1i39vsHSF1i37-rep2_"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 39,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Beginning execution of dms_diffselection in directory /Users/annapon/Documents/NP_DMS_2018/2018_NP_DMS/preferences\n",
      "\n",
      "Version information for dms_tools and associated packages.\n",
      "\tTime and date: Sat Mar 24 16:33:39 2018\n",
      "\tPlatform: Darwin-15.6.0-x86_64-i386-64bit\n",
      "\tPython version: 2.7.13 |Anaconda 4.4.0 (x86_64)| (default, Dec 20 2016, 23:05:08)  [GCC 4.2.1 Compatible Apple LLVM 6.0 (clang-600.0.57)]\n",
      "\tdms_tools version: 1.2.2\n",
      "\tBio version: 1.70\n",
      "\tnumpy version: 1.12.1\n",
      "\tscipy version: 0.19.0\n",
      "\tmatplotlib version: 2.0.2\n",
      "\tcython version: 0.25.2\n",
      "\tpystan version: 2.16.0.0\n",
      "\tweblogolib version: 3.5.0\n",
      "\tPyPDF2 version: 1.26.0\n",
      "\tpandas version: 0.20.1\n",
      "Parsed the following arguments:\n",
      "\tchartype = codon_to_aa\n",
      "\tpseudocount = 10\n",
      "\tmincounts = 0\n",
      "\terrorcontrolcounts = None\n",
      "\tmockcounts = Rep2-DMSO39_counts.txt\n",
      "\tno_scale_pseudocounts = False\n",
      "\toutprefix = HSF1i39vsDMSO39-rep2_\n",
      "\tselectedcounts = Rep2-HSF1i39_counts.txt\n",
      "\tincludestop = False\n",
      "\n",
      "\n",
      "Read counts data for 499 sites.\n",
      "\n",
      "Mutations with the lowest differential selection:\n",
      "site wt mut  diffsel\n",
      "426  M   C    -2.05\n",
      "397  N   Q    -1.59\n",
      "466  L   I    -1.58\n",
      "453  P   C    -1.57\n",
      " 27  A   S    -1.56\n",
      "422  K   I    -1.46\n",
      "290  D   E    -1.44\n",
      "397  N   K    -1.42\n",
      "447  M   C    -1.42\n",
      "268  G   T    -1.40\n",
      "\n",
      "Mutations with the highest differential selection:\n",
      "site wt mut  diffsel\n",
      "283  P   M     3.89\n",
      "353  S   L     3.51\n",
      "283  P   A     3.07\n",
      "273  K   W     3.06\n",
      "377  S   M     2.79\n",
      "111  Y   V     2.67\n",
      "283  P   F     2.62\n",
      "457  S   Q     2.45\n",
      "283  P   W     2.39\n",
      "283  P   S     2.33\n",
      "\n",
      "Sites with greatest total absolute differential selection:\n",
      "site  abs_diffsel  positive_diffsel  negative_diffsel\n",
      "283        28.36             27.59             -0.77\n",
      "353        11.69             10.96             -0.73\n",
      " 34        11.38              7.37             -4.01\n",
      "273        10.78              8.30             -2.49\n",
      "498         9.44              3.11             -6.34\n",
      "420         9.42              1.35             -8.08\n",
      "417         9.18              3.38             -5.80\n",
      "397         9.11              1.98             -7.14\n",
      " 81         8.99              3.85             -5.14\n",
      "456         8.97              4.61             -4.36\n",
      "\n",
      "Writing mutation differential selection to HSF1i39vsDMSO39-rep2_mutdiffsel.txt\n",
      "\n",
      "Writing site differential selection to HSF1i39vsDMSO39-rep2_sitediffsel.txt\n",
      "\n",
      "Program complete.\n"
     ]
    }
   ],
   "source": [
    "#REPLICATE 2: HSF1i 39C vs DMSO 39C\n",
    "\n",
    "!dms_diffselection Rep2-DMSO39_counts.txt Rep2-HSF1i39_counts.txt HSF1i39vsDMSO39-rep2_"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 40,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Beginning execution of dms_diffselection in directory /Users/annapon/Documents/NP_DMS_2018/2018_NP_DMS/preferences\n",
      "\n",
      "Version information for dms_tools and associated packages.\n",
      "\tTime and date: Sat Mar 24 16:34:15 2018\n",
      "\tPlatform: Darwin-15.6.0-x86_64-i386-64bit\n",
      "\tPython version: 2.7.13 |Anaconda 4.4.0 (x86_64)| (default, Dec 20 2016, 23:05:08)  [GCC 4.2.1 Compatible Apple LLVM 6.0 (clang-600.0.57)]\n",
      "\tdms_tools version: 1.2.2\n",
      "\tBio version: 1.70\n",
      "\tnumpy version: 1.12.1\n",
      "\tscipy version: 0.19.0\n",
      "\tmatplotlib version: 2.0.2\n",
      "\tcython version: 0.25.2\n",
      "\tpystan version: 2.16.0.0\n",
      "\tweblogolib version: 3.5.0\n",
      "\tPyPDF2 version: 1.26.0\n",
      "\tpandas version: 0.20.1\n",
      "Parsed the following arguments:\n",
      "\tchartype = codon_to_aa\n",
      "\tpseudocount = 10\n",
      "\tmincounts = 0\n",
      "\terrorcontrolcounts = None\n",
      "\tmockcounts = Rep3-DMSO37_counts.txt\n",
      "\tno_scale_pseudocounts = False\n",
      "\toutprefix = HSF1i37vsDMSO37-rep3_\n",
      "\tselectedcounts = Rep3-HSF1i37_counts.txt\n",
      "\tincludestop = False\n",
      "\n",
      "\n",
      "Read counts data for 499 sites.\n",
      "\n",
      "Mutations with the lowest differential selection:\n",
      "site wt mut  diffsel\n",
      " 43  M   N    -1.59\n",
      "431  G   T    -1.54\n",
      " 64  E   H    -1.41\n",
      " 59  N   P    -1.28\n",
      "401  A   K    -1.24\n",
      "108  L   S    -1.23\n",
      "183  V   A    -1.19\n",
      "421  D   K    -1.19\n",
      " 29  V   P    -1.14\n",
      "319  N   V    -1.14\n",
      "\n",
      "Mutations with the highest differential selection:\n",
      "site wt mut  diffsel\n",
      " 78  Y   F     1.35\n",
      "225  I   L     1.20\n",
      "157  T   Q     1.10\n",
      " 56  L   K     1.06\n",
      "431  G   S     1.04\n",
      "164  C   T     1.03\n",
      " 65  R   Y     1.02\n",
      "332  A   S     1.01\n",
      "447  M   S     1.00\n",
      "428  A   M     0.99\n",
      "\n",
      "Sites with greatest total absolute differential selection:\n",
      "site  abs_diffsel  positive_diffsel  negative_diffsel\n",
      " 50         8.23              2.42             -5.81\n",
      "421         7.88              1.58             -6.30\n",
      " 83         7.77              4.15             -3.62\n",
      "399         7.38              3.45             -3.94\n",
      " 64         7.36              2.95             -4.41\n",
      " 61         7.20              1.36             -5.84\n",
      " 74         7.15              3.20             -3.95\n",
      "427         7.12              2.57             -4.55\n",
      " 90         7.00              4.44             -2.56\n",
      "319         6.75              2.87             -3.89\n",
      "\n",
      "Writing mutation differential selection to HSF1i37vsDMSO37-rep3_mutdiffsel.txt\n",
      "\n",
      "Writing site differential selection to HSF1i37vsDMSO37-rep3_sitediffsel.txt\n",
      "\n",
      "Program complete.\n"
     ]
    }
   ],
   "source": [
    "#REPLICATE 3: -/+ HSF1i 37C\n",
    "\n",
    "!dms_diffselection Rep3-DMSO37_counts.txt Rep3-HSF1i37_counts.txt HSF1i37vsDMSO37-rep3_"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 41,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Beginning execution of dms_diffselection in directory /Users/annapon/Documents/NP_DMS_2018/2018_NP_DMS/preferences\n",
      "\n",
      "Version information for dms_tools and associated packages.\n",
      "\tTime and date: Sat Mar 24 16:34:39 2018\n",
      "\tPlatform: Darwin-15.6.0-x86_64-i386-64bit\n",
      "\tPython version: 2.7.13 |Anaconda 4.4.0 (x86_64)| (default, Dec 20 2016, 23:05:08)  [GCC 4.2.1 Compatible Apple LLVM 6.0 (clang-600.0.57)]\n",
      "\tdms_tools version: 1.2.2\n",
      "\tBio version: 1.70\n",
      "\tnumpy version: 1.12.1\n",
      "\tscipy version: 0.19.0\n",
      "\tmatplotlib version: 2.0.2\n",
      "\tcython version: 0.25.2\n",
      "\tpystan version: 2.16.0.0\n",
      "\tweblogolib version: 3.5.0\n",
      "\tPyPDF2 version: 1.26.0\n",
      "\tpandas version: 0.20.1\n",
      "Parsed the following arguments:\n",
      "\tchartype = codon_to_aa\n",
      "\tpseudocount = 10\n",
      "\tmincounts = 0\n",
      "\terrorcontrolcounts = None\n",
      "\tmockcounts = Rep3-DMSO37_counts.txt\n",
      "\tno_scale_pseudocounts = False\n",
      "\toutprefix = HSF1i37vsDMSO37-rep3_\n",
      "\tselectedcounts = Rep3-HSF1i37_counts.txt\n",
      "\tincludestop = False\n",
      "\n",
      "\n",
      "Removing existing output file HSF1i37vsDMSO37-rep3_mutdiffsel.txt\n",
      "\n",
      "Removing existing output file HSF1i37vsDMSO37-rep3_sitediffsel.txt\n",
      "\n",
      "Read counts data for 499 sites.\n",
      "\n",
      "Mutations with the lowest differential selection:\n",
      "site wt mut  diffsel\n",
      " 43  M   N    -1.59\n",
      "431  G   T    -1.54\n",
      " 64  E   H    -1.41\n",
      " 59  N   P    -1.28\n",
      "401  A   K    -1.24\n",
      "108  L   S    -1.23\n",
      "183  V   A    -1.19\n",
      "421  D   K    -1.19\n",
      " 29  V   P    -1.14\n",
      "319  N   V    -1.14\n",
      "\n",
      "Mutations with the highest differential selection:\n",
      "site wt mut  diffsel\n",
      " 78  Y   F     1.35\n",
      "225  I   L     1.20\n",
      "157  T   Q     1.10\n",
      " 56  L   K     1.06\n",
      "431  G   S     1.04\n",
      "164  C   T     1.03\n",
      " 65  R   Y     1.02\n",
      "332  A   S     1.01\n",
      "447  M   S     1.00\n",
      "428  A   M     0.99\n",
      "\n",
      "Sites with greatest total absolute differential selection:\n",
      "site  abs_diffsel  positive_diffsel  negative_diffsel\n",
      " 50         8.23              2.42             -5.81\n",
      "421         7.88              1.58             -6.30\n",
      " 83         7.77              4.15             -3.62\n",
      "399         7.38              3.45             -3.94\n",
      " 64         7.36              2.95             -4.41\n",
      " 61         7.20              1.36             -5.84\n",
      " 74         7.15              3.20             -3.95\n",
      "427         7.12              2.57             -4.55\n",
      " 90         7.00              4.44             -2.56\n",
      "319         6.75              2.87             -3.89\n",
      "\n",
      "Writing mutation differential selection to HSF1i37vsDMSO37-rep3_mutdiffsel.txt\n",
      "\n",
      "Writing site differential selection to HSF1i37vsDMSO37-rep3_sitediffsel.txt\n",
      "\n",
      "Program complete.\n"
     ]
    }
   ],
   "source": [
    "#REPLICATE 3: -/+ Hsp90i 37C\n",
    "\n",
    "!dms_diffselection Rep3-DMSO37_counts.txt Rep3-HSF1i37_counts.txt HSF1i37vsDMSO37-rep3_"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 42,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Beginning execution of dms_diffselection in directory /Users/annapon/Documents/NP_DMS_2018/2018_NP_DMS/preferences\n",
      "\n",
      "Version information for dms_tools and associated packages.\n",
      "\tTime and date: Sat Mar 24 16:35:08 2018\n",
      "\tPlatform: Darwin-15.6.0-x86_64-i386-64bit\n",
      "\tPython version: 2.7.13 |Anaconda 4.4.0 (x86_64)| (default, Dec 20 2016, 23:05:08)  [GCC 4.2.1 Compatible Apple LLVM 6.0 (clang-600.0.57)]\n",
      "\tdms_tools version: 1.2.2\n",
      "\tBio version: 1.70\n",
      "\tnumpy version: 1.12.1\n",
      "\tscipy version: 0.19.0\n",
      "\tmatplotlib version: 2.0.2\n",
      "\tcython version: 0.25.2\n",
      "\tpystan version: 2.16.0.0\n",
      "\tweblogolib version: 3.5.0\n",
      "\tPyPDF2 version: 1.26.0\n",
      "\tpandas version: 0.20.1\n",
      "Parsed the following arguments:\n",
      "\tchartype = codon_to_aa\n",
      "\tpseudocount = 10\n",
      "\tmincounts = 0\n",
      "\terrorcontrolcounts = None\n",
      "\tmockcounts = Rep3-DMSO37_counts.txt\n",
      "\tno_scale_pseudocounts = False\n",
      "\toutprefix = HSF1i39vsDMSO37-rep3_\n",
      "\tselectedcounts = Rep3-HSF1i39_counts.txt\n",
      "\tincludestop = False\n",
      "\n",
      "\n",
      "Read counts data for 499 sites.\n",
      "\n",
      "Mutations with the lowest differential selection:\n",
      "site wt mut  diffsel\n",
      "397  N   R    -3.04\n",
      " 88  D   N    -2.79\n",
      "367  S   G    -2.61\n",
      "363  V   C    -2.54\n",
      "401  A   R    -2.54\n",
      " 50  S   A    -2.52\n",
      "167  M   L    -2.49\n",
      "401  A   K    -2.47\n",
      " 44  C   V    -2.34\n",
      " 18  E   S    -2.32\n",
      "\n",
      "Mutations with the highest differential selection:\n",
      "site wt mut  diffsel\n",
      "353  S   L     6.58\n",
      "283  P   M     6.53\n",
      "420  F   H     5.88\n",
      "334  N   M     5.47\n",
      "353  S   F     5.12\n",
      "283  P   A     4.85\n",
      "283  P   W     4.37\n",
      "353  S   H     4.27\n",
      " 81  E   I     4.08\n",
      "283  P   C     3.92\n",
      "\n",
      "Sites with greatest total absolute differential selection:\n",
      "site  abs_diffsel  positive_diffsel  negative_diffsel\n",
      "283        52.19             50.20             -1.99\n",
      "353        25.25             24.18             -1.07\n",
      "466        17.95              4.46            -13.49\n",
      " 50        17.76              0.34            -17.42\n",
      " 81        17.58              5.82            -11.77\n",
      " 80        17.06              3.28            -13.78\n",
      "497        16.66              0.97            -15.69\n",
      " 59        16.40              1.02            -15.37\n",
      " 83        15.73              0.08            -15.65\n",
      "498        15.67              2.28            -13.39\n",
      "\n",
      "Writing mutation differential selection to HSF1i39vsDMSO37-rep3_mutdiffsel.txt\n",
      "\n",
      "Writing site differential selection to HSF1i39vsDMSO37-rep3_sitediffsel.txt\n",
      "\n",
      "Program complete.\n"
     ]
    }
   ],
   "source": [
    "#REPLICATE 3: -/+ HSF1i 39C\n",
    "\n",
    "!dms_diffselection Rep3-DMSO37_counts.txt Rep3-HSF1i39_counts.txt HSF1i39vsDMSO37-rep3_"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 43,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Beginning execution of dms_diffselection in directory /Users/annapon/Documents/NP_DMS_2018/2018_NP_DMS/preferences\n",
      "\n",
      "Version information for dms_tools and associated packages.\n",
      "\tTime and date: Sat Mar 24 16:36:00 2018\n",
      "\tPlatform: Darwin-15.6.0-x86_64-i386-64bit\n",
      "\tPython version: 2.7.13 |Anaconda 4.4.0 (x86_64)| (default, Dec 20 2016, 23:05:08)  [GCC 4.2.1 Compatible Apple LLVM 6.0 (clang-600.0.57)]\n",
      "\tdms_tools version: 1.2.2\n",
      "\tBio version: 1.70\n",
      "\tnumpy version: 1.12.1\n",
      "\tscipy version: 0.19.0\n",
      "\tmatplotlib version: 2.0.2\n",
      "\tcython version: 0.25.2\n",
      "\tpystan version: 2.16.0.0\n",
      "\tweblogolib version: 3.5.0\n",
      "\tPyPDF2 version: 1.26.0\n",
      "\tpandas version: 0.20.1\n",
      "Parsed the following arguments:\n",
      "\tchartype = codon_to_aa\n",
      "\tpseudocount = 10\n",
      "\tmincounts = 0\n",
      "\terrorcontrolcounts = None\n",
      "\tmockcounts = Rep3-DMSO37_counts.txt\n",
      "\tno_scale_pseudocounts = False\n",
      "\toutprefix = Hsp90i39vsDMSO37-rep3_\n",
      "\tselectedcounts = Rep3-Hsp90i39_counts.txt\n",
      "\tincludestop = False\n",
      "\n",
      "\n",
      "Read counts data for 499 sites.\n",
      "\n",
      "Mutations with the lowest differential selection:\n",
      "site wt mut  diffsel\n",
      "280  V   C    -1.70\n",
      " 88  D   N    -1.67\n",
      "404  G   A    -1.51\n",
      "346  F   A    -1.50\n",
      "108  L   S    -1.49\n",
      " 59  N   P    -1.48\n",
      "431  G   T    -1.47\n",
      "265  I   L    -1.47\n",
      " 63  I   N    -1.45\n",
      " 52  Y   S    -1.45\n",
      "\n",
      "Mutations with the highest differential selection:\n",
      "site wt mut  diffsel\n",
      "353  S   L     1.78\n",
      "334  N   H     1.53\n",
      "450  G   K     1.53\n",
      "425  I   L     1.44\n",
      "419  P   A     1.44\n",
      "334  N   M     1.40\n",
      "402  S   Q     1.36\n",
      "264  L   F     1.32\n",
      "250  N   T     1.30\n",
      "267  R   K     1.25\n",
      "\n",
      "Sites with greatest total absolute differential selection:\n",
      "site  abs_diffsel  positive_diffsel  negative_diffsel\n",
      " 50        12.95              0.84            -12.11\n",
      "283        12.27             11.34             -0.93\n",
      " 49        11.23              1.20            -10.03\n",
      "497        10.67              0.20            -10.47\n",
      " 56        10.05              1.40             -8.66\n",
      "466         9.85              1.88             -7.96\n",
      "446         9.77              0.53             -9.23\n",
      " 89         9.38              0.46             -8.92\n",
      "424         9.26              1.34             -7.92\n",
      " 59         9.09              0.38             -8.71\n",
      "\n",
      "Writing mutation differential selection to Hsp90i39vsDMSO37-rep3_mutdiffsel.txt\n",
      "\n",
      "Writing site differential selection to Hsp90i39vsDMSO37-rep3_sitediffsel.txt\n",
      "\n",
      "Program complete.\n"
     ]
    }
   ],
   "source": [
    "#REPLICATE 3: -/+ Hsp90i 39C\n",
    "\n",
    "!dms_diffselection Rep3-DMSO37_counts.txt Rep3-Hsp90i39_counts.txt Hsp90i39vsDMSO37-rep3_"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 44,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Beginning execution of dms_diffselection in directory /Users/annapon/Documents/NP_DMS_2018/2018_NP_DMS/preferences\n",
      "\n",
      "Version information for dms_tools and associated packages.\n",
      "\tTime and date: Sat Mar 24 16:36:52 2018\n",
      "\tPlatform: Darwin-15.6.0-x86_64-i386-64bit\n",
      "\tPython version: 2.7.13 |Anaconda 4.4.0 (x86_64)| (default, Dec 20 2016, 23:05:08)  [GCC 4.2.1 Compatible Apple LLVM 6.0 (clang-600.0.57)]\n",
      "\tdms_tools version: 1.2.2\n",
      "\tBio version: 1.70\n",
      "\tnumpy version: 1.12.1\n",
      "\tscipy version: 0.19.0\n",
      "\tmatplotlib version: 2.0.2\n",
      "\tcython version: 0.25.2\n",
      "\tpystan version: 2.16.0.0\n",
      "\tweblogolib version: 3.5.0\n",
      "\tPyPDF2 version: 1.26.0\n",
      "\tpandas version: 0.20.1\n",
      "Parsed the following arguments:\n",
      "\tchartype = codon_to_aa\n",
      "\tpseudocount = 10\n",
      "\tmincounts = 0\n",
      "\terrorcontrolcounts = None\n",
      "\tmockcounts = Rep3-DMSO37_counts.txt\n",
      "\tno_scale_pseudocounts = False\n",
      "\toutprefix = DMSO39vsDMSO37-rep3_\n",
      "\tselectedcounts = Rep3-DMSO39_counts.txt\n",
      "\tincludestop = False\n",
      "\n",
      "\n",
      "Read counts data for 499 sites.\n",
      "\n",
      "Mutations with the lowest differential selection:\n",
      "site wt mut  diffsel\n",
      "367  S   G    -3.16\n",
      "319  N   L    -2.25\n",
      " 50  S   A    -2.23\n",
      " 88  D   N    -2.13\n",
      " 56  L   T    -2.13\n",
      "363  V   C    -2.01\n",
      " 77  K   S    -2.01\n",
      " 50  S   P    -2.00\n",
      " 52  Y   S    -1.95\n",
      " 50  S   H    -1.94\n",
      "\n",
      "Mutations with the highest differential selection:\n",
      "site wt mut  diffsel\n",
      "457  S   F     2.60\n",
      "353  S   L     2.59\n",
      "283  P   C     2.40\n",
      "278  A   V     2.25\n",
      "283  P   A     2.07\n",
      "283  P   G     2.01\n",
      "409  Q   M     2.01\n",
      "283  P   W     1.95\n",
      "283  P   Q     1.87\n",
      "359  S   Q     1.84\n",
      "\n",
      "Sites with greatest total absolute differential selection:\n",
      "site  abs_diffsel  positive_diffsel  negative_diffsel\n",
      "283        21.48             18.99             -2.49\n",
      " 50        19.17              0.40            -18.78\n",
      " 82        16.60              1.33            -15.26\n",
      " 59        14.93              0.04            -14.89\n",
      " 56        14.32              1.95            -12.38\n",
      "497        14.26              0.93            -13.33\n",
      " 83        14.00              0.15            -13.86\n",
      "466        13.74              2.27            -11.47\n",
      " 41        13.53              1.52            -12.01\n",
      " 49        13.45              0.93            -12.51\n",
      "\n",
      "Writing mutation differential selection to DMSO39vsDMSO37-rep3_mutdiffsel.txt\n",
      "\n",
      "Writing site differential selection to DMSO39vsDMSO37-rep3_sitediffsel.txt\n",
      "\n",
      "Program complete.\n"
     ]
    }
   ],
   "source": [
    "#REPLICATE 3: DMSO 39C vs DMSO 37C\n",
    "\n",
    "!dms_diffselection Rep3-DMSO37_counts.txt Rep3-DMSO39_counts.txt DMSO39vsDMSO37-rep3_"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 45,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Beginning execution of dms_diffselection in directory /Users/annapon/Documents/NP_DMS_2018/2018_NP_DMS/preferences\n",
      "\n",
      "Version information for dms_tools and associated packages.\n",
      "\tTime and date: Sat Mar 24 16:37:45 2018\n",
      "\tPlatform: Darwin-15.6.0-x86_64-i386-64bit\n",
      "\tPython version: 2.7.13 |Anaconda 4.4.0 (x86_64)| (default, Dec 20 2016, 23:05:08)  [GCC 4.2.1 Compatible Apple LLVM 6.0 (clang-600.0.57)]\n",
      "\tdms_tools version: 1.2.2\n",
      "\tBio version: 1.70\n",
      "\tnumpy version: 1.12.1\n",
      "\tscipy version: 0.19.0\n",
      "\tmatplotlib version: 2.0.2\n",
      "\tcython version: 0.25.2\n",
      "\tpystan version: 2.16.0.0\n",
      "\tweblogolib version: 3.5.0\n",
      "\tPyPDF2 version: 1.26.0\n",
      "\tpandas version: 0.20.1\n",
      "Parsed the following arguments:\n",
      "\tchartype = codon_to_aa\n",
      "\tpseudocount = 10\n",
      "\tmincounts = 0\n",
      "\terrorcontrolcounts = None\n",
      "\tmockcounts = Rep3-HSF1i37_counts.txt\n",
      "\tno_scale_pseudocounts = False\n",
      "\toutprefix = HSF1i39vsDHSF1i37-rep3_\n",
      "\tselectedcounts = Rep3-HSF1i39_counts.txt\n",
      "\tincludestop = False\n",
      "\n",
      "\n",
      "Read counts data for 499 sites.\n",
      "\n",
      "Mutations with the lowest differential selection:\n",
      "site wt mut  diffsel\n",
      "397  N   R    -2.86\n",
      " 88  D   N    -2.27\n",
      "397  N   C    -2.20\n",
      "426  M   C    -2.17\n",
      "432  N   R    -2.17\n",
      " 18  E   S    -2.15\n",
      "370  N   V    -2.13\n",
      "223  C   T    -2.09\n",
      "367  S   G    -2.06\n",
      "401  A   R    -2.00\n",
      "\n",
      "Mutations with the highest differential selection:\n",
      "site wt mut  diffsel\n",
      "353  S   L     6.65\n",
      "283  P   M     6.37\n",
      "420  F   H     6.11\n",
      "334  N   M     5.44\n",
      "353  S   F     5.26\n",
      "283  P   A     5.01\n",
      "283  P   Y     4.79\n",
      "353  S   H     4.76\n",
      "283  P   W     4.45\n",
      "283  P   I     4.00\n",
      "\n",
      "Sites with greatest total absolute differential selection:\n",
      "site  abs_diffsel  positive_diffsel  negative_diffsel\n",
      "283        53.10             51.91             -1.18\n",
      "353        25.26             24.92             -0.33\n",
      " 81        18.05              4.98            -13.07\n",
      " 83        16.00              0.00            -16.00\n",
      "498        15.78              2.68            -13.10\n",
      "334        15.43             12.51             -2.92\n",
      " 34        15.29              7.37             -7.92\n",
      "466        15.16              3.89            -11.27\n",
      " 82        14.68              0.06            -14.62\n",
      "397        14.56              0.45            -14.11\n",
      "\n",
      "Writing mutation differential selection to HSF1i39vsDHSF1i37-rep3_mutdiffsel.txt\n",
      "\n",
      "Writing site differential selection to HSF1i39vsDHSF1i37-rep3_sitediffsel.txt\n",
      "\n",
      "Program complete.\n"
     ]
    }
   ],
   "source": [
    "#REPLICATE 3: HSF1i 39C vs HSF1i 37C\n",
    "\n",
    "!dms_diffselection Rep3-HSF1i37_counts.txt Rep3-HSF1i39_counts.txt HSF1i39vsDHSF1i37-rep3_"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 46,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Beginning execution of dms_diffselection in directory /Users/annapon/Documents/NP_DMS_2018/2018_NP_DMS/preferences\n",
      "\n",
      "Version information for dms_tools and associated packages.\n",
      "\tTime and date: Sat Mar 24 16:38:29 2018\n",
      "\tPlatform: Darwin-15.6.0-x86_64-i386-64bit\n",
      "\tPython version: 2.7.13 |Anaconda 4.4.0 (x86_64)| (default, Dec 20 2016, 23:05:08)  [GCC 4.2.1 Compatible Apple LLVM 6.0 (clang-600.0.57)]\n",
      "\tdms_tools version: 1.2.2\n",
      "\tBio version: 1.70\n",
      "\tnumpy version: 1.12.1\n",
      "\tscipy version: 0.19.0\n",
      "\tmatplotlib version: 2.0.2\n",
      "\tcython version: 0.25.2\n",
      "\tpystan version: 2.16.0.0\n",
      "\tweblogolib version: 3.5.0\n",
      "\tPyPDF2 version: 1.26.0\n",
      "\tpandas version: 0.20.1\n",
      "Parsed the following arguments:\n",
      "\tchartype = codon_to_aa\n",
      "\tpseudocount = 10\n",
      "\tmincounts = 0\n",
      "\terrorcontrolcounts = None\n",
      "\tmockcounts = Rep3-DMSO39_counts.txt\n",
      "\tno_scale_pseudocounts = False\n",
      "\toutprefix = HSF1i39vsDMSO39-rep3_\n",
      "\tselectedcounts = Rep3-HSF1i39_counts.txt\n",
      "\tincludestop = False\n",
      "\n",
      "\n",
      "Read counts data for 499 sites.\n",
      "\n",
      "Mutations with the lowest differential selection:\n",
      "site wt mut  diffsel\n",
      "457  S   F    -3.64\n",
      "409  Q   M    -2.96\n",
      "278  A   V    -2.60\n",
      "485  G   C    -2.51\n",
      "397  N   R    -2.28\n",
      "437  T   A    -2.22\n",
      "421  D   A    -2.18\n",
      "427  A   P    -2.14\n",
      "451  A   F    -2.07\n",
      "409  Q   A    -2.05\n",
      "\n",
      "Mutations with the highest differential selection:\n",
      "site wt mut  diffsel\n",
      "420  F   H     6.00\n",
      "283  P   M     5.23\n",
      "334  N   M     4.80\n",
      "467  S   R     4.22\n",
      "283  P   Y     4.11\n",
      "353  S   L     3.99\n",
      "334  N   H     3.77\n",
      "353  S   F     3.60\n",
      "462  G   A     3.06\n",
      " 61  L   G     3.05\n",
      "\n",
      "Sites with greatest total absolute differential selection:\n",
      "site  abs_diffsel  positive_diffsel  negative_diffsel\n",
      "283        32.14             32.08             -0.06\n",
      "420        17.16              9.19             -7.97\n",
      "334        17.02             13.47             -3.55\n",
      "353        16.88             16.52             -0.35\n",
      "409        14.23              2.13            -12.11\n",
      "454        12.27              1.22            -11.05\n",
      "377        12.12             10.07             -2.04\n",
      "397        11.42              1.00            -10.42\n",
      "421        11.35              1.80             -9.54\n",
      "427        10.73              2.64             -8.09\n",
      "\n",
      "Writing mutation differential selection to HSF1i39vsDMSO39-rep3_mutdiffsel.txt\n",
      "\n",
      "Writing site differential selection to HSF1i39vsDMSO39-rep3_sitediffsel.txt\n",
      "\n",
      "Program complete.\n"
     ]
    }
   ],
   "source": [
    "#REPLICATE 3: HSF1i 39C vs DMSO 39C\n",
    "\n",
    "!dms_diffselection Rep3-DMSO39_counts.txt Rep3-HSF1i39_counts.txt HSF1i39vsDMSO39-rep3_"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Generate logo plots"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 31,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Beginning execution of dms_logoplot in directory /Users/annapon/Documents/NP_DMS_2018/2018_NP_DMS/preferences\n",
      "\n",
      "Version information for dms_tools and associated packages.\n",
      "\tTime and date: Tue Mar 27 18:55:35 2018\n",
      "\tPlatform: Darwin-15.6.0-x86_64-i386-64bit\n",
      "\tPython version: 2.7.13 |Anaconda 4.4.0 (x86_64)| (default, Dec 20 2016, 23:05:08)  [GCC 4.2.1 Compatible Apple LLVM 6.0 (clang-600.0.57)]\n",
      "\tdms_tools version: 1.2.2\n",
      "\tBio version: 1.70\n",
      "\tnumpy version: 1.12.1\n",
      "\tscipy version: 0.19.0\n",
      "\tmatplotlib version: 2.0.2\n",
      "\tcython version: 0.25.2\n",
      "\tpystan version: 2.16.0.0\n",
      "\tweblogolib version: 3.5.0\n",
      "\tPyPDF2 version: 1.26.0\n",
      "\tpandas version: 0.20.1\n",
      "\n",
      "Parsed the following arguments:\n",
      "\trestrictdiffsel = None\n",
      "\toverlay3 = None\n",
      "\tcolormap = jet\n",
      "\tnperline = 100\n",
      "\tnumberevery = 10\n",
      "\texcludestop = False\n",
      "\tlogoplot = HSF1i39vsDMSO37-rep1_diffsel_logoplot.pdf\n",
      "\toverlay_cmap = None\n",
      "\tmapmetric = charge\n",
      "\tdiffprefheight = 1.0\n",
      "\tnosepline = False\n",
      "\toverlay1 = None\n",
      "\toverlay2 = None\n",
      "\tdiffselheight = ['DMSO39vsDMSO37-rep1_mutdiffsel.txt', 'DMSO39vsDMSO37-rep2_mutdiffsel.txt', 'DMSO39vsDMSO37-rep3_mutdiffsel.txt', 'HSF1i37vsDMSO37-rep1_mutdiffsel.txt', 'HSF1i37vsDMSO37-rep2_mutdiffsel.txt', 'HSF1i37vsDMSO37-rep3_mutdiffsel.txt', 'HSF1i39vsDMSO37-rep1_mutdiffsel.txt', 'HSF1i39vsDMSO37-rep2_mutdiffsel.txt', 'HSF1i39vsDMSO37-rep3_mutdiffsel.txt', 'HSF1i39vsDMSO39-rep1_mutdiffsel.txt', 'HSF1i39vsDMSO39-rep2_mutdiffsel.txt', 'HSF1i39vsDMSO39-rep3_mutdiffsel.txt', 'HSF1i39vsHSF1i37-rep1_mutdiffsel.txt', 'HSF1i39vsHSF1i37-rep2_mutdiffsel.txt', 'HSF1i39vsHSF1i37-rep3_mutdiffsel.txt', 'Hsp90i37vsDMSO37-rep1_mutdiffsel.txt', 'Hsp90i37vsDMSO37-rep2_mutdiffsel.txt', 'Hsp90i39vsDMSO37-rep3_mutdiffsel.txt', 'Hsp90i39vsDMSO37-rep1_mutdiffsel.txt', 'Hsp90i39vsDMSO37-rep2_mutdiffsel.txt', 'Hsp90i39vsDMSO37-rep3_mutdiffsel.txt']\n",
      "\tstringencyparameter = None\n",
      "\tinfile = HSF1i39vsDMSO37-rep1_mutdiffsel.txt\n",
      "\tletterheight = 2.0\n",
      "\n",
      "\n",
      "Logo plot will be written to HSF1i39vsDMSO37-rep1_diffsel_logoplot.pdf\n",
      "Removing existing file HSF1i39vsDMSO37-rep1_diffsel_logoplot.pdf\n",
      "\n",
      "Reading data from HSF1i39vsDMSO37-rep1_mutdiffsel.txt\n",
      "\n",
      "Now making the logo plot in file HSF1i39vsDMSO37-rep1_diffsel_logoplot.pdf\n",
      "\n",
      "Successfully completed dms_logoplot at Tue Mar 27 18:55:50 2018\n"
     ]
    }
   ],
   "source": [
    "#make logoplots for each selection, same axes for all\n",
    "\n",
    "#rep1 HSF1i 39C vs DMSO 37C\n",
    "!dms_logoplot HSF1i39vsDMSO37-rep1_mutdiffsel.txt HSF1i39vsDMSO37-rep1_diffsel_logoplot.pdf --nperline 100 --mapmetric charge --letterheight 2 --diffselheight DMSO39vsDMSO37-rep1_mutdiffsel.txt DMSO39vsDMSO37-rep2_mutdiffsel.txt DMSO39vsDMSO37-rep3_mutdiffsel.txt HSF1i37vsDMSO37-rep1_mutdiffsel.txt HSF1i37vsDMSO37-rep2_mutdiffsel.txt HSF1i37vsDMSO37-rep3_mutdiffsel.txt HSF1i39vsDMSO37-rep1_mutdiffsel.txt HSF1i39vsDMSO37-rep2_mutdiffsel.txt HSF1i39vsDMSO37-rep3_mutdiffsel.txt HSF1i39vsDMSO39-rep1_mutdiffsel.txt HSF1i39vsDMSO39-rep2_mutdiffsel.txt HSF1i39vsDMSO39-rep3_mutdiffsel.txt HSF1i39vsHSF1i37-rep1_mutdiffsel.txt HSF1i39vsHSF1i37-rep2_mutdiffsel.txt HSF1i39vsHSF1i37-rep3_mutdiffsel.txt Hsp90i37vsDMSO37-rep1_mutdiffsel.txt Hsp90i37vsDMSO37-rep2_mutdiffsel.txt Hsp90i39vsDMSO37-rep3_mutdiffsel.txt Hsp90i39vsDMSO37-rep1_mutdiffsel.txt Hsp90i39vsDMSO37-rep2_mutdiffsel.txt Hsp90i39vsDMSO37-rep3_mutdiffsel.txt  "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 32,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Beginning execution of dms_logoplot in directory /Users/annapon/Documents/NP_DMS_2018/2018_NP_DMS/preferences\n",
      "\n",
      "Version information for dms_tools and associated packages.\n",
      "\tTime and date: Tue Mar 27 18:56:18 2018\n",
      "\tPlatform: Darwin-15.6.0-x86_64-i386-64bit\n",
      "\tPython version: 2.7.13 |Anaconda 4.4.0 (x86_64)| (default, Dec 20 2016, 23:05:08)  [GCC 4.2.1 Compatible Apple LLVM 6.0 (clang-600.0.57)]\n",
      "\tdms_tools version: 1.2.2\n",
      "\tBio version: 1.70\n",
      "\tnumpy version: 1.12.1\n",
      "\tscipy version: 0.19.0\n",
      "\tmatplotlib version: 2.0.2\n",
      "\tcython version: 0.25.2\n",
      "\tpystan version: 2.16.0.0\n",
      "\tweblogolib version: 3.5.0\n",
      "\tPyPDF2 version: 1.26.0\n",
      "\tpandas version: 0.20.1\n",
      "\n",
      "Parsed the following arguments:\n",
      "\trestrictdiffsel = None\n",
      "\toverlay3 = None\n",
      "\tcolormap = jet\n",
      "\tnperline = 100\n",
      "\tnumberevery = 10\n",
      "\texcludestop = False\n",
      "\tlogoplot = HSF1i39vsDMSO37-rep2_diffsel_logoplot.pdf\n",
      "\toverlay_cmap = None\n",
      "\tmapmetric = charge\n",
      "\tdiffprefheight = 1.0\n",
      "\tnosepline = False\n",
      "\toverlay1 = None\n",
      "\toverlay2 = None\n",
      "\tdiffselheight = ['DMSO39vsDMSO37-rep1_mutdiffsel.txt', 'DMSO39vsDMSO37-rep2_mutdiffsel.txt', 'DMSO39vsDMSO37-rep3_mutdiffsel.txt', 'HSF1i37vsDMSO37-rep1_mutdiffsel.txt', 'HSF1i37vsDMSO37-rep2_mutdiffsel.txt', 'HSF1i37vsDMSO37-rep3_mutdiffsel.txt', 'HSF1i39vsDMSO37-rep1_mutdiffsel.txt', 'HSF1i39vsDMSO37-rep2_mutdiffsel.txt', 'HSF1i39vsDMSO37-rep3_mutdiffsel.txt', 'HSF1i39vsDMSO39-rep1_mutdiffsel.txt', 'HSF1i39vsDMSO39-rep2_mutdiffsel.txt', 'HSF1i39vsDMSO39-rep3_mutdiffsel.txt', 'HSF1i39vsHSF1i37-rep1_mutdiffsel.txt', 'HSF1i39vsHSF1i37-rep2_mutdiffsel.txt', 'HSF1i39vsHSF1i37-rep3_mutdiffsel.txt', 'Hsp90i37vsDMSO37-rep1_mutdiffsel.txt', 'Hsp90i37vsDMSO37-rep2_mutdiffsel.txt', 'Hsp90i39vsDMSO37-rep3_mutdiffsel.txt', 'Hsp90i39vsDMSO37-rep1_mutdiffsel.txt', 'Hsp90i39vsDMSO37-rep2_mutdiffsel.txt', 'Hsp90i39vsDMSO37-rep3_mutdiffsel.txt']\n",
      "\tstringencyparameter = None\n",
      "\tinfile = HSF1i39vsDMSO37-rep2_mutdiffsel.txt\n",
      "\tletterheight = 2.0\n",
      "\n",
      "\n",
      "Logo plot will be written to HSF1i39vsDMSO37-rep2_diffsel_logoplot.pdf\n",
      "Removing existing file HSF1i39vsDMSO37-rep2_diffsel_logoplot.pdf\n",
      "\n",
      "Reading data from HSF1i39vsDMSO37-rep2_mutdiffsel.txt\n",
      "\n",
      "Now making the logo plot in file HSF1i39vsDMSO37-rep2_diffsel_logoplot.pdf\n",
      "\n",
      "Successfully completed dms_logoplot at Tue Mar 27 18:56:34 2018\n"
     ]
    }
   ],
   "source": [
    "#rep2 HSF1i 39C vs DMSO 37C\n",
    "!dms_logoplot HSF1i39vsDMSO37-rep2_mutdiffsel.txt HSF1i39vsDMSO37-rep2_diffsel_logoplot.pdf --nperline 100 --mapmetric charge --letterheight 2 --diffselheight DMSO39vsDMSO37-rep1_mutdiffsel.txt DMSO39vsDMSO37-rep2_mutdiffsel.txt DMSO39vsDMSO37-rep3_mutdiffsel.txt HSF1i37vsDMSO37-rep1_mutdiffsel.txt HSF1i37vsDMSO37-rep2_mutdiffsel.txt HSF1i37vsDMSO37-rep3_mutdiffsel.txt HSF1i39vsDMSO37-rep1_mutdiffsel.txt HSF1i39vsDMSO37-rep2_mutdiffsel.txt HSF1i39vsDMSO37-rep3_mutdiffsel.txt HSF1i39vsDMSO39-rep1_mutdiffsel.txt HSF1i39vsDMSO39-rep2_mutdiffsel.txt HSF1i39vsDMSO39-rep3_mutdiffsel.txt HSF1i39vsHSF1i37-rep1_mutdiffsel.txt HSF1i39vsHSF1i37-rep2_mutdiffsel.txt HSF1i39vsHSF1i37-rep3_mutdiffsel.txt Hsp90i37vsDMSO37-rep1_mutdiffsel.txt Hsp90i37vsDMSO37-rep2_mutdiffsel.txt Hsp90i39vsDMSO37-rep3_mutdiffsel.txt Hsp90i39vsDMSO37-rep1_mutdiffsel.txt Hsp90i39vsDMSO37-rep2_mutdiffsel.txt Hsp90i39vsDMSO37-rep3_mutdiffsel.txt  "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 33,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Beginning execution of dms_logoplot in directory /Users/annapon/Documents/NP_DMS_2018/2018_NP_DMS/preferences\n",
      "\n",
      "Version information for dms_tools and associated packages.\n",
      "\tTime and date: Tue Mar 27 18:56:37 2018\n",
      "\tPlatform: Darwin-15.6.0-x86_64-i386-64bit\n",
      "\tPython version: 2.7.13 |Anaconda 4.4.0 (x86_64)| (default, Dec 20 2016, 23:05:08)  [GCC 4.2.1 Compatible Apple LLVM 6.0 (clang-600.0.57)]\n",
      "\tdms_tools version: 1.2.2\n",
      "\tBio version: 1.70\n",
      "\tnumpy version: 1.12.1\n",
      "\tscipy version: 0.19.0\n",
      "\tmatplotlib version: 2.0.2\n",
      "\tcython version: 0.25.2\n",
      "\tpystan version: 2.16.0.0\n",
      "\tweblogolib version: 3.5.0\n",
      "\tPyPDF2 version: 1.26.0\n",
      "\tpandas version: 0.20.1\n",
      "\n",
      "Parsed the following arguments:\n",
      "\trestrictdiffsel = None\n",
      "\toverlay3 = None\n",
      "\tcolormap = jet\n",
      "\tnperline = 100\n",
      "\tnumberevery = 10\n",
      "\texcludestop = False\n",
      "\tlogoplot = HSF1i39vsDMSO37-rep3_diffsel_logoplot.pdf\n",
      "\toverlay_cmap = None\n",
      "\tmapmetric = charge\n",
      "\tdiffprefheight = 1.0\n",
      "\tnosepline = False\n",
      "\toverlay1 = None\n",
      "\toverlay2 = None\n",
      "\tdiffselheight = ['DMSO39vsDMSO37-rep1_mutdiffsel.txt', 'DMSO39vsDMSO37-rep2_mutdiffsel.txt', 'DMSO39vsDMSO37-rep3_mutdiffsel.txt', 'HSF1i37vsDMSO37-rep1_mutdiffsel.txt', 'HSF1i37vsDMSO37-rep2_mutdiffsel.txt', 'HSF1i37vsDMSO37-rep3_mutdiffsel.txt', 'HSF1i39vsDMSO37-rep1_mutdiffsel.txt', 'HSF1i39vsDMSO37-rep2_mutdiffsel.txt', 'HSF1i39vsDMSO37-rep3_mutdiffsel.txt', 'HSF1i39vsDMSO39-rep1_mutdiffsel.txt', 'HSF1i39vsDMSO39-rep2_mutdiffsel.txt', 'HSF1i39vsDMSO39-rep3_mutdiffsel.txt', 'HSF1i39vsHSF1i37-rep1_mutdiffsel.txt', 'HSF1i39vsHSF1i37-rep2_mutdiffsel.txt', 'HSF1i39vsHSF1i37-rep3_mutdiffsel.txt', 'Hsp90i37vsDMSO37-rep1_mutdiffsel.txt', 'Hsp90i37vsDMSO37-rep2_mutdiffsel.txt', 'Hsp90i39vsDMSO37-rep3_mutdiffsel.txt', 'Hsp90i39vsDMSO37-rep1_mutdiffsel.txt', 'Hsp90i39vsDMSO37-rep2_mutdiffsel.txt', 'Hsp90i39vsDMSO37-rep3_mutdiffsel.txt']\n",
      "\tstringencyparameter = None\n",
      "\tinfile = HSF1i39vsDMSO37-rep3_mutdiffsel.txt\n",
      "\tletterheight = 2.0\n",
      "\n",
      "\n",
      "Logo plot will be written to HSF1i39vsDMSO37-rep3_diffsel_logoplot.pdf\n",
      "Removing existing file HSF1i39vsDMSO37-rep3_diffsel_logoplot.pdf\n",
      "\n",
      "Reading data from HSF1i39vsDMSO37-rep3_mutdiffsel.txt\n",
      "\n",
      "Now making the logo plot in file HSF1i39vsDMSO37-rep3_diffsel_logoplot.pdf\n",
      "\n",
      "Successfully completed dms_logoplot at Tue Mar 27 18:56:52 2018\n"
     ]
    }
   ],
   "source": [
    "#rep3 HSF1i 39C vs DMSO 37C\n",
    "!dms_logoplot HSF1i39vsDMSO37-rep3_mutdiffsel.txt HSF1i39vsDMSO37-rep3_diffsel_logoplot.pdf --nperline 100 --mapmetric charge --letterheight 2 --diffselheight DMSO39vsDMSO37-rep1_mutdiffsel.txt DMSO39vsDMSO37-rep2_mutdiffsel.txt DMSO39vsDMSO37-rep3_mutdiffsel.txt HSF1i37vsDMSO37-rep1_mutdiffsel.txt HSF1i37vsDMSO37-rep2_mutdiffsel.txt HSF1i37vsDMSO37-rep3_mutdiffsel.txt HSF1i39vsDMSO37-rep1_mutdiffsel.txt HSF1i39vsDMSO37-rep2_mutdiffsel.txt HSF1i39vsDMSO37-rep3_mutdiffsel.txt HSF1i39vsDMSO39-rep1_mutdiffsel.txt HSF1i39vsDMSO39-rep2_mutdiffsel.txt HSF1i39vsDMSO39-rep3_mutdiffsel.txt HSF1i39vsHSF1i37-rep1_mutdiffsel.txt HSF1i39vsHSF1i37-rep2_mutdiffsel.txt HSF1i39vsHSF1i37-rep3_mutdiffsel.txt Hsp90i37vsDMSO37-rep1_mutdiffsel.txt Hsp90i37vsDMSO37-rep2_mutdiffsel.txt Hsp90i39vsDMSO37-rep3_mutdiffsel.txt Hsp90i39vsDMSO37-rep1_mutdiffsel.txt Hsp90i39vsDMSO37-rep2_mutdiffsel.txt Hsp90i39vsDMSO37-rep3_mutdiffsel.txt  "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 34,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Beginning execution of dms_logoplot in directory /Users/annapon/Documents/NP_DMS_2018/2018_NP_DMS/preferences\n",
      "\n",
      "Version information for dms_tools and associated packages.\n",
      "\tTime and date: Tue Mar 27 18:56:55 2018\n",
      "\tPlatform: Darwin-15.6.0-x86_64-i386-64bit\n",
      "\tPython version: 2.7.13 |Anaconda 4.4.0 (x86_64)| (default, Dec 20 2016, 23:05:08)  [GCC 4.2.1 Compatible Apple LLVM 6.0 (clang-600.0.57)]\n",
      "\tdms_tools version: 1.2.2\n",
      "\tBio version: 1.70\n",
      "\tnumpy version: 1.12.1\n",
      "\tscipy version: 0.19.0\n",
      "\tmatplotlib version: 2.0.2\n",
      "\tcython version: 0.25.2\n",
      "\tpystan version: 2.16.0.0\n",
      "\tweblogolib version: 3.5.0\n",
      "\tPyPDF2 version: 1.26.0\n",
      "\tpandas version: 0.20.1\n",
      "\n",
      "Parsed the following arguments:\n",
      "\trestrictdiffsel = None\n",
      "\toverlay3 = None\n",
      "\tcolormap = jet\n",
      "\tnperline = 100\n",
      "\tnumberevery = 10\n",
      "\texcludestop = False\n",
      "\tlogoplot = DMSO39vsDMSO37-rep1_diffsel_logoplot.pdf\n",
      "\toverlay_cmap = None\n",
      "\tmapmetric = charge\n",
      "\tdiffprefheight = 1.0\n",
      "\tnosepline = False\n",
      "\toverlay1 = None\n",
      "\toverlay2 = None\n",
      "\tdiffselheight = ['DMSO39vsDMSO37-rep1_mutdiffsel.txt', 'DMSO39vsDMSO37-rep2_mutdiffsel.txt', 'DMSO39vsDMSO37-rep3_mutdiffsel.txt', 'HSF1i37vsDMSO37-rep1_mutdiffsel.txt', 'HSF1i37vsDMSO37-rep2_mutdiffsel.txt', 'HSF1i37vsDMSO37-rep3_mutdiffsel.txt', 'HSF1i39vsDMSO37-rep1_mutdiffsel.txt', 'HSF1i39vsDMSO37-rep2_mutdiffsel.txt', 'HSF1i39vsDMSO37-rep3_mutdiffsel.txt', 'HSF1i39vsDMSO39-rep1_mutdiffsel.txt', 'HSF1i39vsDMSO39-rep2_mutdiffsel.txt', 'HSF1i39vsDMSO39-rep3_mutdiffsel.txt', 'HSF1i39vsHSF1i37-rep1_mutdiffsel.txt', 'HSF1i39vsHSF1i37-rep2_mutdiffsel.txt', 'HSF1i39vsHSF1i37-rep3_mutdiffsel.txt', 'Hsp90i37vsDMSO37-rep1_mutdiffsel.txt', 'Hsp90i37vsDMSO37-rep2_mutdiffsel.txt', 'Hsp90i39vsDMSO37-rep3_mutdiffsel.txt', 'Hsp90i39vsDMSO37-rep1_mutdiffsel.txt', 'Hsp90i39vsDMSO37-rep2_mutdiffsel.txt', 'Hsp90i39vsDMSO37-rep3_mutdiffsel.txt']\n",
      "\tstringencyparameter = None\n",
      "\tinfile = DMSO39vsDMSO37-rep1_mutdiffsel.txt\n",
      "\tletterheight = 2.0\n",
      "\n",
      "\n",
      "Logo plot will be written to DMSO39vsDMSO37-rep1_diffsel_logoplot.pdf\n",
      "Removing existing file DMSO39vsDMSO37-rep1_diffsel_logoplot.pdf\n",
      "\n",
      "Reading data from DMSO39vsDMSO37-rep1_mutdiffsel.txt\n",
      "\n",
      "Now making the logo plot in file DMSO39vsDMSO37-rep1_diffsel_logoplot.pdf\n",
      "\n",
      "Successfully completed dms_logoplot at Tue Mar 27 18:57:11 2018\n"
     ]
    }
   ],
   "source": [
    "#rep1 DMSO 39C vs DMSO 37C\n",
    "!dms_logoplot DMSO39vsDMSO37-rep1_mutdiffsel.txt DMSO39vsDMSO37-rep1_diffsel_logoplot.pdf --nperline 100 --mapmetric charge --letterheight 2 --diffselheight DMSO39vsDMSO37-rep1_mutdiffsel.txt DMSO39vsDMSO37-rep2_mutdiffsel.txt DMSO39vsDMSO37-rep3_mutdiffsel.txt HSF1i37vsDMSO37-rep1_mutdiffsel.txt HSF1i37vsDMSO37-rep2_mutdiffsel.txt HSF1i37vsDMSO37-rep3_mutdiffsel.txt HSF1i39vsDMSO37-rep1_mutdiffsel.txt HSF1i39vsDMSO37-rep2_mutdiffsel.txt HSF1i39vsDMSO37-rep3_mutdiffsel.txt HSF1i39vsDMSO39-rep1_mutdiffsel.txt HSF1i39vsDMSO39-rep2_mutdiffsel.txt HSF1i39vsDMSO39-rep3_mutdiffsel.txt HSF1i39vsHSF1i37-rep1_mutdiffsel.txt HSF1i39vsHSF1i37-rep2_mutdiffsel.txt HSF1i39vsHSF1i37-rep3_mutdiffsel.txt Hsp90i37vsDMSO37-rep1_mutdiffsel.txt Hsp90i37vsDMSO37-rep2_mutdiffsel.txt Hsp90i39vsDMSO37-rep3_mutdiffsel.txt Hsp90i39vsDMSO37-rep1_mutdiffsel.txt Hsp90i39vsDMSO37-rep2_mutdiffsel.txt Hsp90i39vsDMSO37-rep3_mutdiffsel.txt  "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 35,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Beginning execution of dms_logoplot in directory /Users/annapon/Documents/NP_DMS_2018/2018_NP_DMS/preferences\n",
      "\n",
      "Version information for dms_tools and associated packages.\n",
      "\tTime and date: Tue Mar 27 18:57:40 2018\n",
      "\tPlatform: Darwin-15.6.0-x86_64-i386-64bit\n",
      "\tPython version: 2.7.13 |Anaconda 4.4.0 (x86_64)| (default, Dec 20 2016, 23:05:08)  [GCC 4.2.1 Compatible Apple LLVM 6.0 (clang-600.0.57)]\n",
      "\tdms_tools version: 1.2.2\n",
      "\tBio version: 1.70\n",
      "\tnumpy version: 1.12.1\n",
      "\tscipy version: 0.19.0\n",
      "\tmatplotlib version: 2.0.2\n",
      "\tcython version: 0.25.2\n",
      "\tpystan version: 2.16.0.0\n",
      "\tweblogolib version: 3.5.0\n",
      "\tPyPDF2 version: 1.26.0\n",
      "\tpandas version: 0.20.1\n",
      "\n",
      "Parsed the following arguments:\n",
      "\trestrictdiffsel = None\n",
      "\toverlay3 = None\n",
      "\tcolormap = jet\n",
      "\tnperline = 100\n",
      "\tnumberevery = 10\n",
      "\texcludestop = False\n",
      "\tlogoplot = DMSO39vsDMSO37-rep2_diffsel_logoplot.pdf\n",
      "\toverlay_cmap = None\n",
      "\tmapmetric = charge\n",
      "\tdiffprefheight = 1.0\n",
      "\tnosepline = False\n",
      "\toverlay1 = None\n",
      "\toverlay2 = None\n",
      "\tdiffselheight = ['DMSO39vsDMSO37-rep1_mutdiffsel.txt', 'DMSO39vsDMSO37-rep2_mutdiffsel.txt', 'DMSO39vsDMSO37-rep3_mutdiffsel.txt', 'HSF1i37vsDMSO37-rep1_mutdiffsel.txt', 'HSF1i37vsDMSO37-rep2_mutdiffsel.txt', 'HSF1i37vsDMSO37-rep3_mutdiffsel.txt', 'HSF1i39vsDMSO37-rep1_mutdiffsel.txt', 'HSF1i39vsDMSO37-rep2_mutdiffsel.txt', 'HSF1i39vsDMSO37-rep3_mutdiffsel.txt', 'HSF1i39vsDMSO39-rep1_mutdiffsel.txt', 'HSF1i39vsDMSO39-rep2_mutdiffsel.txt', 'HSF1i39vsDMSO39-rep3_mutdiffsel.txt', 'HSF1i39vsHSF1i37-rep1_mutdiffsel.txt', 'HSF1i39vsHSF1i37-rep2_mutdiffsel.txt', 'HSF1i39vsHSF1i37-rep3_mutdiffsel.txt', 'Hsp90i37vsDMSO37-rep1_mutdiffsel.txt', 'Hsp90i37vsDMSO37-rep2_mutdiffsel.txt', 'Hsp90i39vsDMSO37-rep3_mutdiffsel.txt', 'Hsp90i39vsDMSO37-rep1_mutdiffsel.txt', 'Hsp90i39vsDMSO37-rep2_mutdiffsel.txt', 'Hsp90i39vsDMSO37-rep3_mutdiffsel.txt']\n",
      "\tstringencyparameter = None\n",
      "\tinfile = DMSO39vsDMSO37-rep2_mutdiffsel.txt\n",
      "\tletterheight = 2.0\n",
      "\n",
      "\n",
      "Logo plot will be written to DMSO39vsDMSO37-rep2_diffsel_logoplot.pdf\n",
      "Removing existing file DMSO39vsDMSO37-rep2_diffsel_logoplot.pdf\n",
      "\n",
      "Reading data from DMSO39vsDMSO37-rep2_mutdiffsel.txt\n",
      "\n",
      "Now making the logo plot in file DMSO39vsDMSO37-rep2_diffsel_logoplot.pdf\n",
      "\n",
      "Successfully completed dms_logoplot at Tue Mar 27 18:57:56 2018\n"
     ]
    }
   ],
   "source": [
    "#rep2 DMSO 39C vs DMSO 37C\n",
    "!dms_logoplot DMSO39vsDMSO37-rep2_mutdiffsel.txt DMSO39vsDMSO37-rep2_diffsel_logoplot.pdf --nperline 100 --mapmetric charge --letterheight 2 --diffselheight DMSO39vsDMSO37-rep1_mutdiffsel.txt DMSO39vsDMSO37-rep2_mutdiffsel.txt DMSO39vsDMSO37-rep3_mutdiffsel.txt HSF1i37vsDMSO37-rep1_mutdiffsel.txt HSF1i37vsDMSO37-rep2_mutdiffsel.txt HSF1i37vsDMSO37-rep3_mutdiffsel.txt HSF1i39vsDMSO37-rep1_mutdiffsel.txt HSF1i39vsDMSO37-rep2_mutdiffsel.txt HSF1i39vsDMSO37-rep3_mutdiffsel.txt HSF1i39vsDMSO39-rep1_mutdiffsel.txt HSF1i39vsDMSO39-rep2_mutdiffsel.txt HSF1i39vsDMSO39-rep3_mutdiffsel.txt HSF1i39vsHSF1i37-rep1_mutdiffsel.txt HSF1i39vsHSF1i37-rep2_mutdiffsel.txt HSF1i39vsHSF1i37-rep3_mutdiffsel.txt Hsp90i37vsDMSO37-rep1_mutdiffsel.txt Hsp90i37vsDMSO37-rep2_mutdiffsel.txt Hsp90i39vsDMSO37-rep3_mutdiffsel.txt Hsp90i39vsDMSO37-rep1_mutdiffsel.txt Hsp90i39vsDMSO37-rep2_mutdiffsel.txt Hsp90i39vsDMSO37-rep3_mutdiffsel.txt  "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 36,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Beginning execution of dms_logoplot in directory /Users/annapon/Documents/NP_DMS_2018/2018_NP_DMS/preferences\n",
      "\n",
      "Version information for dms_tools and associated packages.\n",
      "\tTime and date: Tue Mar 27 18:57:59 2018\n",
      "\tPlatform: Darwin-15.6.0-x86_64-i386-64bit\n",
      "\tPython version: 2.7.13 |Anaconda 4.4.0 (x86_64)| (default, Dec 20 2016, 23:05:08)  [GCC 4.2.1 Compatible Apple LLVM 6.0 (clang-600.0.57)]\n",
      "\tdms_tools version: 1.2.2\n",
      "\tBio version: 1.70\n",
      "\tnumpy version: 1.12.1\n",
      "\tscipy version: 0.19.0\n",
      "\tmatplotlib version: 2.0.2\n",
      "\tcython version: 0.25.2\n",
      "\tpystan version: 2.16.0.0\n",
      "\tweblogolib version: 3.5.0\n",
      "\tPyPDF2 version: 1.26.0\n",
      "\tpandas version: 0.20.1\n",
      "\n",
      "Parsed the following arguments:\n",
      "\trestrictdiffsel = None\n",
      "\toverlay3 = None\n",
      "\tcolormap = jet\n",
      "\tnperline = 100\n",
      "\tnumberevery = 10\n",
      "\texcludestop = False\n",
      "\tlogoplot = DMSO39vsDMSO37-rep3_diffsel_logoplot.pdf\n",
      "\toverlay_cmap = None\n",
      "\tmapmetric = charge\n",
      "\tdiffprefheight = 1.0\n",
      "\tnosepline = False\n",
      "\toverlay1 = None\n",
      "\toverlay2 = None\n",
      "\tdiffselheight = ['DMSO39vsDMSO37-rep1_mutdiffsel.txt', 'DMSO39vsDMSO37-rep2_mutdiffsel.txt', 'DMSO39vsDMSO37-rep3_mutdiffsel.txt', 'HSF1i37vsDMSO37-rep1_mutdiffsel.txt', 'HSF1i37vsDMSO37-rep2_mutdiffsel.txt', 'HSF1i37vsDMSO37-rep3_mutdiffsel.txt', 'HSF1i39vsDMSO37-rep1_mutdiffsel.txt', 'HSF1i39vsDMSO37-rep2_mutdiffsel.txt', 'HSF1i39vsDMSO37-rep3_mutdiffsel.txt', 'HSF1i39vsDMSO39-rep1_mutdiffsel.txt', 'HSF1i39vsDMSO39-rep2_mutdiffsel.txt', 'HSF1i39vsDMSO39-rep3_mutdiffsel.txt', 'HSF1i39vsHSF1i37-rep1_mutdiffsel.txt', 'HSF1i39vsHSF1i37-rep2_mutdiffsel.txt', 'HSF1i39vsHSF1i37-rep3_mutdiffsel.txt', 'Hsp90i37vsDMSO37-rep1_mutdiffsel.txt', 'Hsp90i37vsDMSO37-rep2_mutdiffsel.txt', 'Hsp90i39vsDMSO37-rep3_mutdiffsel.txt', 'Hsp90i39vsDMSO37-rep1_mutdiffsel.txt', 'Hsp90i39vsDMSO37-rep2_mutdiffsel.txt', 'Hsp90i39vsDMSO37-rep3_mutdiffsel.txt']\n",
      "\tstringencyparameter = None\n",
      "\tinfile = DMSO39vsDMSO37-rep3_mutdiffsel.txt\n",
      "\tletterheight = 2.0\n",
      "\n",
      "\n",
      "Logo plot will be written to DMSO39vsDMSO37-rep3_diffsel_logoplot.pdf\n",
      "Removing existing file DMSO39vsDMSO37-rep3_diffsel_logoplot.pdf\n",
      "\n",
      "Reading data from DMSO39vsDMSO37-rep3_mutdiffsel.txt\n",
      "\n",
      "Now making the logo plot in file DMSO39vsDMSO37-rep3_diffsel_logoplot.pdf\n",
      "\n",
      "Successfully completed dms_logoplot at Tue Mar 27 18:58:15 2018\n"
     ]
    }
   ],
   "source": [
    "#rep3 DMSO 39C vs DMSO 37C\n",
    "!dms_logoplot DMSO39vsDMSO37-rep3_mutdiffsel.txt DMSO39vsDMSO37-rep3_diffsel_logoplot.pdf --nperline 100 --mapmetric charge --letterheight 2 --diffselheight DMSO39vsDMSO37-rep1_mutdiffsel.txt DMSO39vsDMSO37-rep2_mutdiffsel.txt DMSO39vsDMSO37-rep3_mutdiffsel.txt HSF1i37vsDMSO37-rep1_mutdiffsel.txt HSF1i37vsDMSO37-rep2_mutdiffsel.txt HSF1i37vsDMSO37-rep3_mutdiffsel.txt HSF1i39vsDMSO37-rep1_mutdiffsel.txt HSF1i39vsDMSO37-rep2_mutdiffsel.txt HSF1i39vsDMSO37-rep3_mutdiffsel.txt HSF1i39vsDMSO39-rep1_mutdiffsel.txt HSF1i39vsDMSO39-rep2_mutdiffsel.txt HSF1i39vsDMSO39-rep3_mutdiffsel.txt HSF1i39vsHSF1i37-rep1_mutdiffsel.txt HSF1i39vsHSF1i37-rep2_mutdiffsel.txt HSF1i39vsHSF1i37-rep3_mutdiffsel.txt Hsp90i37vsDMSO37-rep1_mutdiffsel.txt Hsp90i37vsDMSO37-rep2_mutdiffsel.txt Hsp90i39vsDMSO37-rep3_mutdiffsel.txt Hsp90i39vsDMSO37-rep1_mutdiffsel.txt Hsp90i39vsDMSO37-rep2_mutdiffsel.txt Hsp90i39vsDMSO37-rep3_mutdiffsel.txt  "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 37,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Beginning execution of dms_logoplot in directory /Users/annapon/Documents/NP_DMS_2018/2018_NP_DMS/preferences\n",
      "\n",
      "Version information for dms_tools and associated packages.\n",
      "\tTime and date: Tue Mar 27 18:58:17 2018\n",
      "\tPlatform: Darwin-15.6.0-x86_64-i386-64bit\n",
      "\tPython version: 2.7.13 |Anaconda 4.4.0 (x86_64)| (default, Dec 20 2016, 23:05:08)  [GCC 4.2.1 Compatible Apple LLVM 6.0 (clang-600.0.57)]\n",
      "\tdms_tools version: 1.2.2\n",
      "\tBio version: 1.70\n",
      "\tnumpy version: 1.12.1\n",
      "\tscipy version: 0.19.0\n",
      "\tmatplotlib version: 2.0.2\n",
      "\tcython version: 0.25.2\n",
      "\tpystan version: 2.16.0.0\n",
      "\tweblogolib version: 3.5.0\n",
      "\tPyPDF2 version: 1.26.0\n",
      "\tpandas version: 0.20.1\n",
      "\n",
      "Parsed the following arguments:\n",
      "\trestrictdiffsel = None\n",
      "\toverlay3 = None\n",
      "\tcolormap = jet\n",
      "\tnperline = 100\n",
      "\tnumberevery = 10\n",
      "\texcludestop = False\n",
      "\tlogoplot = HSF1i37vsDMSO37-rep1_diffsel_logoplot.pdf\n",
      "\toverlay_cmap = None\n",
      "\tmapmetric = charge\n",
      "\tdiffprefheight = 1.0\n",
      "\tnosepline = False\n",
      "\toverlay1 = None\n",
      "\toverlay2 = None\n",
      "\tdiffselheight = ['DMSO39vsDMSO37-rep1_mutdiffsel.txt', 'DMSO39vsDMSO37-rep2_mutdiffsel.txt', 'DMSO39vsDMSO37-rep3_mutdiffsel.txt', 'HSF1i37vsDMSO37-rep1_mutdiffsel.txt', 'HSF1i37vsDMSO37-rep2_mutdiffsel.txt', 'HSF1i37vsDMSO37-rep3_mutdiffsel.txt', 'HSF1i39vsDMSO37-rep1_mutdiffsel.txt', 'HSF1i39vsDMSO37-rep2_mutdiffsel.txt', 'HSF1i39vsDMSO37-rep3_mutdiffsel.txt', 'HSF1i39vsDMSO39-rep1_mutdiffsel.txt', 'HSF1i39vsDMSO39-rep2_mutdiffsel.txt', 'HSF1i39vsDMSO39-rep3_mutdiffsel.txt', 'HSF1i39vsHSF1i37-rep1_mutdiffsel.txt', 'HSF1i39vsHSF1i37-rep2_mutdiffsel.txt', 'HSF1i39vsHSF1i37-rep3_mutdiffsel.txt', 'Hsp90i37vsDMSO37-rep1_mutdiffsel.txt', 'Hsp90i37vsDMSO37-rep2_mutdiffsel.txt', 'Hsp90i39vsDMSO37-rep3_mutdiffsel.txt', 'Hsp90i39vsDMSO37-rep1_mutdiffsel.txt', 'Hsp90i39vsDMSO37-rep2_mutdiffsel.txt', 'Hsp90i39vsDMSO37-rep3_mutdiffsel.txt']\n",
      "\tstringencyparameter = None\n",
      "\tinfile = HSF1i37vsDMSO37-rep1_mutdiffsel.txt\n",
      "\tletterheight = 2.0\n",
      "\n",
      "\n",
      "Logo plot will be written to HSF1i37vsDMSO37-rep1_diffsel_logoplot.pdf\n",
      "Removing existing file HSF1i37vsDMSO37-rep1_diffsel_logoplot.pdf\n",
      "\n",
      "Reading data from HSF1i37vsDMSO37-rep1_mutdiffsel.txt\n",
      "\n",
      "Now making the logo plot in file HSF1i37vsDMSO37-rep1_diffsel_logoplot.pdf\n",
      "\n",
      "Successfully completed dms_logoplot at Tue Mar 27 18:58:33 2018\n"
     ]
    }
   ],
   "source": [
    "#rep1 HSF1i 37C vs DMSO 37C\n",
    "!dms_logoplot HSF1i37vsDMSO37-rep1_mutdiffsel.txt HSF1i37vsDMSO37-rep1_diffsel_logoplot.pdf --nperline 100 --mapmetric charge --letterheight 2 --diffselheight DMSO39vsDMSO37-rep1_mutdiffsel.txt DMSO39vsDMSO37-rep2_mutdiffsel.txt DMSO39vsDMSO37-rep3_mutdiffsel.txt HSF1i37vsDMSO37-rep1_mutdiffsel.txt HSF1i37vsDMSO37-rep2_mutdiffsel.txt HSF1i37vsDMSO37-rep3_mutdiffsel.txt HSF1i39vsDMSO37-rep1_mutdiffsel.txt HSF1i39vsDMSO37-rep2_mutdiffsel.txt HSF1i39vsDMSO37-rep3_mutdiffsel.txt HSF1i39vsDMSO39-rep1_mutdiffsel.txt HSF1i39vsDMSO39-rep2_mutdiffsel.txt HSF1i39vsDMSO39-rep3_mutdiffsel.txt HSF1i39vsHSF1i37-rep1_mutdiffsel.txt HSF1i39vsHSF1i37-rep2_mutdiffsel.txt HSF1i39vsHSF1i37-rep3_mutdiffsel.txt Hsp90i37vsDMSO37-rep1_mutdiffsel.txt Hsp90i37vsDMSO37-rep2_mutdiffsel.txt Hsp90i39vsDMSO37-rep3_mutdiffsel.txt Hsp90i39vsDMSO37-rep1_mutdiffsel.txt Hsp90i39vsDMSO37-rep2_mutdiffsel.txt Hsp90i39vsDMSO37-rep3_mutdiffsel.txt  "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 38,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Beginning execution of dms_logoplot in directory /Users/annapon/Documents/NP_DMS_2018/2018_NP_DMS/preferences\n",
      "\n",
      "Version information for dms_tools and associated packages.\n",
      "\tTime and date: Tue Mar 27 18:58:37 2018\n",
      "\tPlatform: Darwin-15.6.0-x86_64-i386-64bit\n",
      "\tPython version: 2.7.13 |Anaconda 4.4.0 (x86_64)| (default, Dec 20 2016, 23:05:08)  [GCC 4.2.1 Compatible Apple LLVM 6.0 (clang-600.0.57)]\n",
      "\tdms_tools version: 1.2.2\n",
      "\tBio version: 1.70\n",
      "\tnumpy version: 1.12.1\n",
      "\tscipy version: 0.19.0\n",
      "\tmatplotlib version: 2.0.2\n",
      "\tcython version: 0.25.2\n",
      "\tpystan version: 2.16.0.0\n",
      "\tweblogolib version: 3.5.0\n",
      "\tPyPDF2 version: 1.26.0\n",
      "\tpandas version: 0.20.1\n",
      "\n",
      "Parsed the following arguments:\n",
      "\trestrictdiffsel = None\n",
      "\toverlay3 = None\n",
      "\tcolormap = jet\n",
      "\tnperline = 100\n",
      "\tnumberevery = 10\n",
      "\texcludestop = False\n",
      "\tlogoplot = HSF1i37vsDMSO37-rep2_diffsel_logoplot.pdf\n",
      "\toverlay_cmap = None\n",
      "\tmapmetric = charge\n",
      "\tdiffprefheight = 1.0\n",
      "\tnosepline = False\n",
      "\toverlay1 = None\n",
      "\toverlay2 = None\n",
      "\tdiffselheight = ['DMSO39vsDMSO37-rep1_mutdiffsel.txt', 'DMSO39vsDMSO37-rep2_mutdiffsel.txt', 'DMSO39vsDMSO37-rep3_mutdiffsel.txt', 'HSF1i37vsDMSO37-rep1_mutdiffsel.txt', 'HSF1i37vsDMSO37-rep2_mutdiffsel.txt', 'HSF1i37vsDMSO37-rep3_mutdiffsel.txt', 'HSF1i39vsDMSO37-rep1_mutdiffsel.txt', 'HSF1i39vsDMSO37-rep2_mutdiffsel.txt', 'HSF1i39vsDMSO37-rep3_mutdiffsel.txt', 'HSF1i39vsDMSO39-rep1_mutdiffsel.txt', 'HSF1i39vsDMSO39-rep2_mutdiffsel.txt', 'HSF1i39vsDMSO39-rep3_mutdiffsel.txt', 'HSF1i39vsHSF1i37-rep1_mutdiffsel.txt', 'HSF1i39vsHSF1i37-rep2_mutdiffsel.txt', 'HSF1i39vsHSF1i37-rep3_mutdiffsel.txt', 'Hsp90i37vsDMSO37-rep1_mutdiffsel.txt', 'Hsp90i37vsDMSO37-rep2_mutdiffsel.txt', 'Hsp90i39vsDMSO37-rep3_mutdiffsel.txt', 'Hsp90i39vsDMSO37-rep1_mutdiffsel.txt', 'Hsp90i39vsDMSO37-rep2_mutdiffsel.txt', 'Hsp90i39vsDMSO37-rep3_mutdiffsel.txt']\n",
      "\tstringencyparameter = None\n",
      "\tinfile = HSF1i37vsDMSO37-rep2_mutdiffsel.txt\n",
      "\tletterheight = 2.0\n",
      "\n",
      "\n",
      "Logo plot will be written to HSF1i37vsDMSO37-rep2_diffsel_logoplot.pdf\n",
      "Removing existing file HSF1i37vsDMSO37-rep2_diffsel_logoplot.pdf\n",
      "\n",
      "Reading data from HSF1i37vsDMSO37-rep2_mutdiffsel.txt\n",
      "\n",
      "Now making the logo plot in file HSF1i37vsDMSO37-rep2_diffsel_logoplot.pdf\n",
      "\n",
      "Successfully completed dms_logoplot at Tue Mar 27 18:58:52 2018\n"
     ]
    }
   ],
   "source": [
    "#rep2 HSF1i 37C vs DMSO 37C\n",
    "!dms_logoplot HSF1i37vsDMSO37-rep2_mutdiffsel.txt HSF1i37vsDMSO37-rep2_diffsel_logoplot.pdf --nperline 100 --mapmetric charge --letterheight 2 --diffselheight DMSO39vsDMSO37-rep1_mutdiffsel.txt DMSO39vsDMSO37-rep2_mutdiffsel.txt DMSO39vsDMSO37-rep3_mutdiffsel.txt HSF1i37vsDMSO37-rep1_mutdiffsel.txt HSF1i37vsDMSO37-rep2_mutdiffsel.txt HSF1i37vsDMSO37-rep3_mutdiffsel.txt HSF1i39vsDMSO37-rep1_mutdiffsel.txt HSF1i39vsDMSO37-rep2_mutdiffsel.txt HSF1i39vsDMSO37-rep3_mutdiffsel.txt HSF1i39vsDMSO39-rep1_mutdiffsel.txt HSF1i39vsDMSO39-rep2_mutdiffsel.txt HSF1i39vsDMSO39-rep3_mutdiffsel.txt HSF1i39vsHSF1i37-rep1_mutdiffsel.txt HSF1i39vsHSF1i37-rep2_mutdiffsel.txt HSF1i39vsHSF1i37-rep3_mutdiffsel.txt Hsp90i37vsDMSO37-rep1_mutdiffsel.txt Hsp90i37vsDMSO37-rep2_mutdiffsel.txt Hsp90i39vsDMSO37-rep3_mutdiffsel.txt Hsp90i39vsDMSO37-rep1_mutdiffsel.txt Hsp90i39vsDMSO37-rep2_mutdiffsel.txt Hsp90i39vsDMSO37-rep3_mutdiffsel.txt  "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 39,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Beginning execution of dms_logoplot in directory /Users/annapon/Documents/NP_DMS_2018/2018_NP_DMS/preferences\n",
      "\n",
      "Version information for dms_tools and associated packages.\n",
      "\tTime and date: Tue Mar 27 18:58:55 2018\n",
      "\tPlatform: Darwin-15.6.0-x86_64-i386-64bit\n",
      "\tPython version: 2.7.13 |Anaconda 4.4.0 (x86_64)| (default, Dec 20 2016, 23:05:08)  [GCC 4.2.1 Compatible Apple LLVM 6.0 (clang-600.0.57)]\n",
      "\tdms_tools version: 1.2.2\n",
      "\tBio version: 1.70\n",
      "\tnumpy version: 1.12.1\n",
      "\tscipy version: 0.19.0\n",
      "\tmatplotlib version: 2.0.2\n",
      "\tcython version: 0.25.2\n",
      "\tpystan version: 2.16.0.0\n",
      "\tweblogolib version: 3.5.0\n",
      "\tPyPDF2 version: 1.26.0\n",
      "\tpandas version: 0.20.1\n",
      "\n",
      "Parsed the following arguments:\n",
      "\trestrictdiffsel = None\n",
      "\toverlay3 = None\n",
      "\tcolormap = jet\n",
      "\tnperline = 100\n",
      "\tnumberevery = 10\n",
      "\texcludestop = False\n",
      "\tlogoplot = HSF1i37vsDMSO37-rep3_diffsel_logoplot.pdf\n",
      "\toverlay_cmap = None\n",
      "\tmapmetric = charge\n",
      "\tdiffprefheight = 1.0\n",
      "\tnosepline = False\n",
      "\toverlay1 = None\n",
      "\toverlay2 = None\n",
      "\tdiffselheight = ['DMSO39vsDMSO37-rep1_mutdiffsel.txt', 'DMSO39vsDMSO37-rep2_mutdiffsel.txt', 'DMSO39vsDMSO37-rep3_mutdiffsel.txt', 'HSF1i37vsDMSO37-rep1_mutdiffsel.txt', 'HSF1i37vsDMSO37-rep2_mutdiffsel.txt', 'HSF1i37vsDMSO37-rep3_mutdiffsel.txt', 'HSF1i39vsDMSO37-rep1_mutdiffsel.txt', 'HSF1i39vsDMSO37-rep2_mutdiffsel.txt', 'HSF1i39vsDMSO37-rep3_mutdiffsel.txt', 'HSF1i39vsDMSO39-rep1_mutdiffsel.txt', 'HSF1i39vsDMSO39-rep2_mutdiffsel.txt', 'HSF1i39vsDMSO39-rep3_mutdiffsel.txt', 'HSF1i39vsHSF1i37-rep1_mutdiffsel.txt', 'HSF1i39vsHSF1i37-rep2_mutdiffsel.txt', 'HSF1i39vsHSF1i37-rep3_mutdiffsel.txt', 'Hsp90i37vsDMSO37-rep1_mutdiffsel.txt', 'Hsp90i37vsDMSO37-rep2_mutdiffsel.txt', 'Hsp90i39vsDMSO37-rep3_mutdiffsel.txt', 'Hsp90i39vsDMSO37-rep1_mutdiffsel.txt', 'Hsp90i39vsDMSO37-rep2_mutdiffsel.txt', 'Hsp90i39vsDMSO37-rep3_mutdiffsel.txt']\n",
      "\tstringencyparameter = None\n",
      "\tinfile = HSF1i37vsDMSO37-rep3_mutdiffsel.txt\n",
      "\tletterheight = 2.0\n",
      "\n",
      "\n",
      "Logo plot will be written to HSF1i37vsDMSO37-rep3_diffsel_logoplot.pdf\n",
      "Removing existing file HSF1i37vsDMSO37-rep3_diffsel_logoplot.pdf\n",
      "\n",
      "Reading data from HSF1i37vsDMSO37-rep3_mutdiffsel.txt\n",
      "\n",
      "Now making the logo plot in file HSF1i37vsDMSO37-rep3_diffsel_logoplot.pdf\n",
      "\n",
      "Successfully completed dms_logoplot at Tue Mar 27 18:59:11 2018\n"
     ]
    }
   ],
   "source": [
    "#rep3 HSF1i 37C vs DMSO 37C\n",
    "!dms_logoplot HSF1i37vsDMSO37-rep3_mutdiffsel.txt HSF1i37vsDMSO37-rep3_diffsel_logoplot.pdf --nperline 100 --mapmetric charge --letterheight 2 --diffselheight DMSO39vsDMSO37-rep1_mutdiffsel.txt DMSO39vsDMSO37-rep2_mutdiffsel.txt DMSO39vsDMSO37-rep3_mutdiffsel.txt HSF1i37vsDMSO37-rep1_mutdiffsel.txt HSF1i37vsDMSO37-rep2_mutdiffsel.txt HSF1i37vsDMSO37-rep3_mutdiffsel.txt HSF1i39vsDMSO37-rep1_mutdiffsel.txt HSF1i39vsDMSO37-rep2_mutdiffsel.txt HSF1i39vsDMSO37-rep3_mutdiffsel.txt HSF1i39vsDMSO39-rep1_mutdiffsel.txt HSF1i39vsDMSO39-rep2_mutdiffsel.txt HSF1i39vsDMSO39-rep3_mutdiffsel.txt HSF1i39vsHSF1i37-rep1_mutdiffsel.txt HSF1i39vsHSF1i37-rep2_mutdiffsel.txt HSF1i39vsHSF1i37-rep3_mutdiffsel.txt Hsp90i37vsDMSO37-rep1_mutdiffsel.txt Hsp90i37vsDMSO37-rep2_mutdiffsel.txt Hsp90i39vsDMSO37-rep3_mutdiffsel.txt Hsp90i39vsDMSO37-rep1_mutdiffsel.txt Hsp90i39vsDMSO37-rep2_mutdiffsel.txt Hsp90i39vsDMSO37-rep3_mutdiffsel.txt  "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 40,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Beginning execution of dms_logoplot in directory /Users/annapon/Documents/NP_DMS_2018/2018_NP_DMS/preferences\n",
      "\n",
      "Version information for dms_tools and associated packages.\n",
      "\tTime and date: Tue Mar 27 18:59:14 2018\n",
      "\tPlatform: Darwin-15.6.0-x86_64-i386-64bit\n",
      "\tPython version: 2.7.13 |Anaconda 4.4.0 (x86_64)| (default, Dec 20 2016, 23:05:08)  [GCC 4.2.1 Compatible Apple LLVM 6.0 (clang-600.0.57)]\n",
      "\tdms_tools version: 1.2.2\n",
      "\tBio version: 1.70\n",
      "\tnumpy version: 1.12.1\n",
      "\tscipy version: 0.19.0\n",
      "\tmatplotlib version: 2.0.2\n",
      "\tcython version: 0.25.2\n",
      "\tpystan version: 2.16.0.0\n",
      "\tweblogolib version: 3.5.0\n",
      "\tPyPDF2 version: 1.26.0\n",
      "\tpandas version: 0.20.1\n",
      "\n",
      "Parsed the following arguments:\n",
      "\trestrictdiffsel = None\n",
      "\toverlay3 = None\n",
      "\tcolormap = jet\n",
      "\tnperline = 100\n",
      "\tnumberevery = 10\n",
      "\texcludestop = False\n",
      "\tlogoplot = HSF1i39vsDMSO39-rep1_diffsel_logoplot.pdf\n",
      "\toverlay_cmap = None\n",
      "\tmapmetric = charge\n",
      "\tdiffprefheight = 1.0\n",
      "\tnosepline = False\n",
      "\toverlay1 = None\n",
      "\toverlay2 = None\n",
      "\tdiffselheight = ['DMSO39vsDMSO37-rep1_mutdiffsel.txt', 'DMSO39vsDMSO37-rep2_mutdiffsel.txt', 'DMSO39vsDMSO37-rep3_mutdiffsel.txt', 'HSF1i37vsDMSO37-rep1_mutdiffsel.txt', 'HSF1i37vsDMSO37-rep2_mutdiffsel.txt', 'HSF1i37vsDMSO37-rep3_mutdiffsel.txt', 'HSF1i39vsDMSO37-rep1_mutdiffsel.txt', 'HSF1i39vsDMSO37-rep2_mutdiffsel.txt', 'HSF1i39vsDMSO37-rep3_mutdiffsel.txt', 'HSF1i39vsDMSO39-rep1_mutdiffsel.txt', 'HSF1i39vsDMSO39-rep2_mutdiffsel.txt', 'HSF1i39vsDMSO39-rep3_mutdiffsel.txt', 'HSF1i39vsHSF1i37-rep1_mutdiffsel.txt', 'HSF1i39vsHSF1i37-rep2_mutdiffsel.txt', 'HSF1i39vsHSF1i37-rep3_mutdiffsel.txt', 'Hsp90i37vsDMSO37-rep1_mutdiffsel.txt', 'Hsp90i37vsDMSO37-rep2_mutdiffsel.txt', 'Hsp90i39vsDMSO37-rep3_mutdiffsel.txt', 'Hsp90i39vsDMSO37-rep1_mutdiffsel.txt', 'Hsp90i39vsDMSO37-rep2_mutdiffsel.txt', 'Hsp90i39vsDMSO37-rep3_mutdiffsel.txt']\n",
      "\tstringencyparameter = None\n",
      "\tinfile = HSF1i39vsDMSO39-rep1_mutdiffsel.txt\n",
      "\tletterheight = 2.0\n",
      "\n",
      "\n",
      "Logo plot will be written to HSF1i39vsDMSO39-rep1_diffsel_logoplot.pdf\n",
      "Removing existing file HSF1i39vsDMSO39-rep1_diffsel_logoplot.pdf\n",
      "\n",
      "Reading data from HSF1i39vsDMSO39-rep1_mutdiffsel.txt\n",
      "\n",
      "Now making the logo plot in file HSF1i39vsDMSO39-rep1_diffsel_logoplot.pdf\n",
      "\n",
      "Successfully completed dms_logoplot at Tue Mar 27 18:59:29 2018\n"
     ]
    }
   ],
   "source": [
    "#rep1 HSF1i 39C vs DMSO 39C\n",
    "!dms_logoplot HSF1i39vsDMSO39-rep1_mutdiffsel.txt HSF1i39vsDMSO39-rep1_diffsel_logoplot.pdf --nperline 100 --mapmetric charge --letterheight 2 --diffselheight DMSO39vsDMSO37-rep1_mutdiffsel.txt DMSO39vsDMSO37-rep2_mutdiffsel.txt DMSO39vsDMSO37-rep3_mutdiffsel.txt HSF1i37vsDMSO37-rep1_mutdiffsel.txt HSF1i37vsDMSO37-rep2_mutdiffsel.txt HSF1i37vsDMSO37-rep3_mutdiffsel.txt HSF1i39vsDMSO37-rep1_mutdiffsel.txt HSF1i39vsDMSO37-rep2_mutdiffsel.txt HSF1i39vsDMSO37-rep3_mutdiffsel.txt HSF1i39vsDMSO39-rep1_mutdiffsel.txt HSF1i39vsDMSO39-rep2_mutdiffsel.txt HSF1i39vsDMSO39-rep3_mutdiffsel.txt HSF1i39vsHSF1i37-rep1_mutdiffsel.txt HSF1i39vsHSF1i37-rep2_mutdiffsel.txt HSF1i39vsHSF1i37-rep3_mutdiffsel.txt Hsp90i37vsDMSO37-rep1_mutdiffsel.txt Hsp90i37vsDMSO37-rep2_mutdiffsel.txt Hsp90i39vsDMSO37-rep3_mutdiffsel.txt Hsp90i39vsDMSO37-rep1_mutdiffsel.txt Hsp90i39vsDMSO37-rep2_mutdiffsel.txt Hsp90i39vsDMSO37-rep3_mutdiffsel.txt  "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 41,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Beginning execution of dms_logoplot in directory /Users/annapon/Documents/NP_DMS_2018/2018_NP_DMS/preferences\n",
      "\n",
      "Version information for dms_tools and associated packages.\n",
      "\tTime and date: Tue Mar 27 18:59:35 2018\n",
      "\tPlatform: Darwin-15.6.0-x86_64-i386-64bit\n",
      "\tPython version: 2.7.13 |Anaconda 4.4.0 (x86_64)| (default, Dec 20 2016, 23:05:08)  [GCC 4.2.1 Compatible Apple LLVM 6.0 (clang-600.0.57)]\n",
      "\tdms_tools version: 1.2.2\n",
      "\tBio version: 1.70\n",
      "\tnumpy version: 1.12.1\n",
      "\tscipy version: 0.19.0\n",
      "\tmatplotlib version: 2.0.2\n",
      "\tcython version: 0.25.2\n",
      "\tpystan version: 2.16.0.0\n",
      "\tweblogolib version: 3.5.0\n",
      "\tPyPDF2 version: 1.26.0\n",
      "\tpandas version: 0.20.1\n",
      "\n",
      "Parsed the following arguments:\n",
      "\trestrictdiffsel = None\n",
      "\toverlay3 = None\n",
      "\tcolormap = jet\n",
      "\tnperline = 100\n",
      "\tnumberevery = 10\n",
      "\texcludestop = False\n",
      "\tlogoplot = HSF1i39vsDMSO39-rep2_diffsel_logoplot.pdf\n",
      "\toverlay_cmap = None\n",
      "\tmapmetric = charge\n",
      "\tdiffprefheight = 1.0\n",
      "\tnosepline = False\n",
      "\toverlay1 = None\n",
      "\toverlay2 = None\n",
      "\tdiffselheight = ['DMSO39vsDMSO37-rep1_mutdiffsel.txt', 'DMSO39vsDMSO37-rep2_mutdiffsel.txt', 'DMSO39vsDMSO37-rep3_mutdiffsel.txt', 'HSF1i37vsDMSO37-rep1_mutdiffsel.txt', 'HSF1i37vsDMSO37-rep2_mutdiffsel.txt', 'HSF1i37vsDMSO37-rep3_mutdiffsel.txt', 'HSF1i39vsDMSO37-rep1_mutdiffsel.txt', 'HSF1i39vsDMSO37-rep2_mutdiffsel.txt', 'HSF1i39vsDMSO37-rep3_mutdiffsel.txt', 'HSF1i39vsDMSO39-rep1_mutdiffsel.txt', 'HSF1i39vsDMSO39-rep2_mutdiffsel.txt', 'HSF1i39vsDMSO39-rep3_mutdiffsel.txt', 'HSF1i39vsHSF1i37-rep1_mutdiffsel.txt', 'HSF1i39vsHSF1i37-rep2_mutdiffsel.txt', 'HSF1i39vsHSF1i37-rep3_mutdiffsel.txt', 'Hsp90i37vsDMSO37-rep1_mutdiffsel.txt', 'Hsp90i37vsDMSO37-rep2_mutdiffsel.txt', 'Hsp90i39vsDMSO37-rep3_mutdiffsel.txt', 'Hsp90i39vsDMSO37-rep1_mutdiffsel.txt', 'Hsp90i39vsDMSO37-rep2_mutdiffsel.txt', 'Hsp90i39vsDMSO37-rep3_mutdiffsel.txt']\n",
      "\tstringencyparameter = None\n",
      "\tinfile = HSF1i39vsDMSO39-rep2_mutdiffsel.txt\n",
      "\tletterheight = 2.0\n",
      "\n",
      "\n",
      "Logo plot will be written to HSF1i39vsDMSO39-rep2_diffsel_logoplot.pdf\n",
      "Removing existing file HSF1i39vsDMSO39-rep2_diffsel_logoplot.pdf\n",
      "\n",
      "Reading data from HSF1i39vsDMSO39-rep2_mutdiffsel.txt\n",
      "\n",
      "Now making the logo plot in file HSF1i39vsDMSO39-rep2_diffsel_logoplot.pdf\n",
      "\n",
      "Successfully completed dms_logoplot at Tue Mar 27 18:59:50 2018\n"
     ]
    }
   ],
   "source": [
    "#rep2 HSF1i 39C vs DMSO 39C\n",
    "!dms_logoplot HSF1i39vsDMSO39-rep2_mutdiffsel.txt HSF1i39vsDMSO39-rep2_diffsel_logoplot.pdf --nperline 100 --mapmetric charge --letterheight 2 --diffselheight DMSO39vsDMSO37-rep1_mutdiffsel.txt DMSO39vsDMSO37-rep2_mutdiffsel.txt DMSO39vsDMSO37-rep3_mutdiffsel.txt HSF1i37vsDMSO37-rep1_mutdiffsel.txt HSF1i37vsDMSO37-rep2_mutdiffsel.txt HSF1i37vsDMSO37-rep3_mutdiffsel.txt HSF1i39vsDMSO37-rep1_mutdiffsel.txt HSF1i39vsDMSO37-rep2_mutdiffsel.txt HSF1i39vsDMSO37-rep3_mutdiffsel.txt HSF1i39vsDMSO39-rep1_mutdiffsel.txt HSF1i39vsDMSO39-rep2_mutdiffsel.txt HSF1i39vsDMSO39-rep3_mutdiffsel.txt HSF1i39vsHSF1i37-rep1_mutdiffsel.txt HSF1i39vsHSF1i37-rep2_mutdiffsel.txt HSF1i39vsHSF1i37-rep3_mutdiffsel.txt Hsp90i37vsDMSO37-rep1_mutdiffsel.txt Hsp90i37vsDMSO37-rep2_mutdiffsel.txt Hsp90i39vsDMSO37-rep3_mutdiffsel.txt Hsp90i39vsDMSO37-rep1_mutdiffsel.txt Hsp90i39vsDMSO37-rep2_mutdiffsel.txt Hsp90i39vsDMSO37-rep3_mutdiffsel.txt  "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 42,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Beginning execution of dms_logoplot in directory /Users/annapon/Documents/NP_DMS_2018/2018_NP_DMS/preferences\n",
      "\n",
      "Version information for dms_tools and associated packages.\n",
      "\tTime and date: Tue Mar 27 19:00:04 2018\n",
      "\tPlatform: Darwin-15.6.0-x86_64-i386-64bit\n",
      "\tPython version: 2.7.13 |Anaconda 4.4.0 (x86_64)| (default, Dec 20 2016, 23:05:08)  [GCC 4.2.1 Compatible Apple LLVM 6.0 (clang-600.0.57)]\n",
      "\tdms_tools version: 1.2.2\n",
      "\tBio version: 1.70\n",
      "\tnumpy version: 1.12.1\n",
      "\tscipy version: 0.19.0\n",
      "\tmatplotlib version: 2.0.2\n",
      "\tcython version: 0.25.2\n",
      "\tpystan version: 2.16.0.0\n",
      "\tweblogolib version: 3.5.0\n",
      "\tPyPDF2 version: 1.26.0\n",
      "\tpandas version: 0.20.1\n",
      "\n",
      "Parsed the following arguments:\n",
      "\trestrictdiffsel = None\n",
      "\toverlay3 = None\n",
      "\tcolormap = jet\n",
      "\tnperline = 100\n",
      "\tnumberevery = 10\n",
      "\texcludestop = False\n",
      "\tlogoplot = HSF1i39vsDMSO39-rep3_diffsel_logoplot.pdf\n",
      "\toverlay_cmap = None\n",
      "\tmapmetric = charge\n",
      "\tdiffprefheight = 1.0\n",
      "\tnosepline = False\n",
      "\toverlay1 = None\n",
      "\toverlay2 = None\n",
      "\tdiffselheight = ['DMSO39vsDMSO37-rep1_mutdiffsel.txt', 'DMSO39vsDMSO37-rep2_mutdiffsel.txt', 'DMSO39vsDMSO37-rep3_mutdiffsel.txt', 'HSF1i37vsDMSO37-rep1_mutdiffsel.txt', 'HSF1i37vsDMSO37-rep2_mutdiffsel.txt', 'HSF1i37vsDMSO37-rep3_mutdiffsel.txt', 'HSF1i39vsDMSO37-rep1_mutdiffsel.txt', 'HSF1i39vsDMSO37-rep2_mutdiffsel.txt', 'HSF1i39vsDMSO37-rep3_mutdiffsel.txt', 'HSF1i39vsDMSO39-rep1_mutdiffsel.txt', 'HSF1i39vsDMSO39-rep2_mutdiffsel.txt', 'HSF1i39vsDMSO39-rep3_mutdiffsel.txt', 'HSF1i39vsHSF1i37-rep1_mutdiffsel.txt', 'HSF1i39vsHSF1i37-rep2_mutdiffsel.txt', 'HSF1i39vsHSF1i37-rep3_mutdiffsel.txt', 'Hsp90i37vsDMSO37-rep1_mutdiffsel.txt', 'Hsp90i37vsDMSO37-rep2_mutdiffsel.txt', 'Hsp90i39vsDMSO37-rep3_mutdiffsel.txt', 'Hsp90i39vsDMSO37-rep1_mutdiffsel.txt', 'Hsp90i39vsDMSO37-rep2_mutdiffsel.txt', 'Hsp90i39vsDMSO37-rep3_mutdiffsel.txt']\n",
      "\tstringencyparameter = None\n",
      "\tinfile = HSF1i39vsDMSO39-rep3_mutdiffsel.txt\n",
      "\tletterheight = 2.0\n",
      "\n",
      "\n",
      "Logo plot will be written to HSF1i39vsDMSO39-rep3_diffsel_logoplot.pdf\n",
      "Removing existing file HSF1i39vsDMSO39-rep3_diffsel_logoplot.pdf\n",
      "\n",
      "Reading data from HSF1i39vsDMSO39-rep3_mutdiffsel.txt\n",
      "\n",
      "Now making the logo plot in file HSF1i39vsDMSO39-rep3_diffsel_logoplot.pdf\n",
      "\n",
      "Successfully completed dms_logoplot at Tue Mar 27 19:00:19 2018\n"
     ]
    }
   ],
   "source": [
    "#rep3 HSF1i 39C vs DMSO 39C\n",
    "!dms_logoplot HSF1i39vsDMSO39-rep3_mutdiffsel.txt HSF1i39vsDMSO39-rep3_diffsel_logoplot.pdf --nperline 100 --mapmetric charge --letterheight 2 --diffselheight DMSO39vsDMSO37-rep1_mutdiffsel.txt DMSO39vsDMSO37-rep2_mutdiffsel.txt DMSO39vsDMSO37-rep3_mutdiffsel.txt HSF1i37vsDMSO37-rep1_mutdiffsel.txt HSF1i37vsDMSO37-rep2_mutdiffsel.txt HSF1i37vsDMSO37-rep3_mutdiffsel.txt HSF1i39vsDMSO37-rep1_mutdiffsel.txt HSF1i39vsDMSO37-rep2_mutdiffsel.txt HSF1i39vsDMSO37-rep3_mutdiffsel.txt HSF1i39vsDMSO39-rep1_mutdiffsel.txt HSF1i39vsDMSO39-rep2_mutdiffsel.txt HSF1i39vsDMSO39-rep3_mutdiffsel.txt HSF1i39vsHSF1i37-rep1_mutdiffsel.txt HSF1i39vsHSF1i37-rep2_mutdiffsel.txt HSF1i39vsHSF1i37-rep3_mutdiffsel.txt Hsp90i37vsDMSO37-rep1_mutdiffsel.txt Hsp90i37vsDMSO37-rep2_mutdiffsel.txt Hsp90i39vsDMSO37-rep3_mutdiffsel.txt Hsp90i39vsDMSO37-rep1_mutdiffsel.txt Hsp90i39vsDMSO37-rep2_mutdiffsel.txt Hsp90i39vsDMSO37-rep3_mutdiffsel.txt  "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 43,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Beginning execution of dms_logoplot in directory /Users/annapon/Documents/NP_DMS_2018/2018_NP_DMS/preferences\n",
      "\n",
      "Version information for dms_tools and associated packages.\n",
      "\tTime and date: Tue Mar 27 19:00:21 2018\n",
      "\tPlatform: Darwin-15.6.0-x86_64-i386-64bit\n",
      "\tPython version: 2.7.13 |Anaconda 4.4.0 (x86_64)| (default, Dec 20 2016, 23:05:08)  [GCC 4.2.1 Compatible Apple LLVM 6.0 (clang-600.0.57)]\n",
      "\tdms_tools version: 1.2.2\n",
      "\tBio version: 1.70\n",
      "\tnumpy version: 1.12.1\n",
      "\tscipy version: 0.19.0\n",
      "\tmatplotlib version: 2.0.2\n",
      "\tcython version: 0.25.2\n",
      "\tpystan version: 2.16.0.0\n",
      "\tweblogolib version: 3.5.0\n",
      "\tPyPDF2 version: 1.26.0\n",
      "\tpandas version: 0.20.1\n",
      "\n",
      "Parsed the following arguments:\n",
      "\trestrictdiffsel = None\n",
      "\toverlay3 = None\n",
      "\tcolormap = jet\n",
      "\tnperline = 100\n",
      "\tnumberevery = 10\n",
      "\texcludestop = False\n",
      "\tlogoplot = HSF1i39vsHSF1i37-rep1_diffsel_logoplot.pdf\n",
      "\toverlay_cmap = None\n",
      "\tmapmetric = charge\n",
      "\tdiffprefheight = 1.0\n",
      "\tnosepline = False\n",
      "\toverlay1 = None\n",
      "\toverlay2 = None\n",
      "\tdiffselheight = ['DMSO39vsDMSO37-rep1_mutdiffsel.txt', 'DMSO39vsDMSO37-rep2_mutdiffsel.txt', 'DMSO39vsDMSO37-rep3_mutdiffsel.txt', 'HSF1i37vsDMSO37-rep1_mutdiffsel.txt', 'HSF1i37vsDMSO37-rep2_mutdiffsel.txt', 'HSF1i37vsDMSO37-rep3_mutdiffsel.txt', 'HSF1i39vsDMSO37-rep1_mutdiffsel.txt', 'HSF1i39vsDMSO37-rep2_mutdiffsel.txt', 'HSF1i39vsDMSO37-rep3_mutdiffsel.txt', 'HSF1i39vsDMSO39-rep1_mutdiffsel.txt', 'HSF1i39vsDMSO39-rep2_mutdiffsel.txt', 'HSF1i39vsDMSO39-rep3_mutdiffsel.txt', 'HSF1i39vsHSF1i37-rep1_mutdiffsel.txt', 'HSF1i39vsHSF1i37-rep2_mutdiffsel.txt', 'HSF1i39vsHSF1i37-rep3_mutdiffsel.txt', 'Hsp90i37vsDMSO37-rep1_mutdiffsel.txt', 'Hsp90i37vsDMSO37-rep2_mutdiffsel.txt', 'Hsp90i39vsDMSO37-rep3_mutdiffsel.txt', 'Hsp90i39vsDMSO37-rep1_mutdiffsel.txt', 'Hsp90i39vsDMSO37-rep2_mutdiffsel.txt', 'Hsp90i39vsDMSO37-rep3_mutdiffsel.txt']\n",
      "\tstringencyparameter = None\n",
      "\tinfile = HSF1i39vsHSF1i37-rep1_mutdiffsel.txt\n",
      "\tletterheight = 2.0\n",
      "\n",
      "\n",
      "Logo plot will be written to HSF1i39vsHSF1i37-rep1_diffsel_logoplot.pdf\n",
      "Removing existing file HSF1i39vsHSF1i37-rep1_diffsel_logoplot.pdf\n",
      "\n",
      "Reading data from HSF1i39vsHSF1i37-rep1_mutdiffsel.txt\n",
      "\n",
      "Now making the logo plot in file HSF1i39vsHSF1i37-rep1_diffsel_logoplot.pdf\n",
      "\n",
      "Successfully completed dms_logoplot at Tue Mar 27 19:00:37 2018\n"
     ]
    }
   ],
   "source": [
    "#rep1 HSF1i 39C vs HSF1i 37C\n",
    "!dms_logoplot HSF1i39vsHSF1i37-rep1_mutdiffsel.txt HSF1i39vsHSF1i37-rep1_diffsel_logoplot.pdf --nperline 100 --mapmetric charge --letterheight 2 --diffselheight DMSO39vsDMSO37-rep1_mutdiffsel.txt DMSO39vsDMSO37-rep2_mutdiffsel.txt DMSO39vsDMSO37-rep3_mutdiffsel.txt HSF1i37vsDMSO37-rep1_mutdiffsel.txt HSF1i37vsDMSO37-rep2_mutdiffsel.txt HSF1i37vsDMSO37-rep3_mutdiffsel.txt HSF1i39vsDMSO37-rep1_mutdiffsel.txt HSF1i39vsDMSO37-rep2_mutdiffsel.txt HSF1i39vsDMSO37-rep3_mutdiffsel.txt HSF1i39vsDMSO39-rep1_mutdiffsel.txt HSF1i39vsDMSO39-rep2_mutdiffsel.txt HSF1i39vsDMSO39-rep3_mutdiffsel.txt HSF1i39vsHSF1i37-rep1_mutdiffsel.txt HSF1i39vsHSF1i37-rep2_mutdiffsel.txt HSF1i39vsHSF1i37-rep3_mutdiffsel.txt Hsp90i37vsDMSO37-rep1_mutdiffsel.txt Hsp90i37vsDMSO37-rep2_mutdiffsel.txt Hsp90i39vsDMSO37-rep3_mutdiffsel.txt Hsp90i39vsDMSO37-rep1_mutdiffsel.txt Hsp90i39vsDMSO37-rep2_mutdiffsel.txt Hsp90i39vsDMSO37-rep3_mutdiffsel.txt  "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 44,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Beginning execution of dms_logoplot in directory /Users/annapon/Documents/NP_DMS_2018/2018_NP_DMS/preferences\n",
      "\n",
      "Version information for dms_tools and associated packages.\n",
      "\tTime and date: Tue Mar 27 19:00:39 2018\n",
      "\tPlatform: Darwin-15.6.0-x86_64-i386-64bit\n",
      "\tPython version: 2.7.13 |Anaconda 4.4.0 (x86_64)| (default, Dec 20 2016, 23:05:08)  [GCC 4.2.1 Compatible Apple LLVM 6.0 (clang-600.0.57)]\n",
      "\tdms_tools version: 1.2.2\n",
      "\tBio version: 1.70\n",
      "\tnumpy version: 1.12.1\n",
      "\tscipy version: 0.19.0\n",
      "\tmatplotlib version: 2.0.2\n",
      "\tcython version: 0.25.2\n",
      "\tpystan version: 2.16.0.0\n",
      "\tweblogolib version: 3.5.0\n",
      "\tPyPDF2 version: 1.26.0\n",
      "\tpandas version: 0.20.1\n",
      "\n",
      "Parsed the following arguments:\n",
      "\trestrictdiffsel = None\n",
      "\toverlay3 = None\n",
      "\tcolormap = jet\n",
      "\tnperline = 100\n",
      "\tnumberevery = 10\n",
      "\texcludestop = False\n",
      "\tlogoplot = HSF1i39vsHSF1i37-rep2_diffsel_logoplot.pdf\n",
      "\toverlay_cmap = None\n",
      "\tmapmetric = charge\n",
      "\tdiffprefheight = 1.0\n",
      "\tnosepline = False\n",
      "\toverlay1 = None\n",
      "\toverlay2 = None\n",
      "\tdiffselheight = ['DMSO39vsDMSO37-rep1_mutdiffsel.txt', 'DMSO39vsDMSO37-rep2_mutdiffsel.txt', 'DMSO39vsDMSO37-rep3_mutdiffsel.txt', 'HSF1i37vsDMSO37-rep1_mutdiffsel.txt', 'HSF1i37vsDMSO37-rep2_mutdiffsel.txt', 'HSF1i37vsDMSO37-rep3_mutdiffsel.txt', 'HSF1i39vsDMSO37-rep1_mutdiffsel.txt', 'HSF1i39vsDMSO37-rep2_mutdiffsel.txt', 'HSF1i39vsDMSO37-rep3_mutdiffsel.txt', 'HSF1i39vsDMSO39-rep1_mutdiffsel.txt', 'HSF1i39vsDMSO39-rep2_mutdiffsel.txt', 'HSF1i39vsDMSO39-rep3_mutdiffsel.txt', 'HSF1i39vsHSF1i37-rep1_mutdiffsel.txt', 'HSF1i39vsHSF1i37-rep2_mutdiffsel.txt', 'HSF1i39vsHSF1i37-rep3_mutdiffsel.txt', 'Hsp90i37vsDMSO37-rep1_mutdiffsel.txt', 'Hsp90i37vsDMSO37-rep2_mutdiffsel.txt', 'Hsp90i39vsDMSO37-rep3_mutdiffsel.txt', 'Hsp90i39vsDMSO37-rep1_mutdiffsel.txt', 'Hsp90i39vsDMSO37-rep2_mutdiffsel.txt', 'Hsp90i39vsDMSO37-rep3_mutdiffsel.txt']\n",
      "\tstringencyparameter = None\n",
      "\tinfile = HSF1i39vsHSF1i37-rep2_mutdiffsel.txt\n",
      "\tletterheight = 2.0\n",
      "\n",
      "\n",
      "Logo plot will be written to HSF1i39vsHSF1i37-rep2_diffsel_logoplot.pdf\n",
      "Removing existing file HSF1i39vsHSF1i37-rep2_diffsel_logoplot.pdf\n",
      "\n",
      "Reading data from HSF1i39vsHSF1i37-rep2_mutdiffsel.txt\n",
      "\n",
      "Now making the logo plot in file HSF1i39vsHSF1i37-rep2_diffsel_logoplot.pdf\n",
      "\n",
      "Successfully completed dms_logoplot at Tue Mar 27 19:00:54 2018\n"
     ]
    }
   ],
   "source": [
    "#rep2 HSF1i 39C vs HSF1i 37C\n",
    "!dms_logoplot HSF1i39vsHSF1i37-rep2_mutdiffsel.txt HSF1i39vsHSF1i37-rep2_diffsel_logoplot.pdf --nperline 100 --mapmetric charge --letterheight 2 --diffselheight DMSO39vsDMSO37-rep1_mutdiffsel.txt DMSO39vsDMSO37-rep2_mutdiffsel.txt DMSO39vsDMSO37-rep3_mutdiffsel.txt HSF1i37vsDMSO37-rep1_mutdiffsel.txt HSF1i37vsDMSO37-rep2_mutdiffsel.txt HSF1i37vsDMSO37-rep3_mutdiffsel.txt HSF1i39vsDMSO37-rep1_mutdiffsel.txt HSF1i39vsDMSO37-rep2_mutdiffsel.txt HSF1i39vsDMSO37-rep3_mutdiffsel.txt HSF1i39vsDMSO39-rep1_mutdiffsel.txt HSF1i39vsDMSO39-rep2_mutdiffsel.txt HSF1i39vsDMSO39-rep3_mutdiffsel.txt HSF1i39vsHSF1i37-rep1_mutdiffsel.txt HSF1i39vsHSF1i37-rep2_mutdiffsel.txt HSF1i39vsHSF1i37-rep3_mutdiffsel.txt Hsp90i37vsDMSO37-rep1_mutdiffsel.txt Hsp90i37vsDMSO37-rep2_mutdiffsel.txt Hsp90i39vsDMSO37-rep3_mutdiffsel.txt Hsp90i39vsDMSO37-rep1_mutdiffsel.txt Hsp90i39vsDMSO37-rep2_mutdiffsel.txt Hsp90i39vsDMSO37-rep3_mutdiffsel.txt  "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 45,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Beginning execution of dms_logoplot in directory /Users/annapon/Documents/NP_DMS_2018/2018_NP_DMS/preferences\n",
      "\n",
      "Version information for dms_tools and associated packages.\n",
      "\tTime and date: Tue Mar 27 19:00:57 2018\n",
      "\tPlatform: Darwin-15.6.0-x86_64-i386-64bit\n",
      "\tPython version: 2.7.13 |Anaconda 4.4.0 (x86_64)| (default, Dec 20 2016, 23:05:08)  [GCC 4.2.1 Compatible Apple LLVM 6.0 (clang-600.0.57)]\n",
      "\tdms_tools version: 1.2.2\n",
      "\tBio version: 1.70\n",
      "\tnumpy version: 1.12.1\n",
      "\tscipy version: 0.19.0\n",
      "\tmatplotlib version: 2.0.2\n",
      "\tcython version: 0.25.2\n",
      "\tpystan version: 2.16.0.0\n",
      "\tweblogolib version: 3.5.0\n",
      "\tPyPDF2 version: 1.26.0\n",
      "\tpandas version: 0.20.1\n",
      "\n",
      "Parsed the following arguments:\n",
      "\trestrictdiffsel = None\n",
      "\toverlay3 = None\n",
      "\tcolormap = jet\n",
      "\tnperline = 100\n",
      "\tnumberevery = 10\n",
      "\texcludestop = False\n",
      "\tlogoplot = HSF1i39vsHSF1i37-rep3_diffsel_logoplot.pdf\n",
      "\toverlay_cmap = None\n",
      "\tmapmetric = charge\n",
      "\tdiffprefheight = 1.0\n",
      "\tnosepline = False\n",
      "\toverlay1 = None\n",
      "\toverlay2 = None\n",
      "\tdiffselheight = ['DMSO39vsDMSO37-rep1_mutdiffsel.txt', 'DMSO39vsDMSO37-rep2_mutdiffsel.txt', 'DMSO39vsDMSO37-rep3_mutdiffsel.txt', 'HSF1i37vsDMSO37-rep1_mutdiffsel.txt', 'HSF1i37vsDMSO37-rep2_mutdiffsel.txt', 'HSF1i37vsDMSO37-rep3_mutdiffsel.txt', 'HSF1i39vsDMSO37-rep1_mutdiffsel.txt', 'HSF1i39vsDMSO37-rep2_mutdiffsel.txt', 'HSF1i39vsDMSO37-rep3_mutdiffsel.txt', 'HSF1i39vsDMSO39-rep1_mutdiffsel.txt', 'HSF1i39vsDMSO39-rep2_mutdiffsel.txt', 'HSF1i39vsDMSO39-rep3_mutdiffsel.txt', 'HSF1i39vsHSF1i37-rep1_mutdiffsel.txt', 'HSF1i39vsHSF1i37-rep2_mutdiffsel.txt', 'HSF1i39vsHSF1i37-rep3_mutdiffsel.txt', 'Hsp90i37vsDMSO37-rep1_mutdiffsel.txt', 'Hsp90i37vsDMSO37-rep2_mutdiffsel.txt', 'Hsp90i39vsDMSO37-rep3_mutdiffsel.txt', 'Hsp90i39vsDMSO37-rep1_mutdiffsel.txt', 'Hsp90i39vsDMSO37-rep2_mutdiffsel.txt', 'Hsp90i39vsDMSO37-rep3_mutdiffsel.txt']\n",
      "\tstringencyparameter = None\n",
      "\tinfile = HSF1i39vsHSF1i37-rep3_mutdiffsel.txt\n",
      "\tletterheight = 2.0\n",
      "\n",
      "\n",
      "Logo plot will be written to HSF1i39vsHSF1i37-rep3_diffsel_logoplot.pdf\n",
      "Removing existing file HSF1i39vsHSF1i37-rep3_diffsel_logoplot.pdf\n",
      "\n",
      "Reading data from HSF1i39vsHSF1i37-rep3_mutdiffsel.txt\n",
      "\n",
      "Now making the logo plot in file HSF1i39vsHSF1i37-rep3_diffsel_logoplot.pdf\n",
      "\n",
      "Successfully completed dms_logoplot at Tue Mar 27 19:01:13 2018\n"
     ]
    }
   ],
   "source": [
    "#rep3 HSF1i 39C vs HSF1i 37C\n",
    "!dms_logoplot HSF1i39vsHSF1i37-rep3_mutdiffsel.txt HSF1i39vsHSF1i37-rep3_diffsel_logoplot.pdf --nperline 100 --mapmetric charge --letterheight 2 --diffselheight DMSO39vsDMSO37-rep1_mutdiffsel.txt DMSO39vsDMSO37-rep2_mutdiffsel.txt DMSO39vsDMSO37-rep3_mutdiffsel.txt HSF1i37vsDMSO37-rep1_mutdiffsel.txt HSF1i37vsDMSO37-rep2_mutdiffsel.txt HSF1i37vsDMSO37-rep3_mutdiffsel.txt HSF1i39vsDMSO37-rep1_mutdiffsel.txt HSF1i39vsDMSO37-rep2_mutdiffsel.txt HSF1i39vsDMSO37-rep3_mutdiffsel.txt HSF1i39vsDMSO39-rep1_mutdiffsel.txt HSF1i39vsDMSO39-rep2_mutdiffsel.txt HSF1i39vsDMSO39-rep3_mutdiffsel.txt HSF1i39vsHSF1i37-rep1_mutdiffsel.txt HSF1i39vsHSF1i37-rep2_mutdiffsel.txt HSF1i39vsHSF1i37-rep3_mutdiffsel.txt Hsp90i37vsDMSO37-rep1_mutdiffsel.txt Hsp90i37vsDMSO37-rep2_mutdiffsel.txt Hsp90i39vsDMSO37-rep3_mutdiffsel.txt Hsp90i39vsDMSO37-rep1_mutdiffsel.txt Hsp90i39vsDMSO37-rep2_mutdiffsel.txt Hsp90i39vsDMSO37-rep3_mutdiffsel.txt  "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 46,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Beginning execution of dms_logoplot in directory /Users/annapon/Documents/NP_DMS_2018/2018_NP_DMS/preferences\n",
      "\n",
      "Version information for dms_tools and associated packages.\n",
      "\tTime and date: Tue Mar 27 19:01:15 2018\n",
      "\tPlatform: Darwin-15.6.0-x86_64-i386-64bit\n",
      "\tPython version: 2.7.13 |Anaconda 4.4.0 (x86_64)| (default, Dec 20 2016, 23:05:08)  [GCC 4.2.1 Compatible Apple LLVM 6.0 (clang-600.0.57)]\n",
      "\tdms_tools version: 1.2.2\n",
      "\tBio version: 1.70\n",
      "\tnumpy version: 1.12.1\n",
      "\tscipy version: 0.19.0\n",
      "\tmatplotlib version: 2.0.2\n",
      "\tcython version: 0.25.2\n",
      "\tpystan version: 2.16.0.0\n",
      "\tweblogolib version: 3.5.0\n",
      "\tPyPDF2 version: 1.26.0\n",
      "\tpandas version: 0.20.1\n",
      "\n",
      "Parsed the following arguments:\n",
      "\trestrictdiffsel = None\n",
      "\toverlay3 = None\n",
      "\tcolormap = jet\n",
      "\tnperline = 100\n",
      "\tnumberevery = 10\n",
      "\texcludestop = False\n",
      "\tlogoplot = Hsp90i37vsDMSO37-rep1_diffsel_logoplot.pdf\n",
      "\toverlay_cmap = None\n",
      "\tmapmetric = charge\n",
      "\tdiffprefheight = 1.0\n",
      "\tnosepline = False\n",
      "\toverlay1 = None\n",
      "\toverlay2 = None\n",
      "\tdiffselheight = ['DMSO39vsDMSO37-rep1_mutdiffsel.txt', 'DMSO39vsDMSO37-rep2_mutdiffsel.txt', 'DMSO39vsDMSO37-rep3_mutdiffsel.txt', 'HSF1i37vsDMSO37-rep1_mutdiffsel.txt', 'HSF1i37vsDMSO37-rep2_mutdiffsel.txt', 'HSF1i37vsDMSO37-rep3_mutdiffsel.txt', 'HSF1i39vsDMSO37-rep1_mutdiffsel.txt', 'HSF1i39vsDMSO37-rep2_mutdiffsel.txt', 'HSF1i39vsDMSO37-rep3_mutdiffsel.txt', 'HSF1i39vsDMSO39-rep1_mutdiffsel.txt', 'HSF1i39vsDMSO39-rep2_mutdiffsel.txt', 'HSF1i39vsDMSO39-rep3_mutdiffsel.txt', 'HSF1i39vsHSF1i37-rep1_mutdiffsel.txt', 'HSF1i39vsHSF1i37-rep2_mutdiffsel.txt', 'HSF1i39vsHSF1i37-rep3_mutdiffsel.txt', 'Hsp90i37vsDMSO37-rep1_mutdiffsel.txt', 'Hsp90i37vsDMSO37-rep2_mutdiffsel.txt', 'Hsp90i39vsDMSO37-rep3_mutdiffsel.txt', 'Hsp90i39vsDMSO37-rep1_mutdiffsel.txt', 'Hsp90i39vsDMSO37-rep2_mutdiffsel.txt', 'Hsp90i39vsDMSO37-rep3_mutdiffsel.txt']\n",
      "\tstringencyparameter = None\n",
      "\tinfile = Hsp90i37vsDMSO37-rep1_mutdiffsel.txt\n",
      "\tletterheight = 2.0\n",
      "\n",
      "\n",
      "Logo plot will be written to Hsp90i37vsDMSO37-rep1_diffsel_logoplot.pdf\n",
      "Removing existing file Hsp90i37vsDMSO37-rep1_diffsel_logoplot.pdf\n",
      "\n",
      "Reading data from Hsp90i37vsDMSO37-rep1_mutdiffsel.txt\n",
      "\n",
      "Now making the logo plot in file Hsp90i37vsDMSO37-rep1_diffsel_logoplot.pdf\n",
      "\n",
      "Successfully completed dms_logoplot at Tue Mar 27 19:01:31 2018\n"
     ]
    }
   ],
   "source": [
    "#rep1 Hsp90i 37C vs DMSO 37C\n",
    "!dms_logoplot Hsp90i37vsDMSO37-rep1_mutdiffsel.txt Hsp90i37vsDMSO37-rep1_diffsel_logoplot.pdf --nperline 100 --mapmetric charge --letterheight 2 --diffselheight DMSO39vsDMSO37-rep1_mutdiffsel.txt DMSO39vsDMSO37-rep2_mutdiffsel.txt DMSO39vsDMSO37-rep3_mutdiffsel.txt HSF1i37vsDMSO37-rep1_mutdiffsel.txt HSF1i37vsDMSO37-rep2_mutdiffsel.txt HSF1i37vsDMSO37-rep3_mutdiffsel.txt HSF1i39vsDMSO37-rep1_mutdiffsel.txt HSF1i39vsDMSO37-rep2_mutdiffsel.txt HSF1i39vsDMSO37-rep3_mutdiffsel.txt HSF1i39vsDMSO39-rep1_mutdiffsel.txt HSF1i39vsDMSO39-rep2_mutdiffsel.txt HSF1i39vsDMSO39-rep3_mutdiffsel.txt HSF1i39vsHSF1i37-rep1_mutdiffsel.txt HSF1i39vsHSF1i37-rep2_mutdiffsel.txt HSF1i39vsHSF1i37-rep3_mutdiffsel.txt Hsp90i37vsDMSO37-rep1_mutdiffsel.txt Hsp90i37vsDMSO37-rep2_mutdiffsel.txt Hsp90i39vsDMSO37-rep3_mutdiffsel.txt Hsp90i39vsDMSO37-rep1_mutdiffsel.txt Hsp90i39vsDMSO37-rep2_mutdiffsel.txt Hsp90i39vsDMSO37-rep3_mutdiffsel.txt  "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 47,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Beginning execution of dms_logoplot in directory /Users/annapon/Documents/NP_DMS_2018/2018_NP_DMS/preferences\n",
      "\n",
      "Version information for dms_tools and associated packages.\n",
      "\tTime and date: Tue Mar 27 19:01:33 2018\n",
      "\tPlatform: Darwin-15.6.0-x86_64-i386-64bit\n",
      "\tPython version: 2.7.13 |Anaconda 4.4.0 (x86_64)| (default, Dec 20 2016, 23:05:08)  [GCC 4.2.1 Compatible Apple LLVM 6.0 (clang-600.0.57)]\n",
      "\tdms_tools version: 1.2.2\n",
      "\tBio version: 1.70\n",
      "\tnumpy version: 1.12.1\n",
      "\tscipy version: 0.19.0\n",
      "\tmatplotlib version: 2.0.2\n",
      "\tcython version: 0.25.2\n",
      "\tpystan version: 2.16.0.0\n",
      "\tweblogolib version: 3.5.0\n",
      "\tPyPDF2 version: 1.26.0\n",
      "\tpandas version: 0.20.1\n",
      "\n",
      "Parsed the following arguments:\n",
      "\trestrictdiffsel = None\n",
      "\toverlay3 = None\n",
      "\tcolormap = jet\n",
      "\tnperline = 100\n",
      "\tnumberevery = 10\n",
      "\texcludestop = False\n",
      "\tlogoplot = Hsp90i37vsDMSO37-rep2_diffsel_logoplot.pdf\n",
      "\toverlay_cmap = None\n",
      "\tmapmetric = charge\n",
      "\tdiffprefheight = 1.0\n",
      "\tnosepline = False\n",
      "\toverlay1 = None\n",
      "\toverlay2 = None\n",
      "\tdiffselheight = ['DMSO39vsDMSO37-rep1_mutdiffsel.txt', 'DMSO39vsDMSO37-rep2_mutdiffsel.txt', 'DMSO39vsDMSO37-rep3_mutdiffsel.txt', 'HSF1i37vsDMSO37-rep1_mutdiffsel.txt', 'HSF1i37vsDMSO37-rep2_mutdiffsel.txt', 'HSF1i37vsDMSO37-rep3_mutdiffsel.txt', 'HSF1i39vsDMSO37-rep1_mutdiffsel.txt', 'HSF1i39vsDMSO37-rep2_mutdiffsel.txt', 'HSF1i39vsDMSO37-rep3_mutdiffsel.txt', 'HSF1i39vsDMSO39-rep1_mutdiffsel.txt', 'HSF1i39vsDMSO39-rep2_mutdiffsel.txt', 'HSF1i39vsDMSO39-rep3_mutdiffsel.txt', 'HSF1i39vsHSF1i37-rep1_mutdiffsel.txt', 'HSF1i39vsHSF1i37-rep2_mutdiffsel.txt', 'HSF1i39vsHSF1i37-rep3_mutdiffsel.txt', 'Hsp90i37vsDMSO37-rep1_mutdiffsel.txt', 'Hsp90i37vsDMSO37-rep2_mutdiffsel.txt', 'Hsp90i39vsDMSO37-rep3_mutdiffsel.txt', 'Hsp90i39vsDMSO37-rep1_mutdiffsel.txt', 'Hsp90i39vsDMSO37-rep2_mutdiffsel.txt', 'Hsp90i39vsDMSO37-rep3_mutdiffsel.txt']\n",
      "\tstringencyparameter = None\n",
      "\tinfile = Hsp90i37vsDMSO37-rep2_mutdiffsel.txt\n",
      "\tletterheight = 2.0\n",
      "\n",
      "\n",
      "Logo plot will be written to Hsp90i37vsDMSO37-rep2_diffsel_logoplot.pdf\n",
      "Removing existing file Hsp90i37vsDMSO37-rep2_diffsel_logoplot.pdf\n",
      "\n",
      "Reading data from Hsp90i37vsDMSO37-rep2_mutdiffsel.txt\n",
      "\n",
      "Now making the logo plot in file Hsp90i37vsDMSO37-rep2_diffsel_logoplot.pdf\n",
      "\n",
      "Successfully completed dms_logoplot at Tue Mar 27 19:01:49 2018\n"
     ]
    }
   ],
   "source": [
    "#rep2 Hsp90i 37C vs DMSO 37C\n",
    "!dms_logoplot Hsp90i37vsDMSO37-rep2_mutdiffsel.txt Hsp90i37vsDMSO37-rep2_diffsel_logoplot.pdf --nperline 100 --mapmetric charge --letterheight 2 --diffselheight DMSO39vsDMSO37-rep1_mutdiffsel.txt DMSO39vsDMSO37-rep2_mutdiffsel.txt DMSO39vsDMSO37-rep3_mutdiffsel.txt HSF1i37vsDMSO37-rep1_mutdiffsel.txt HSF1i37vsDMSO37-rep2_mutdiffsel.txt HSF1i37vsDMSO37-rep3_mutdiffsel.txt HSF1i39vsDMSO37-rep1_mutdiffsel.txt HSF1i39vsDMSO37-rep2_mutdiffsel.txt HSF1i39vsDMSO37-rep3_mutdiffsel.txt HSF1i39vsDMSO39-rep1_mutdiffsel.txt HSF1i39vsDMSO39-rep2_mutdiffsel.txt HSF1i39vsDMSO39-rep3_mutdiffsel.txt HSF1i39vsHSF1i37-rep1_mutdiffsel.txt HSF1i39vsHSF1i37-rep2_mutdiffsel.txt HSF1i39vsHSF1i37-rep3_mutdiffsel.txt Hsp90i37vsDMSO37-rep1_mutdiffsel.txt Hsp90i37vsDMSO37-rep2_mutdiffsel.txt Hsp90i39vsDMSO37-rep3_mutdiffsel.txt Hsp90i39vsDMSO37-rep1_mutdiffsel.txt Hsp90i39vsDMSO37-rep2_mutdiffsel.txt Hsp90i39vsDMSO37-rep3_mutdiffsel.txt  "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 50,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Beginning execution of dms_logoplot in directory /Users/annapon/Documents/NP_DMS_2018/2018_NP_DMS/preferences\n",
      "\n",
      "Version information for dms_tools and associated packages.\n",
      "\tTime and date: Tue Mar 27 19:02:46 2018\n",
      "\tPlatform: Darwin-15.6.0-x86_64-i386-64bit\n",
      "\tPython version: 2.7.13 |Anaconda 4.4.0 (x86_64)| (default, Dec 20 2016, 23:05:08)  [GCC 4.2.1 Compatible Apple LLVM 6.0 (clang-600.0.57)]\n",
      "\tdms_tools version: 1.2.2\n",
      "\tBio version: 1.70\n",
      "\tnumpy version: 1.12.1\n",
      "\tscipy version: 0.19.0\n",
      "\tmatplotlib version: 2.0.2\n",
      "\tcython version: 0.25.2\n",
      "\tpystan version: 2.16.0.0\n",
      "\tweblogolib version: 3.5.0\n",
      "\tPyPDF2 version: 1.26.0\n",
      "\tpandas version: 0.20.1\n",
      "\n",
      "Parsed the following arguments:\n",
      "\trestrictdiffsel = None\n",
      "\toverlay3 = None\n",
      "\tcolormap = jet\n",
      "\tnperline = 100\n",
      "\tnumberevery = 10\n",
      "\texcludestop = False\n",
      "\tlogoplot = Hsp90i37vsDMSO37-rep3_diffsel_logoplot.pdf\n",
      "\toverlay_cmap = None\n",
      "\tmapmetric = charge\n",
      "\tdiffprefheight = 1.0\n",
      "\tnosepline = False\n",
      "\toverlay1 = None\n",
      "\toverlay2 = None\n",
      "\tdiffselheight = ['DMSO39vsDMSO37-rep1_mutdiffsel.txt', 'DMSO39vsDMSO37-rep2_mutdiffsel.txt', 'DMSO39vsDMSO37-rep3_mutdiffsel.txt', 'HSF1i37vsDMSO37-rep1_mutdiffsel.txt', 'HSF1i37vsDMSO37-rep2_mutdiffsel.txt', 'HSF1i37vsDMSO37-rep3_mutdiffsel.txt', 'HSF1i39vsDMSO37-rep1_mutdiffsel.txt', 'HSF1i39vsDMSO37-rep2_mutdiffsel.txt', 'HSF1i39vsDMSO37-rep3_mutdiffsel.txt', 'HSF1i39vsDMSO39-rep1_mutdiffsel.txt', 'HSF1i39vsDMSO39-rep2_mutdiffsel.txt', 'HSF1i39vsDMSO39-rep3_mutdiffsel.txt', 'HSF1i39vsHSF1i37-rep1_mutdiffsel.txt', 'HSF1i39vsHSF1i37-rep2_mutdiffsel.txt', 'HSF1i39vsHSF1i37-rep3_mutdiffsel.txt', 'Hsp90i37vsDMSO37-rep1_mutdiffsel.txt', 'Hsp90i37vsDMSO37-rep2_mutdiffsel.txt', 'Hsp90i39vsDMSO37-rep3_mutdiffsel.txt', 'Hsp90i39vsDMSO37-rep1_mutdiffsel.txt', 'Hsp90i39vsDMSO37-rep2_mutdiffsel.txt', 'Hsp90i39vsDMSO37-rep3_mutdiffsel.txt']\n",
      "\tstringencyparameter = None\n",
      "\tinfile = Hsp90i39vsDMSO37-rep3_mutdiffsel.txt\n",
      "\tletterheight = 2.0\n",
      "\n",
      "\n",
      "Logo plot will be written to Hsp90i37vsDMSO37-rep3_diffsel_logoplot.pdf\n",
      "\n",
      "Reading data from Hsp90i39vsDMSO37-rep3_mutdiffsel.txt\n",
      "\n",
      "Now making the logo plot in file Hsp90i37vsDMSO37-rep3_diffsel_logoplot.pdf\n",
      "\n",
      "Successfully completed dms_logoplot at Tue Mar 27 19:03:02 2018\n"
     ]
    }
   ],
   "source": [
    "#rep3 Hsp90i 37C vs DMSO 37C\n",
    "!dms_logoplot Hsp90i39vsDMSO37-rep3_mutdiffsel.txt Hsp90i37vsDMSO37-rep3_diffsel_logoplot.pdf --nperline 100 --mapmetric charge --letterheight 2 --diffselheight DMSO39vsDMSO37-rep1_mutdiffsel.txt DMSO39vsDMSO37-rep2_mutdiffsel.txt DMSO39vsDMSO37-rep3_mutdiffsel.txt HSF1i37vsDMSO37-rep1_mutdiffsel.txt HSF1i37vsDMSO37-rep2_mutdiffsel.txt HSF1i37vsDMSO37-rep3_mutdiffsel.txt HSF1i39vsDMSO37-rep1_mutdiffsel.txt HSF1i39vsDMSO37-rep2_mutdiffsel.txt HSF1i39vsDMSO37-rep3_mutdiffsel.txt HSF1i39vsDMSO39-rep1_mutdiffsel.txt HSF1i39vsDMSO39-rep2_mutdiffsel.txt HSF1i39vsDMSO39-rep3_mutdiffsel.txt HSF1i39vsHSF1i37-rep1_mutdiffsel.txt HSF1i39vsHSF1i37-rep2_mutdiffsel.txt HSF1i39vsHSF1i37-rep3_mutdiffsel.txt Hsp90i37vsDMSO37-rep1_mutdiffsel.txt Hsp90i37vsDMSO37-rep2_mutdiffsel.txt Hsp90i39vsDMSO37-rep3_mutdiffsel.txt Hsp90i39vsDMSO37-rep1_mutdiffsel.txt Hsp90i39vsDMSO37-rep2_mutdiffsel.txt Hsp90i39vsDMSO37-rep3_mutdiffsel.txt  "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 51,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Beginning execution of dms_logoplot in directory /Users/annapon/Documents/NP_DMS_2018/2018_NP_DMS/preferences\n",
      "\n",
      "Version information for dms_tools and associated packages.\n",
      "\tTime and date: Tue Mar 27 19:03:05 2018\n",
      "\tPlatform: Darwin-15.6.0-x86_64-i386-64bit\n",
      "\tPython version: 2.7.13 |Anaconda 4.4.0 (x86_64)| (default, Dec 20 2016, 23:05:08)  [GCC 4.2.1 Compatible Apple LLVM 6.0 (clang-600.0.57)]\n",
      "\tdms_tools version: 1.2.2\n",
      "\tBio version: 1.70\n",
      "\tnumpy version: 1.12.1\n",
      "\tscipy version: 0.19.0\n",
      "\tmatplotlib version: 2.0.2\n",
      "\tcython version: 0.25.2\n",
      "\tpystan version: 2.16.0.0\n",
      "\tweblogolib version: 3.5.0\n",
      "\tPyPDF2 version: 1.26.0\n",
      "\tpandas version: 0.20.1\n",
      "\n",
      "Parsed the following arguments:\n",
      "\trestrictdiffsel = None\n",
      "\toverlay3 = None\n",
      "\tcolormap = jet\n",
      "\tnperline = 100\n",
      "\tnumberevery = 10\n",
      "\texcludestop = False\n",
      "\tlogoplot = Hsp90i39vsDMSO37-rep1_diffsel_logoplot.pdf\n",
      "\toverlay_cmap = None\n",
      "\tmapmetric = charge\n",
      "\tdiffprefheight = 1.0\n",
      "\tnosepline = False\n",
      "\toverlay1 = None\n",
      "\toverlay2 = None\n",
      "\tdiffselheight = ['DMSO39vsDMSO37-rep1_mutdiffsel.txt', 'DMSO39vsDMSO37-rep2_mutdiffsel.txt', 'DMSO39vsDMSO37-rep3_mutdiffsel.txt', 'HSF1i37vsDMSO37-rep1_mutdiffsel.txt', 'HSF1i37vsDMSO37-rep2_mutdiffsel.txt', 'HSF1i37vsDMSO37-rep3_mutdiffsel.txt', 'HSF1i39vsDMSO37-rep1_mutdiffsel.txt', 'HSF1i39vsDMSO37-rep2_mutdiffsel.txt', 'HSF1i39vsDMSO37-rep3_mutdiffsel.txt', 'HSF1i39vsDMSO39-rep1_mutdiffsel.txt', 'HSF1i39vsDMSO39-rep2_mutdiffsel.txt', 'HSF1i39vsDMSO39-rep3_mutdiffsel.txt', 'HSF1i39vsHSF1i37-rep1_mutdiffsel.txt', 'HSF1i39vsHSF1i37-rep2_mutdiffsel.txt', 'HSF1i39vsHSF1i37-rep3_mutdiffsel.txt', 'Hsp90i37vsDMSO37-rep1_mutdiffsel.txt', 'Hsp90i37vsDMSO37-rep2_mutdiffsel.txt', 'Hsp90i39vsDMSO37-rep3_mutdiffsel.txt', 'Hsp90i39vsDMSO37-rep1_mutdiffsel.txt', 'Hsp90i39vsDMSO37-rep2_mutdiffsel.txt', 'Hsp90i39vsDMSO37-rep3_mutdiffsel.txt']\n",
      "\tstringencyparameter = None\n",
      "\tinfile = Hsp90i39vsDMSO37-rep1_mutdiffsel.txt\n",
      "\tletterheight = 2.0\n",
      "\n",
      "\n",
      "Logo plot will be written to Hsp90i39vsDMSO37-rep1_diffsel_logoplot.pdf\n",
      "Removing existing file Hsp90i39vsDMSO37-rep1_diffsel_logoplot.pdf\n",
      "\n",
      "Reading data from Hsp90i39vsDMSO37-rep1_mutdiffsel.txt\n",
      "\n",
      "Now making the logo plot in file Hsp90i39vsDMSO37-rep1_diffsel_logoplot.pdf\n",
      "\n",
      "Successfully completed dms_logoplot at Tue Mar 27 19:03:20 2018\n"
     ]
    }
   ],
   "source": [
    "#rep1 Hsp90i 39C vs DMSO 37C\n",
    "!dms_logoplot Hsp90i39vsDMSO37-rep1_mutdiffsel.txt Hsp90i39vsDMSO37-rep1_diffsel_logoplot.pdf --nperline 100 --mapmetric charge --letterheight 2 --diffselheight DMSO39vsDMSO37-rep1_mutdiffsel.txt DMSO39vsDMSO37-rep2_mutdiffsel.txt DMSO39vsDMSO37-rep3_mutdiffsel.txt HSF1i37vsDMSO37-rep1_mutdiffsel.txt HSF1i37vsDMSO37-rep2_mutdiffsel.txt HSF1i37vsDMSO37-rep3_mutdiffsel.txt HSF1i39vsDMSO37-rep1_mutdiffsel.txt HSF1i39vsDMSO37-rep2_mutdiffsel.txt HSF1i39vsDMSO37-rep3_mutdiffsel.txt HSF1i39vsDMSO39-rep1_mutdiffsel.txt HSF1i39vsDMSO39-rep2_mutdiffsel.txt HSF1i39vsDMSO39-rep3_mutdiffsel.txt HSF1i39vsHSF1i37-rep1_mutdiffsel.txt HSF1i39vsHSF1i37-rep2_mutdiffsel.txt HSF1i39vsHSF1i37-rep3_mutdiffsel.txt Hsp90i37vsDMSO37-rep1_mutdiffsel.txt Hsp90i37vsDMSO37-rep2_mutdiffsel.txt Hsp90i39vsDMSO37-rep3_mutdiffsel.txt Hsp90i39vsDMSO37-rep1_mutdiffsel.txt Hsp90i39vsDMSO37-rep2_mutdiffsel.txt Hsp90i39vsDMSO37-rep3_mutdiffsel.txt  "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 52,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Beginning execution of dms_logoplot in directory /Users/annapon/Documents/NP_DMS_2018/2018_NP_DMS/preferences\n",
      "\n",
      "Version information for dms_tools and associated packages.\n",
      "\tTime and date: Tue Mar 27 19:03:23 2018\n",
      "\tPlatform: Darwin-15.6.0-x86_64-i386-64bit\n",
      "\tPython version: 2.7.13 |Anaconda 4.4.0 (x86_64)| (default, Dec 20 2016, 23:05:08)  [GCC 4.2.1 Compatible Apple LLVM 6.0 (clang-600.0.57)]\n",
      "\tdms_tools version: 1.2.2\n",
      "\tBio version: 1.70\n",
      "\tnumpy version: 1.12.1\n",
      "\tscipy version: 0.19.0\n",
      "\tmatplotlib version: 2.0.2\n",
      "\tcython version: 0.25.2\n",
      "\tpystan version: 2.16.0.0\n",
      "\tweblogolib version: 3.5.0\n",
      "\tPyPDF2 version: 1.26.0\n",
      "\tpandas version: 0.20.1\n",
      "\n",
      "Parsed the following arguments:\n",
      "\trestrictdiffsel = None\n",
      "\toverlay3 = None\n",
      "\tcolormap = jet\n",
      "\tnperline = 100\n",
      "\tnumberevery = 10\n",
      "\texcludestop = False\n",
      "\tlogoplot = Hsp90i39vsDMSO37-rep2_diffsel_logoplot.pdf\n",
      "\toverlay_cmap = None\n",
      "\tmapmetric = charge\n",
      "\tdiffprefheight = 1.0\n",
      "\tnosepline = False\n",
      "\toverlay1 = None\n",
      "\toverlay2 = None\n",
      "\tdiffselheight = ['DMSO39vsDMSO37-rep1_mutdiffsel.txt', 'DMSO39vsDMSO37-rep2_mutdiffsel.txt', 'DMSO39vsDMSO37-rep3_mutdiffsel.txt', 'HSF1i37vsDMSO37-rep1_mutdiffsel.txt', 'HSF1i37vsDMSO37-rep2_mutdiffsel.txt', 'HSF1i37vsDMSO37-rep3_mutdiffsel.txt', 'HSF1i39vsDMSO37-rep1_mutdiffsel.txt', 'HSF1i39vsDMSO37-rep2_mutdiffsel.txt', 'HSF1i39vsDMSO37-rep3_mutdiffsel.txt', 'HSF1i39vsDMSO39-rep1_mutdiffsel.txt', 'HSF1i39vsDMSO39-rep2_mutdiffsel.txt', 'HSF1i39vsDMSO39-rep3_mutdiffsel.txt', 'HSF1i39vsHSF1i37-rep1_mutdiffsel.txt', 'HSF1i39vsHSF1i37-rep2_mutdiffsel.txt', 'HSF1i39vsHSF1i37-rep3_mutdiffsel.txt', 'Hsp90i37vsDMSO37-rep1_mutdiffsel.txt', 'Hsp90i37vsDMSO37-rep2_mutdiffsel.txt', 'Hsp90i39vsDMSO37-rep3_mutdiffsel.txt', 'Hsp90i39vsDMSO37-rep1_mutdiffsel.txt', 'Hsp90i39vsDMSO37-rep2_mutdiffsel.txt', 'Hsp90i39vsDMSO37-rep3_mutdiffsel.txt']\n",
      "\tstringencyparameter = None\n",
      "\tinfile = Hsp90i39vsDMSO37-rep2_mutdiffsel.txt\n",
      "\tletterheight = 2.0\n",
      "\n",
      "\n",
      "Logo plot will be written to Hsp90i39vsDMSO37-rep2_diffsel_logoplot.pdf\n",
      "Removing existing file Hsp90i39vsDMSO37-rep2_diffsel_logoplot.pdf\n",
      "\n",
      "Reading data from Hsp90i39vsDMSO37-rep2_mutdiffsel.txt\n",
      "\n",
      "Now making the logo plot in file Hsp90i39vsDMSO37-rep2_diffsel_logoplot.pdf\n",
      "\n",
      "Successfully completed dms_logoplot at Tue Mar 27 19:03:38 2018\n"
     ]
    }
   ],
   "source": [
    "#rep2 Hsp90i 39C vs DMSO 37C\n",
    "!dms_logoplot Hsp90i39vsDMSO37-rep2_mutdiffsel.txt Hsp90i39vsDMSO37-rep2_diffsel_logoplot.pdf --nperline 100 --mapmetric charge --letterheight 2 --diffselheight DMSO39vsDMSO37-rep1_mutdiffsel.txt DMSO39vsDMSO37-rep2_mutdiffsel.txt DMSO39vsDMSO37-rep3_mutdiffsel.txt HSF1i37vsDMSO37-rep1_mutdiffsel.txt HSF1i37vsDMSO37-rep2_mutdiffsel.txt HSF1i37vsDMSO37-rep3_mutdiffsel.txt HSF1i39vsDMSO37-rep1_mutdiffsel.txt HSF1i39vsDMSO37-rep2_mutdiffsel.txt HSF1i39vsDMSO37-rep3_mutdiffsel.txt HSF1i39vsDMSO39-rep1_mutdiffsel.txt HSF1i39vsDMSO39-rep2_mutdiffsel.txt HSF1i39vsDMSO39-rep3_mutdiffsel.txt HSF1i39vsHSF1i37-rep1_mutdiffsel.txt HSF1i39vsHSF1i37-rep2_mutdiffsel.txt HSF1i39vsHSF1i37-rep3_mutdiffsel.txt Hsp90i37vsDMSO37-rep1_mutdiffsel.txt Hsp90i37vsDMSO37-rep2_mutdiffsel.txt Hsp90i39vsDMSO37-rep3_mutdiffsel.txt Hsp90i39vsDMSO37-rep1_mutdiffsel.txt Hsp90i39vsDMSO37-rep2_mutdiffsel.txt Hsp90i39vsDMSO37-rep3_mutdiffsel.txt  "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 53,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Beginning execution of dms_logoplot in directory /Users/annapon/Documents/NP_DMS_2018/2018_NP_DMS/preferences\n",
      "\n",
      "Version information for dms_tools and associated packages.\n",
      "\tTime and date: Tue Mar 27 19:03:41 2018\n",
      "\tPlatform: Darwin-15.6.0-x86_64-i386-64bit\n",
      "\tPython version: 2.7.13 |Anaconda 4.4.0 (x86_64)| (default, Dec 20 2016, 23:05:08)  [GCC 4.2.1 Compatible Apple LLVM 6.0 (clang-600.0.57)]\n",
      "\tdms_tools version: 1.2.2\n",
      "\tBio version: 1.70\n",
      "\tnumpy version: 1.12.1\n",
      "\tscipy version: 0.19.0\n",
      "\tmatplotlib version: 2.0.2\n",
      "\tcython version: 0.25.2\n",
      "\tpystan version: 2.16.0.0\n",
      "\tweblogolib version: 3.5.0\n",
      "\tPyPDF2 version: 1.26.0\n",
      "\tpandas version: 0.20.1\n",
      "\n",
      "Parsed the following arguments:\n",
      "\trestrictdiffsel = None\n",
      "\toverlay3 = None\n",
      "\tcolormap = jet\n",
      "\tnperline = 100\n",
      "\tnumberevery = 10\n",
      "\texcludestop = False\n",
      "\tlogoplot = Hsp90i39vsDMSO37-rep3_diffsel_logoplot.pdf\n",
      "\toverlay_cmap = None\n",
      "\tmapmetric = charge\n",
      "\tdiffprefheight = 1.0\n",
      "\tnosepline = False\n",
      "\toverlay1 = None\n",
      "\toverlay2 = None\n",
      "\tdiffselheight = ['DMSO39vsDMSO37-rep1_mutdiffsel.txt', 'DMSO39vsDMSO37-rep2_mutdiffsel.txt', 'DMSO39vsDMSO37-rep3_mutdiffsel.txt', 'HSF1i37vsDMSO37-rep1_mutdiffsel.txt', 'HSF1i37vsDMSO37-rep2_mutdiffsel.txt', 'HSF1i37vsDMSO37-rep3_mutdiffsel.txt', 'HSF1i39vsDMSO37-rep1_mutdiffsel.txt', 'HSF1i39vsDMSO37-rep2_mutdiffsel.txt', 'HSF1i39vsDMSO37-rep3_mutdiffsel.txt', 'HSF1i39vsDMSO39-rep1_mutdiffsel.txt', 'HSF1i39vsDMSO39-rep2_mutdiffsel.txt', 'HSF1i39vsDMSO39-rep3_mutdiffsel.txt', 'HSF1i39vsHSF1i37-rep1_mutdiffsel.txt', 'HSF1i39vsHSF1i37-rep2_mutdiffsel.txt', 'HSF1i39vsHSF1i37-rep3_mutdiffsel.txt', 'Hsp90i37vsDMSO37-rep1_mutdiffsel.txt', 'Hsp90i37vsDMSO37-rep2_mutdiffsel.txt', 'Hsp90i39vsDMSO37-rep3_mutdiffsel.txt', 'Hsp90i39vsDMSO37-rep1_mutdiffsel.txt', 'Hsp90i39vsDMSO37-rep2_mutdiffsel.txt', 'Hsp90i39vsDMSO37-rep3_mutdiffsel.txt']\n",
      "\tstringencyparameter = None\n",
      "\tinfile = Hsp90i39vsDMSO37-rep3_mutdiffsel.txt\n",
      "\tletterheight = 2.0\n",
      "\n",
      "\n",
      "Logo plot will be written to Hsp90i39vsDMSO37-rep3_diffsel_logoplot.pdf\n",
      "Removing existing file Hsp90i39vsDMSO37-rep3_diffsel_logoplot.pdf\n",
      "\n",
      "Reading data from Hsp90i39vsDMSO37-rep3_mutdiffsel.txt\n",
      "\n",
      "Now making the logo plot in file Hsp90i39vsDMSO37-rep3_diffsel_logoplot.pdf\n",
      "\n",
      "Successfully completed dms_logoplot at Tue Mar 27 19:03:56 2018\n"
     ]
    }
   ],
   "source": [
    "#rep3 Hsp90i 39C vs DMSO 37C\n",
    "!dms_logoplot Hsp90i39vsDMSO37-rep3_mutdiffsel.txt Hsp90i39vsDMSO37-rep3_diffsel_logoplot.pdf --nperline 100 --mapmetric charge --letterheight 2 --diffselheight DMSO39vsDMSO37-rep1_mutdiffsel.txt DMSO39vsDMSO37-rep2_mutdiffsel.txt DMSO39vsDMSO37-rep3_mutdiffsel.txt HSF1i37vsDMSO37-rep1_mutdiffsel.txt HSF1i37vsDMSO37-rep2_mutdiffsel.txt HSF1i37vsDMSO37-rep3_mutdiffsel.txt HSF1i39vsDMSO37-rep1_mutdiffsel.txt HSF1i39vsDMSO37-rep2_mutdiffsel.txt HSF1i39vsDMSO37-rep3_mutdiffsel.txt HSF1i39vsDMSO39-rep1_mutdiffsel.txt HSF1i39vsDMSO39-rep2_mutdiffsel.txt HSF1i39vsDMSO39-rep3_mutdiffsel.txt HSF1i39vsHSF1i37-rep1_mutdiffsel.txt HSF1i39vsHSF1i37-rep2_mutdiffsel.txt HSF1i39vsHSF1i37-rep3_mutdiffsel.txt Hsp90i37vsDMSO37-rep1_mutdiffsel.txt Hsp90i37vsDMSO37-rep2_mutdiffsel.txt Hsp90i39vsDMSO37-rep3_mutdiffsel.txt Hsp90i39vsDMSO37-rep1_mutdiffsel.txt Hsp90i39vsDMSO37-rep2_mutdiffsel.txt Hsp90i39vsDMSO37-rep3_mutdiffsel.txt  "
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Generate logo plot with MxA selection"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Using the data from O. Ashenberg et al 2017 to compare with the differential selection observed in the present work."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 11,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "/Users/annapon/Documents/NP_DMS_2018/2018_NP_DMS/preferences\n",
      "\n",
      "Beginning execution of dms_logoplot in directory /Users/annapon/Documents/NP_DMS_2018/2018_NP_DMS/preferences\n",
      "\n",
      "Version information for dms_tools and associated packages.\n",
      "\tTime and date: Wed Mar 28 15:39:25 2018\n",
      "\tPlatform: Darwin-15.6.0-x86_64-i386-64bit\n",
      "\tPython version: 2.7.13 |Anaconda 4.4.0 (x86_64)| (default, Dec 20 2016, 23:05:08)  [GCC 4.2.1 Compatible Apple LLVM 6.0 (clang-600.0.57)]\n",
      "\tdms_tools version: 1.2.2\n",
      "\tBio version: 1.70\n",
      "\tnumpy version: 1.12.1\n",
      "\tscipy version: 0.19.0\n",
      "\tmatplotlib version: 2.0.2\n",
      "\tcython version: 0.25.2\n",
      "\tpystan version: 2.16.0.0\n",
      "\tweblogolib version: 3.5.0\n",
      "\tPyPDF2 version: 1.26.0\n",
      "\tpandas version: 0.20.1\n",
      "\n",
      "Parsed the following arguments:\n",
      "\trestrictdiffsel = None\n",
      "\toverlay3 = None\n",
      "\tcolormap = jet\n",
      "\tnperline = 100\n",
      "\tnumberevery = 10\n",
      "\texcludestop = False\n",
      "\tlogoplot = MxA_logoplot.pdf\n",
      "\toverlay_cmap = None\n",
      "\tmapmetric = charge\n",
      "\tdiffprefheight = 1.0\n",
      "\tnosepline = False\n",
      "\toverlay1 = None\n",
      "\toverlay2 = None\n",
      "\tdiffselheight = ['DMSO39vsDMSO37-rep1_mutdiffsel.txt', 'DMSO39vsDMSO37-rep2_mutdiffsel.txt', 'DMSO39vsDMSO37-rep3_mutdiffsel.txt', 'HSF1i37vsDMSO37-rep1_mutdiffsel.txt', 'HSF1i37vsDMSO37-rep2_mutdiffsel.txt', 'HSF1i37vsDMSO37-rep3_mutdiffsel.txt', 'HSF1i39vsDMSO37-rep1_mutdiffsel.txt', 'HSF1i39vsDMSO37-rep2_mutdiffsel.txt', 'HSF1i39vsDMSO37-rep3_mutdiffsel.txt', 'HSF1i39vsDMSO39-rep1_mutdiffsel.txt', 'HSF1i39vsDMSO39-rep2_mutdiffsel.txt', 'HSF1i39vsDMSO39-rep3_mutdiffsel.txt', 'HSF1i39vsHSF1i37-rep1_mutdiffsel.txt', 'HSF1i39vsHSF1i37-rep2_mutdiffsel.txt', 'HSF1i39vsHSF1i37-rep3_mutdiffsel.txt', 'Hsp90i37vsDMSO37-rep1_mutdiffsel.txt', 'Hsp90i37vsDMSO37-rep2_mutdiffsel.txt', 'Hsp90i39vsDMSO37-rep3_mutdiffsel.txt', 'Hsp90i39vsDMSO37-rep1_mutdiffsel.txt', 'Hsp90i39vsDMSO37-rep2_mutdiffsel.txt', 'Hsp90i39vsDMSO37-rep3_mutdiffsel.txt', 'MxA.txt']\n",
      "\tstringencyparameter = None\n",
      "\tinfile = MxA.txt\n",
      "\tletterheight = 2.0\n",
      "\n",
      "\n",
      "Logo plot will be written to MxA_logoplot.pdf\n",
      "\n",
      "Reading data from MxA.txt\n",
      "\n",
      "Now making the logo plot in file MxA_logoplot.pdf\n",
      "\n",
      "Successfully completed dms_logoplot at Wed Mar 28 15:39:41 2018\n"
     ]
    }
   ],
   "source": [
    "#MxA logo plot with height scaled to perturbed proteostasis selection\n",
    "%cd preferences\n",
    "!dms_logoplot MxA.txt MxA_logoplot.pdf --nperline 100 --mapmetric charge --letterheight 2 --diffselheight DMSO39vsDMSO37-rep1_mutdiffsel.txt DMSO39vsDMSO37-rep2_mutdiffsel.txt DMSO39vsDMSO37-rep3_mutdiffsel.txt HSF1i37vsDMSO37-rep1_mutdiffsel.txt HSF1i37vsDMSO37-rep2_mutdiffsel.txt HSF1i37vsDMSO37-rep3_mutdiffsel.txt HSF1i39vsDMSO37-rep1_mutdiffsel.txt HSF1i39vsDMSO37-rep2_mutdiffsel.txt HSF1i39vsDMSO37-rep3_mutdiffsel.txt HSF1i39vsDMSO39-rep1_mutdiffsel.txt HSF1i39vsDMSO39-rep2_mutdiffsel.txt HSF1i39vsDMSO39-rep3_mutdiffsel.txt HSF1i39vsHSF1i37-rep1_mutdiffsel.txt HSF1i39vsHSF1i37-rep2_mutdiffsel.txt HSF1i39vsHSF1i37-rep3_mutdiffsel.txt Hsp90i37vsDMSO37-rep1_mutdiffsel.txt Hsp90i37vsDMSO37-rep2_mutdiffsel.txt Hsp90i39vsDMSO37-rep3_mutdiffsel.txt Hsp90i39vsDMSO37-rep1_mutdiffsel.txt Hsp90i39vsDMSO37-rep2_mutdiffsel.txt Hsp90i39vsDMSO37-rep3_mutdiffsel.txt MxA.txt "
   ]
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 2",
   "language": "python",
   "name": "python2"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 2
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython2",
   "version": "2.7.13"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 2
}
