{
 "cells": [
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Analysis of deep mutational scanning of influenza A/Aichi/1968 (H3N2) nucleoprotein in perturbed host proteostasis environments"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Anna Ponomarenko, March 2018"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Load required modules"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "import os\n",
    "import glob\n",
    "import sys\n",
    "import string\n",
    "import time\n",
    "import multiprocessing\n",
    "\n",
    "import pandas as pd\n",
    "\n",
    "basedir = os.getcwd()\n",
    "if not os.path.isdir('notebook_figures'):\n",
    "    !mkdir notebook_figures\n",
    "if not os.path.isdir('FASTQ_files'):\n",
    "    !mkdir FASTQ_files\n",
    "if not os.path.isdir('preferences'):\n",
    "    !mkdir preferences\n",
    "    \n",
    "import matplotlib\n",
    "print \"Using matplotlib version %s\" % matplotlib.__version__\n",
    "matplotlib.use(\"Pdf\")\n",
    "orange = \"#E69F00\"\n",
    "lightblue = \"#56B4E9\"\n",
    "green = \"#009E73\"\n",
    "yellow = \"#F0E442\"\n",
    "darkblue = \"#0072B2\"\n",
    "rust =  \"#D55E00\"\n",
    "purple =  \"#CC79A7\"\n",
    "from matplotlib.ticker import MaxNLocator\n",
    "from matplotlib.patches import Polygon\n",
    "from matplotlib import gridspec\n",
    "import pylab as plt\n",
    "\n",
    "from IPython.display import Image, display\n",
    "\n",
    "import dms_tools\n",
    "print \"Using dms_tools version %s\" % dms_tools.__version__\n",
    "import dms_tools.utils\n",
    "import dms_tools.file_io\n",
    "import dms_tools.plot"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Define some functions and variables used below by multiple cells"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "def RunScriptDms(rundir, run_name, script_name, commands, use_sbatch, sbatch_cpus, walltime=None):\n",
    "    \"\"\"Runs a ``dmstools`` script.\n",
    "\n",
    "    *rundir* is the directory in which we run the job. Created if it does\n",
    "    not exist.\n",
    "\n",
    "    *run_name* is the name of the run, which should be a string without\n",
    "    spaces. The input file has this prefix followed by ``_infile.txt``.\n",
    "\n",
    "    *script_name* is the name of the script that we run.\n",
    "\n",
    "    *commands* is a list of command-line arguments. These arguments should appear \n",
    "    in the order in which one would place them on the command line. The elements in this\n",
    "    list are joined together by spaces, and given to *script_name*. All elements\n",
    "    should be strings.\n",
    "\n",
    "    *use_sbatch* is a Boolean switch specifying whether we use ``sbatch``\n",
    "    to run the script. If *False*, the script is just run with the command\n",
    "    line instruction. If *True*, then ``sbatch`` is used, and the command file\n",
    "    has the prefix *run_name* followed by the suffix ``.sbatch``.\n",
    "\n",
    "    *sbatch_cpus* is an option that is only meaningful if *use_sbatch* is \n",
    "    *True*. It gives the integer number of CPUs that are claimed via\n",
    "    ``sbatch`` using the option ``sbatch -c``. \n",
    "\n",
    "    *walltime* is an option that is only meaningful if *use_sbatch* is\n",
    "    *True*. If so, it should be an integer giving the number of hours \n",
    "    to allocate for the job. If *walltime* has its default value of \n",
    "    *None*, no wall time for the job is specified.\n",
    "\n",
    "    It is assumed that the script can be run at the command line using::\n",
    "\n",
    "        script_name infile\n",
    "\n",
    "    Returns *runfailed*: *True* if run failed, and *False* otherwise.\n",
    "    \"\"\"\n",
    "    print \"Running %s for %s in directory %s...\" % (script_name, run_name, rundir)\n",
    "    currdir = os.getcwd()\n",
    "    if not os.path.isdir(rundir):\n",
    "        os.mkdir(rundir)\n",
    "    os.chdir(rundir)\n",
    "    if (not run_name) or not all([x not in string.whitespace for x in run_name]):\n",
    "        raise ValueError(\"Invalid run_name of %s\" % run_name)\n",
    "    commandline = ' '.join(['%s' % command for command in commands])\n",
    "    if use_sbatch:\n",
    "        sbatchfile = '%s.sbatch' % run_name # sbatch command file\n",
    "        jobidfile = 'sbatch_%s_jobid' % run_name # holds sbatch job id\n",
    "        jobstatusfile = 'sbatch_%s_jobstatus' % run_name # holds sbatch job status\n",
    "        joberrorsfile = 'sbatch_%s_errors' % run_name # holds sbatch job errors\n",
    "        sbatch_f = open(sbatchfile, 'w')\n",
    "        sbatch_f.write('#!/bin/sh\\n#SBATCH\\n')\n",
    "        if walltime:\n",
    "            sbatch_f.write('#PBS -l walltime=%d:00:00\\n' % walltime)\n",
    "        sbatch_f.write('%s %s' % (script_name, commandline))\n",
    "        sbatch_f.close()\n",
    "        os.system('sbatch -c %d -e %s %s > %s' % (sbatch_cpus, joberrorsfile, sbatchfile, jobidfile))\n",
    "        time.sleep(60) # short 1 minute delay\n",
    "        jobid = int(open(jobidfile).read().split()[-1])\n",
    "        nslurmfails = 0\n",
    "        while True:\n",
    "            time.sleep(60) # delay 1 minute\n",
    "            returncode = os.system('squeue -j %d > %s' % (jobid, jobstatusfile))\n",
    "            if returncode != 0:\n",
    "                nslurmfails += 1\n",
    "                if nslurmfails > 180: # error if squeue fails at least 180 consecutive times\n",
    "                    raise ValueError(\"squeue is continually failing, which means that slurm is not working on your system. Note that although this script has crashed, many of the jobs submitted via slurm may still be running. You'll want to monitor (squeue) or kill them (scancel) -- unfortunately you can't do that until slurm starts working again.\")\n",
    "                continue # we got an error while trying to run squeue\n",
    "            nslurmfails = 0\n",
    "            lines = open(jobstatusfile).readlines()\n",
    "            if len(lines) < 2:\n",
    "                break # no longer in slurm queue\n",
    "        errors = open(joberrorsfile).read().strip()\n",
    "    else:\n",
    "        errors = os.system('%s %s' % (script_name, commandline))\n",
    "    os.chdir(currdir)\n",
    "    if errors:\n",
    "        print \"ERROR running %s for %s in directory %s.\" % (script_name, run_name, rundir)\n",
    "        return True\n",
    "    else:\n",
    "        print \"Successfully completed running %s for %s in directory %s.\" % (script_name, run_name, rundir)\n",
    "        return False\n",
    "\n",
    "    \n",
    "def RunProcesses(processes, nmultiruns):\n",
    "    \"\"\"Runs a list *multiprocessing.Process* processes.\n",
    "\n",
    "    *processes* is a list of *multiprocessing.Process* objects that\n",
    "    have not yet been started. If an empty list, then just returns \n",
    "    with no action taken.\n",
    "\n",
    "    *nmultiruns* is an integer >= 1 indicating the number of simultaneous\n",
    "    processes to run.\n",
    "\n",
    "    Runs the processes in *processes*, making sure to never have more than\n",
    "    *nmultiruns* running at a time. If any of the processes fail (return\n",
    "    an exitcode with a boolean value other than *False*), an exception\n",
    "    is raised immediately. Otherwise, this function finishes when all\n",
    "    processes have completed.\n",
    "    \"\"\"\n",
    "    if not processes:\n",
    "        return\n",
    "    if not (nmultiruns >= 1 and isinstance(nmultiruns, int)):\n",
    "        raise ValueError(\"nmultiruns must be an integer >= 1\")\n",
    "    processes_started = [False] * len(processes)\n",
    "    processes_running = [False] * len(processes)\n",
    "    processes_finished = [False] * len(processes)\n",
    "    while not all(processes_finished):\n",
    "        if (processes_running.count(True) < nmultiruns) and not all(processes_started):\n",
    "            i = processes_started.index(False)\n",
    "            processes[i].start()\n",
    "            processes_started[i] = True\n",
    "            processes_running[i] = True\n",
    "        for i in range(len(processes)):\n",
    "            if processes_running[i]:\n",
    "                if not processes[i].is_alive():\n",
    "                    processes_running[i] = False\n",
    "                    processes_finished[i] = True\n",
    "                    if processes[i].exitcode:\n",
    "                        raise IOError(\"One of the processes failed to complete.\")\n",
    "        time.sleep(60)\n",
    "\n",
    "max_cpus = 1\n",
    "# Do we use sbatch when appropriate? Some simple operations will still\n",
    "# be run without sbatch even when True\n",
    "use_sbatch = False\n",
    "# Specify reference sequence to map reads to, numbering must follow what is specified by alignspecs\n",
    "refseq = '%s/Aichi_NP_reference.fasta' % basedir\n",
    "\n",
    "# Specify the biological replicates and sample names. Sample names are stored as list of 2-tuples or in a dictionary\n",
    "replicates = ['replicate-1', 'replicate-2']\n",
    "replicate_sample = [('replicate-1', 'plasmid'),('replicate-1', 'p0mut'),('replicate-1', 'WT37'),('replicate-1', 'DMSO37'),('replicate-1', 'dnHSF137'), ('replicate-1', '17AAG37'), ('replicate-1', 'PU29F37'),('replicate-1', 'WT39'),('replicate-1', 'DMSO39'),('replicate-1', 'dnHSF139'), ('replicate-1', '17AAG39'), ('replicate-1', 'PU29F39'),('replicate-2', 'plasmid'),('replicate-2', 'p0mut'),('replicate-2', 'WT37'),('replicate-2', 'DMSO37'),('replicate-2', 'dnHSF137'), ('replicate-2', '17AAG37'), ('replicate-2', 'PU29F37'),('replicate-2', 'WT39'),('replicate-2', 'DMSO39'),('replicate-2', 'dnHSF139'), ('replicate-2', '17AAG39'), ('replicate-2', 'PU29F39')]\n",
    "#, ('replicate-1', 'plasmid'),('replicate-1', 'WTplasmid')]\n",
    "replicate_sample_dict = dict([ (replicate, []) for replicate in replicates])\n",
    "for replicate, sample in replicate_sample:\n",
    "    replicate_sample_dict[replicate].append(sample)\n",
    "        \n",
    "print(\"Shared functions and variables are now defined.\")"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Create reference file"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "%%writefile $refseq\n",
    ">gi|392340231|gb|CY121120.1| Influenza A virus (A/Aichi/2/1968(H3N2)) nucleocapsid protein (NP) gene, complete cds\n",
    "ATGGCGTCCCAAGGCACCAAACGGTCTTATGAACAGATGGAAACTGATGGGGAACGCCAGAATGCAACTGAGATCAGAGCATCCGTCGGGAAGATGATTGATGGAAT\n",
    "TGGACGATTCTACATCCAAATGTGCACTGAACTTAAACTCAGTGATTATGAGGGGCGACTGATCCAGAAC\n",
    "AGCTTAACAATAGAGAGAATGGTGCTCTCTGCTTTTGACGAAAGAAGGAATAAATATCTGGAAGAACATC\n",
    "CCAGCGCGGGGAAGGATCCTAAGAAAACTGGAGGACCCATATACAAGAGAGTAGATAGAAAGTGGATGAG\n",
    "GGAACTCGTCCTTTATGACAAAGAAGAAATAAGGCGAATCTGGCGCCAAGCCAATAATGGTGATGATGCA\n",
    "ACAGCTGGTCTGACTCACATGATGATCTGGCATTCCAATTTGAATGATACAACATACCAGAGGACAAGAG\n",
    "CTCTTGTTCGCACCGGCATGGATCCCAGGATGTGCTCTCTGATGCAGGGTTCGACTCTCCCTAGGAGGTC\n",
    "TGGAGCTGCAGGCGCTGCAGTCAAAGGAGTTGGGACAATGGTGATGGAGTTGATAAGGATGATCAAACGT\n",
    "GGGATCAATGATCGGAACTTCTGGAGAGGTGAAAATGGACGAAAAACAAGGAGTGCTTACGAGAGAATGT\n",
    "GCAACATTCTCAAAGGAAAATTTCAAACAGCTGCACAAAGGGCAATGATGGATCAAGTGAGAGAAAGTCG\n",
    "GAACCCAGGAAATGCTGAGATCGAAGATCTCATCTTTCTGGCACGGTCTGCACTCATATTGAGAGGGTCA\n",
    "GTTGCTCACAAATCTTGTCTGCCCGCCTGTGTGTATGGACCTGCCGTAGCCAGTGGCTACGACTTCGAAA\n",
    "AAGAGGGATACTCTTTAGTGGGAATAGACCCTTTCAAACTGCTTCAAAACAGCCAAGTATACAGCCTAAT\n",
    "CAGACCGAACGAGAATCCAGCACACAAGAGTCAGCTGGTGTGGATGGCATGCAATTCTGCTGCATTTGAA\n",
    "GATCTAAGAGTATTAAGCTTCATCAGAGGGACCAAAGTATCCCCAAGGGGGAAACTTTCCACTAGAGGAG\n",
    "TACAAATTGCTTCAAATGAAAACATGGATGCTATGGAATCAAGTACTCTTGAACTGAGAAGCAGGTACTG\n",
    "GGCCATAAGAACCAGAAGTGGAGGAAACACTAATCAACAGAGGGCCTCTGCAGGTCAAATCAGTGTGCAA\n",
    "CCTGCATTTTCTGTGCAAAGAAACCTCCCATTTGACAAACCAACCATCATGGCAGCATTCACTGGGAATA\n",
    "CAGAGGGAAGAACATCAGACATGAGGGCAGAAATTATAAGGATGATGGAAGGTGCAAAACCAGAAGAAAT\n",
    "GTCCTTCCAGGGGCGGGGAGTCTTCGAGCTCTCGGACGAAAGGGCAGCGAACCCGATCGTGCCCTCTTTT\n",
    "GACATGAGTAATGAAGGATCTTATTTCTTCGGAGACAATGCAGAGGAGTACGACAATTAA\n"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Process and analyze deep sequencing data"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Align paired-end reads"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "def main():\n",
    "    \"\"\"Main body of script.\"\"\"\n",
    "\n",
    "    # Specify R1 and R2 read lengths and gene range of each subamplicon for dms_barcodedsubamplicons\n",
    "    alignspecs = '1,252,34,32 253,501,38,30 502,750,30,36 751,1002,32,41 1003,1251,29,34 1252,1497,34,41'\n",
    "\n",
    "    # Specify range of residues to consider\n",
    "    sitesrange = ' '.join(['%s' %i for i in range(1,499)]) # range of residues to consider\n",
    "\n",
    "    # Set R1 and R2 trim lengths for dms_barcodedsubamplicons\n",
    "    R1trimlength = 200\n",
    "    R2trimlength = 160\n",
    "\n",
    "    # Group all barcodes with this many reads or more for dms_summarizealignments\n",
    "    maxperbarcode = 4\n",
    " \n",
    "   # Specify for which mutvir libraries to infer amino-acid preferences\n",
    "    # I will not be running this since I didn't sequence the WT viruses\n",
    "    # prefs_mutvir = {'replicate-1': ['mutvir', 'mutvir-MS-1A', 'mutvir-MS-1B', 'mutvir-MxA-1A', 'mutvir-MxA-1B'], \\\n",
    "                      #'replicate-2': ['mutvir', 'mutvir-MS', 'mutvir-MxA']}\n",
    "\n",
    "    # Specify which mutvir libraries to use as control selection and actual selection to infer differential amino-acid preferences\n",
    "    # I will be doing this, we want to infer differential AA preferences between folding environments\n",
    "    controls = {'replicate-1': ['Sample1', 'Sample3', 'Sample5'], 'replicate-2': ['Sample9', 'Sample11', 'Sample13']}\n",
    "    selections = {'replicate-1': ['Sample2', 'Sample4', 'Sample6'], 'replicate-2': ['Sample10', 'Sample12', 'Sample14']}\n",
    "\n",
    "    # Directory path for FASTQ files for each replicate and amplicon\n",
    "    FASTQ_path = '%s/FASTQ_files/' % basedir\n",
    "    \n",
    "    print \"Found FASTQ_path\"\n",
    "\n",
    "    # Specify which pairs of mutvir libraries to get correlation of amino-acid preferences between replicate-1 and replicate-2 as 2-tuples\n",
    "    # I dont think im running this since im not running prefs_mutvir\n",
    "    #correlate_prefs_mutvir = ( ('mutvir', 'mutvir'), ('mutvir-MS-1A', 'mutvir-MS'), ('mutvir-MS-1B', 'mutvir-MS'), \\\n",
    "                              # ('mutvir-MxA-1A', 'mutvir-MxA'), ('mutvir-MxA-1B', 'mutvir-MxA'))\n",
    "\n",
    "    # Specify location of preference files from older sequencing study. These will get compared to current sequencing study.\n",
    "    # not relevant for me, i think.\n",
    "    # old_basedir = '/fh/fast/bloom_j/grp/mapmuts/examples/2014Analysis_Influenza_NP_MxA'\n",
    "    #files_old_prefs_mutvir = [('replicate-%s' %i, '%s/replicate_%s/preferences_mutvir/replicate_%s_mutvir_equilibriumpreferences.txt' % (old_basedir, i, i)) for i in ('1', '2')]\n",
    "    #print files_old_prefs_mutvir\n",
    "\n",
    "    # Toggle which scripts to run or skip:\n",
    "    # however, this script is not smart enough to know if the existing output \n",
    "    # exists necessary to run the next strip exist - so use with caution!\n",
    "    run_barcodedsubamplicons = True\n",
    "    run_summarizealignments = False\n",
    "\n",
    "# again, I'm not infering prefs\n",
    "    #run_inferprefs = False\n",
    "    #run_prefs_editsites = False\n",
    "    #run_prefs_merge = False\n",
    "    #run_prefs_correlate = False\n",
    "    #run_prefs_logoplot = False\n",
    "\n",
    "    run_inferdiffprefs = False\n",
    "    run_diffprefs_editsites = False\n",
    "    run_diffprefs_merge = False\n",
    "    run_diffprefs_correlate = False\n",
    "    run_diffprefs_logoplot = False\n",
    "    #########################################\n",
    "    #########################################\n",
    "    # Now run dms_barcodedsubamplicons\n",
    "    # Each sample is run its own subdirectory\n",
    "    #########################################\n",
    "    #########################################\n",
    "    if run_barcodedsubamplicons:\n",
    "        print \"entering barcodedsubamplicons section of masterscript\"\n",
    "\n",
    "        processes = []\n",
    "\n",
    "        # Store (r1files, r2files) in dictionary keyed by (replicate, sample)\n",
    "        fastqfiles = {} \n",
    "        for replicate, sample in replicate_sample:\n",
    "            r1files = sorted(glob.glob('%s/%s/%s/%s_1.fastq' % (FASTQ_path, replicate, sample, sample)))\n",
    "            r2files = sorted(glob.glob('%s/%s/%s/%s_2.fastq' % (FASTQ_path, replicate, sample, sample)))\n",
    "            assert r1files and r2files\n",
    "            fastqfiles[(replicate, sample)] = (r1files, r2files)\n",
    "            #print replicate, sample, r1files, r2files\n",
    "\n",
    "            if not os.path.isdir(replicate):\n",
    "                os.mkdir(replicate)\n",
    "            if not os.path.isdir(\"%s/%s\" % (replicate, sample)):\n",
    "                os.mkdir(\"%s/%s\" % (replicate, sample))\n",
    "\n",
    "        # Make the barcoded alignments for each sample\n",
    "        for replicate, sample in replicate_sample:\n",
    "            subdir = '%s/%s' % (replicate, sample)\n",
    "            alignprefix = '%s_' % sample\n",
    "            (r1files, r2files) = fastqfiles[(replicate, sample)]\n",
    "            commands = ['--R1trimlength %s' % R1trimlength, '--R2trimlength %s' % R2trimlength, alignprefix, refseq, ','.join(r1files), ','.join(r2files), alignspecs]\n",
    "\n",
    "            # Add this makealignments.py call to the list of processes.\n",
    "            # Use *list* to make a copy of *commands* since lists are mutable on subsequent runs\n",
    "            # of making replicate-amplicon combinations.\n",
    "            processes.append(multiprocessing.Process(target=RunScriptDms, args=(subdir, 'barcodedsubamplicons', 'dms_barcodedsubamplicons', list(commands), use_sbatch, 1)))\n",
    "    \n",
    "\n",
    "        # run ALL the barcodedsubamplicons processes\n",
    "        RunProcesses(processes, nmultiruns=max_cpus)\n",
    "\n",
    "        # end barcodedsubamplicons block of masterscript\n",
    "    else:\n",
    "        print \"Skipping barcodedsubamplicons section of masterscript\"\n",
    "        \n",
    "if __name__ == '__main__':\n",
    "    main() # run the script"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Summarize alignments"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "def main():\n",
    "    \"\"\"Main body of script.\"\"\"\n",
    "\n",
    "    # Group all barcodes with this many reads or more for dms_summarizealignments\n",
    "    maxperbarcode = 4\n",
    "\n",
    "    # Directory path for FASTQ files for each replicate and amplicon\n",
    "    FASTQ_path = '%s/FASTQ_files/' % basedir\n",
    "\n",
    "\n",
    "    # Run summarizealignments on individual samples\n",
    "    for replicate, sample in replicate_sample:\n",
    "        subdir = '%s/%s' % (replicate, sample)\n",
    "        alignprefix_name = '%s_,%s' % (sample, sample)\n",
    "        outprefix = 'alignmentsummary_'\n",
    "        commands = ['--maxperbarcode %s' % maxperbarcode, outprefix, 'barcodedsubamplicons', alignprefix_name]            #print commands\n",
    "#        RunScriptDms(subdir, 'summarizealignments', 'dms_summarizealignments', commands, False, 1)\n",
    "\n",
    "    # Run summarizealignments on all samples from a replicate     \n",
    "    for replicate in replicates:\n",
    "        alignprefixes_names = []\n",
    "        subdir = replicate\n",
    "        outprefix = 'alignmentsummary_'\n",
    "        for sample in replicate_sample_dict[replicate]:\n",
    "            if replicate == 'replicate-1' and sample != 'DNA':   # only 7 alignments can be summarized currently\n",
    "                alignprefix = '%s/%s_' % (sample, sample)\n",
    "                alignprefixes_names.append(\"%s,%s\" % (alignprefix, sample))\n",
    "            if replicate != 'replicate-1':\n",
    "                alignprefix = '%s/%s_' % (sample, sample)\n",
    "                alignprefixes_names.append(\"%s,%s\" % (alignprefix, sample))\n",
    "        commands = ['--maxperbarcode %s' % maxperbarcode, outprefix, 'barcodedsubamplicons', ' '.join(alignprefixes_names)]\n",
    "        print commands\n",
    "        RunScriptDms(subdir, 'summarizealignments', 'dms_summarizealignments', commands, False, 1)\n",
    "        \n",
    "if __name__ == '__main__':\n",
    "    main() # run the script"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Plot mutational and error rates, depth of coverage"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "from IPython.display import Image, display\n",
    "\n",
    "filepath = 'replicate-1/alignmentsummary_barcodes.pdf'\n",
    "\n",
    "png = filepath.rstrip('.pdf') + '.png'\n",
    "!convert -density 192 -trim $filepath $png\n",
    "display(Image(png, width=500))\n",
    "\n",
    "filepath = 'replicate-1/alignmentsummary_depth.pdf'\n",
    "\n",
    "png = filepath.rstrip('.pdf') + '.png'\n",
    "!convert -density 192 -trim $filepath $png\n",
    "display(Image(png, width=500))\n",
    "\n",
    "filepath = 'replicate-1/alignmentsummary_mutcounts_all.pdf'\n",
    "\n",
    "png = filepath.rstrip('.pdf') + '.png'\n",
    "!convert -density 192 -trim $filepath $png\n",
    "display(Image(png, width=500))\n",
    "\n",
    "filepath = 'replicate-1/alignmentsummary_mutcounts_multi_nt.pdf'\n",
    "\n",
    "png = filepath.rstrip('.pdf') + '.png'\n",
    "!convert -density 192 -trim $filepath $png\n",
    "display(Image(png, width=500))\n",
    "\n",
    "filepath = 'replicate-1/alignmentsummary_mutdepth.pdf'\n",
    "\n",
    "png = filepath.rstrip('.pdf') + '.png'\n",
    "!convert -density 192 -trim $filepath $png\n",
    "display(Image(png, width=500))\n",
    "\n",
    "filepath = 'replicate-1/alignmentsummary_mutfreqs.pdf'\n",
    "\n",
    "png = filepath.rstrip('.pdf') + '.png'\n",
    "!convert -density 192 -trim $filepath $png\n",
    "display(Image(png, width=500))\n",
    "\n",
    "filepath = 'replicate-1/alignmentsummary_reads.pdf'\n",
    "\n",
    "png = filepath.rstrip('.pdf') + '.png'\n",
    "!convert -density 192 -trim $filepath $png\n",
    "display(Image(png, width=500))"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Calculate differential selection for each mutation and each site in NP"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Run dms_diffselection on all perturbed proteostasis environments relative to basal conditions at 37 C for each of three replicates."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "#REPLICATE 1: -/+ 17AAG 37C\n",
    "%cd preferences\n",
    "!ls\n",
    "\n",
    "!dms_diffselection Rep1-DMSO-37_counts.txt Rep1-17AAG-37_counts.txt 17AAG-rep1-37_"
   ]
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 2",
   "language": "python",
   "name": "python2"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 2
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython2",
   "version": "2.7.13"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 2
}
